# Results

  5. SGC R10 metapangenome graphs
  6. Dominating set differential abundance analysis 

We developed a computational approach to resolve sub-species level differences between groups of short read shotgun metagenomes (**Figure @fig:overview**).
While our pipeline relies on many published algorithms, we developed two new approaches that, when combined with existing tools, generated insights into microbial sequences associated with IBD subtype.
After consistent pre-processing, we used FracMinHash sketching to produce subsampled k-mer abundance profiles of metagenomes that reflected the sequence diversity in a sample [@doi:10.12688/f1000research.19675.1] (CITE: gather), and used these profiles to perform metagenome-wide k-mer association with IBD subtype. 
We refer to FracMinHash sketches as *signatures*, and for simplicity, continue referring to the sub-sampled k-mers in a signature as *k-mers*.
Retaining only k-mers associated with IBD, we used the minimum set cover approach to identify the genomes that best encompassed these k-mers.

Next, we developed an approach to perform differential abundance analysis directly on assembly graphs in order to recover all sequences that were more or less abundant in each IBD subtype when compared to nonIBD.
Using the genomes identified by our k-mer association analysis, we first performed assembly graph genome queries to recover all sequences associated with a given species within a metagenome.
For each genome query, we combined these sequences into a single assembly graph, which we refer to as a *metapangenome graph*. 
We estimated the abundance of each piece in this graph within each metagenome, and used these abundances to perform differential abundance analysis.

![
**Overview of the metagenome analysis technique used in this paper.** Steps that are outlined in grey were developed in this paper. **Step 1:** Using quality controlled sequencing reads from many metagenomes, we decomposed reads into k-mers and subsample these k-mers using FracMinHash, thereby selecting k-mers that evenly represent the sequence diversity within a sample. We then identified interesting k-mers using random forests, and associate these k-mers with genomes in reference databases. **Step 2:** For each metagenome, we constructed a compact de Bruijn assembly graph that contains all k-mers from a metagenome. We used dominating sets to carve the graph into pieces. We queried this graph with genomes associated with interesting k-mers identified in Step 1, recovering sequence diversity nearby in the assembly graph. We refer to these sequences as genome query neighborhoods. Step 2 is the workflow published in [@doi:10.1186/s13059-020-02066-4]. **Step 3:** We combined genome query neighborhoods for a single genome from all metagenomes. We constructed a compact de Bruijn assembly graph from these sequences, and used a dominating set with a large radius to carve the graph into large pieces. Here, we diagram construction of R=2 dominating set pieces, but in practice we used R=10. We estimated the abundance of k-mers in each metagenome for each dominating set piece, and used these abundances to perform differential abundance analysis.
](images/ibd_overview_steps.svg "Pipeline overview"){#fig:overview}

We applied this approach to the analysis of IBD gut microbiomes via meta-analysis.
Meta-analyses have recently shown success in improving detection of microbial signatures of colorectal cancer [@doi:10.1038/s41591-019-0406-6; @doi:10.1038/s41591-019-0405-7; @doi:10.1128/mSystems.00332-18]. 
To this end, we identified studies that performed metagenomic sequencing of individuals with CD, UC, or nonIBD and combined these to perform a meta-analysis (**Table @tbl:cohorts**).
All studies profiled fecal gut microbiomes via Illumina shotgun metagenome sequencing.
Individuals were from five distinct countries and seven cohorts (**Table @tbl:cohorts**).
In many studies, samples were taken in time series to profile disease progression or individual response to treatment. 
In these cases we included the first sample in the time series so organized interventions would not skew our results.
In addition, many of the nonIBD samples, particularly those from the iHMP, profiled sick individuals that were not diagnosed with IBD, meaning some of these samples are not healthy controls.


| Cohort      |   Name        | Country      |Total   |CD      |UC      |nonIBD  | Reference |
|:------------|:-------------:|:------------:|:------:|:------:|:------:|:------:|:----------|
| iHMP        | IBDMDB                   | USA              | 106   | 50  | 30  | 26 | [@doi:10.1038/s41586-019-1237-9] |
| PRJEB2054   | MetaHIT                  | Denmark, Spain   | 124   | 4   | 21  | 99 | [@doi:10.1038/nature08821] |
| SRP057027   | NA                       | Canada, USA      | 112   | 87  | 0   | 25 | [@doi:10.1016/j.chom.2015.09.008] |
| PRJNA385949 | PRISM, STiNKi            | USA              | 17    | 9   | 5   | 3 | [@doi:10.1186/s13073-017-0490-5] |
| PRJNA400072 | PRISM, LLDeep, and NLIBD | USA, Netherlands | 218   | 87  | 76  | 55 | [@doi:10.1038/s41564-018-0306-4] |
| PRJNA237362 | RISK                     | North America    | 28    | 23  | 0   | 5 | [@doi:10.1016/j.chom.2014.02.005] | 
| Total       |                          |                  | 605   | 260 | 132 | 213| |
Table: Six IBD shotgun metagenome sequencing cohorts used in this meta-cohort analysis.
{#tbl:cohorts}

## K-mers are weakly predictive of IBD subtype

<!--I don't love these first two sentences-->
We first sought an approach to compare many metagenomes without data loss accompanied by mapping to incomplete reference databases or *de novo* analyses like assembly and binning.
We reasoned that FracMinHash sketches randomly subsample k-mers to allow comparisons, which may provide an unbiased approach to quickly compare across many metagenomes.
In total, we profiled 7,376,151 subsampled k-mers across all samples in all cohorts, representing approximately 14 billion total k-mers. 

We detected variation due to IBD diagnosis in k-mer profiles of gut metagenomes from different cohorts.
We calculated a pairwise distance matrix using angular distance between k-mer abundance profiles to assess sample diversity. 
We performed principle coordinate analysis and PERMANOVA with this distance matrix (**Figure @fig:kmers A**), using the variables study accession, diagnosis, library size, and number of k-mers observed in a sample (**Figure @fig:kmers A**).
Study accounted for highest variation, emphasizing that technical artifacts can introduce strong signals that may influence heterogeneity in results across IBD microbiome studies but that can be mitigated through meta-analysis [@doi:10.1038/s41591-019-0406-6].
The number of k-mers observed in a sample accounted for the second highest variation, possibly reflecting reduced diversity in stool metagenomes of CD and UC patients (reviewed in [@doi:10.1038/s41579-019-0213-6]). 
Diagnosis accounted for a significant amount of variation as well, indicating that there is a small but detectable signal of IBD subtype in stool metagenomes.

<!--
|Variable          |Jaccard distance | Angular distance|
|:-----------------|:---------------:|:---------------:|
| Number of k-mers | 9.9%           | 6.2% |
| Study accession  | 6.6%           | 13.5% |
| Diagnosis        | 6.2%           | 3.3% |
| Library size     | 0.009%        | 0.01% |
Table: Results from PERMANOVA performed on Jaccard and Angular distance matrices. Number of k-mers refers to the number of k-mers in a signature, while library size refers to the number of raw reads per sample. All test were significant at p < .001.
{#tbl:permanova}
-->

To evaluate whether the variation captured by diagnosis is predictive of IBD subtype, we built random forests classifiers to predict CD, UC, or nonIBD subtype.
Random forests is a supervised learning classification model that estimates how predictive k-mers are of IBD subtype, and weights individual k-mers as more or less predictive using a metric called variable importance.
To assess whether disease signatures generalize across study populations, we used a leave-one-study-out cross-validation approach where we built and optimized a classifier using five cohorts and validated on the sixth.
We built each model six times, using a different random seed each time, to hone in on cross-study and cross-model signal.
Given the high-dimensional structure of this data set (e.g. many more k-mers than metagenomes), we first used variable selection to narrow the set of predictive k-mers in the training set [@doi:10.1007/s11634-016-0276-4; @doi:10.1093/bib/bbx124].
Variable selection reduced the number of k-mers used in each model by two orders of magnitude, from 7,376,151 to 28,684-41,701 (mean = 35,673.1, sd = 4090.3) (**Figure @fig:kmers B**). 

Using this reduced set of k-mers, we optimized each random forests classifier on the training set, producing 36 optimized models. 
We validated each model on the left-out study.
The accuracy on the validation studies ranged from 49%-77% (**Figure @fig:kmers B**), outperforming a previously published model built on metagenomic data alone [@doi:10.1038/s41564-018-0306-4].


![
**Long nucleotide k-mers retain information about IBD subtype classification.** **A.** Principal coordinate analysis of distance matrices obtained from comparing FracMinHash signatures with abundances and PERMANOVA results that explain the variance. Number of k-mers refers to the number of k-mers in a signature, while library size refers to the number of raw reads per sample. All test were significant at p < .001. **B.** Random forests models built on FracMinHash signatures predicted IBD subtype better than chance. Variable selection reduced the number of k-mers used to build each model, and model performance varied by validation study.
](images/kmers.png "kmer results"){#fig:kmers}

To understand which genomes were responsible for disease signatures detected by our models, we anchored k-mers in the models against genomes in reference databases using sourmash gather (CITE: gather).
Sourmash gather determines the minimum set of genomes in a database necessary to cover all of the k-mers in a query (CITE: gather).
We used the GTDB rs202 representatives database, which contains bacterial and archaeal genomes, and the GenBank viral, fungal, and protozoa databases.
We found that a substantial fraction of genomes were shared between models, indicating there is a consistent biological signal captured among classifiers: of 3,889 total genomes detected across all classifiers, 360 genomes were shared between all classifiers (**Figure @fig:tree**).
The presence of shared k-mers between classifiers indicates that there is a weak but consistent biological signal in metagenomes for IBD subtype between cohorts.

K-mers that anchored to these shared genomes represented 65% of all k-mers used to build the optimized classifiers, but accounted for an outsize proportion of variable importance in the optimized classifiers.
After normalizing variable importance across classifiers, 76% of the total variable importance was held by these k-mers. 
These k-mers contribute a large fraction of predictive power for classification of IBD subtype, and the genomes in which they are found represent a microbial core that contains predictive power in IBD subtype classification.

Given that 360 genomes anchored the majority of k-mers and variable importance across all models, we were curious whether a smaller number of genomes could still retain the majority of variable importance. 
Limiting genomes to those that could hold at least 1% of the normalized variable importance, we found that 54 genomes accounted for 50% of the variable importance (**Figure @fig:tree**). 
We assumed these genomes represent the strongest candidates for discriminating IBD subtype and focused on them for the remainder of our analyses.

![
**Phylogenetic tree of 360 bacterial species that were predictive of IBD subtype in all models.** Tree was built from the GTDB rs202 tree with all tips except those represented by the 360 genomes removed. Tree tips are labelled by genomes that anchored at least 1% of the normalized variable importance. The inner ring annotates the rank of the genomes, with the genome holding the most normalized variable importance across models ranked as 1. The outer ring is the species name within the GTDB database.
](images/genome360.png "tree"){#fig:tree}

## Genome queries into metagenome assembly graphs recover neighborhoods of sequence variation and establish species umbrellas

While we were able to identify the majority of k-mers that were important for predicting IBD subtype, 26% of k-mers remained unannotated. 
We hypothesized that these k-mers represented strain variable sequences not in reference databases but belonging to species represented by annotated k-mers.
To test this hypothesis, we performed genome queries on assembly graphs of each metagenome using the 54 candidate genomes that discriminated IBD subtype (**Figure @fig:overview**).
Assembly graph genome queries recover sequences in a metagenome that match the query, as well as those that are nearby in the assembly graph, making queries akin to but more sensitive than read mapping against reference genomes (**Figure @fig:overview**) [@doi:10.1186/s13059-020-02066-4].
<!-- is more sensitive the right thing to say here? Should it be more general? -->
The resulting genome query neighborhood represents a species-level umbrella that contains all sequence variation contained in the metagenome for a query.

After performing genome queries, we re-anchored k-mers against the resulting query neighborhoods as well as the databases used previously. 
We observed that the percent of unassigned k-mers decreased from 26% to 8% (**Figure @fig:alluvial**), supporting our hypothesis that many of these k-mers are sequence variants belonging to species identified in k-mers important for predicting IBD subtype.
We further observed that many other k-mers previously anchored by other genomes were reassigned to the genome query neighborhoods (**Figure @fig:alluvial**).
This suggests that the genome queries create species umbrellas that represent sequence variation for the query genome itself as well as other closely related genomes that occur within a metagenome.
 
![
**Alluvial plot depicting the set of genomes that anchored k-mers that were important for predicting IBD subtype.** 
The blocks on the left represent the breakdown of k-mer assignations from greedy exact matching against databases alone, while the blocks on the right represent k-mer assignations after metagenome assembly graph queries.
](images/alluvial_percent_of_hashes.png){#fig:alluvial}

<!--
I think part of why this might occur is because a lot of the sequences that were important for predicting IBD subtype, some of which were not in reference databases, encoded marker genes, including 16s and 23s. But the reason I know this is from multifasta queries. I would really rather not get in to the whole 16s/23s marker gene stuff, as it now seems tangential to the dominating set differential abundance (e.g., I'm trying to minimize the random forests component, bc while it's cool from a discovery perspective, all I use it for now it to pick out candidate genomes to analyze more closely). Also, I don't want to go into all of this stuff bc a good 1/2 of the results were generated from the old catlases and I have no intention of regenerating all of those results. So. It feels good knowing that some of the stuff gets scooped in bc 16s/23s should organize stuff close together in graph land, but i think I can safely hide that insight from the reads and not feel too silly about it.
-->

## IBD gut microbiomes have decreased diversity in strict anaerobes that is punctuated by strain switches for some facultative anaerobes

After recovering all sequences in metagenomes in the neighborhoods of the species that discriminate IBD subtypes, we next sought to determine the specific genome segments that were differentially abundant in IBD. 
Differential abundance analysis is a common step in metagenome analysis, however it is typically applied to gene counts [@doi:10.1186/s12864-018-4637-6; @doi:10.1186/s12864-016-2386-y], which requires assembly or mapping prior to abundance estimation.
<!--TR: Also commonly applied to species counts. Should this be added? -->
To avoid assembly or mapping and the accompanying loss of reads [@doi:doi.org/10.1186/s13059-020-02066-4], we developed an abundance estimation approach that works directly on the assembly graphs, enabling differential abundance analysis from the assembly graph.
Our abundance estimation approach was based on *R*-dominating sets, an algorithm introduced in [@doi:10.1186/s13059-020-02066-4] that efficiently computes the dominating nodes in a cDBG so that every node is no more that distance *R* from a dominator.
The dominating set is used to carve the graph into pieces, each of which contains one dominating node. 
Here, we first build a species-level assembly graph that contains all genome query neighborhood sequences for a given genome from every metagenome, which we call a *metapangenome graph*.
We then partition the graph into pieces using a large radius (*R* = 10). 
The large radius carves the graph into pieces that average XX k-mers in size, enabling meaningful comparisons between groups; we aimed for a piece size similar to the average bacterial gene length of approximately 1000 base pairs [@doi:10.1093/dnares/dsn027]. 
We next estimated the abundance of each piece within each metagenome using average k-mer abundance.
We also annotated the graph pieces using XXX.
Using this information, we performed dominating set differential abundance analysis using corncob, a statistical package that tests for differential relative abundance in the presence of variable sequencing depth and excessive zeroes for unobserved observations [@doi:10.1214/19-aoas1283], conditions which occur in abundances from dominating sets. 

We applied this method for each of our genome queries, building 54 metapangenome graphs and performing dominating set differential abundance analysis on each. 
Given that corncob fits a model for each dominating set piece and therefore does not require abundance information for all pieces [@doi:10.1214/19-aoas1283], and given that we sought differences that characterized the majority of our samples within a group, we tested for differential abundance in pieces that occurred in at least 100 metagenomes.
On average, this condition was met in XX% of dominating set pieces.

![
**Dominating set differential abundance analysis revealed genome segments that were significantly different in CD and UC compared to nonIBD.** Results are organized by GTDB taxonomy, with a tree representing the 54 species and colored by family on the far left. The percent of dominating set pieces tested is labelled in grey, and the percent of significantly differentially abundant pieces are colored by increased (blue) or decreased (brown) abundance.
](images/gtdb_and_corncob_res.png){#fig:dda}

We found that the majority of species decreased in abundance in CD, and to a lesser extent, UC (**Figure @fig:dda**).
Many of these species are generally regarded as beneficial bacteria. 
For example, nine of the 54 species we investigated were *Faecalibacterium prausnitzii*, the phylogroups of which are separated in the GTDB taxonomy but combined into a single species in the NCBI taxonomy. 
*F. prausnitzii* is a key butyrate producer in the gut and plays a crucial role in reducing intestinal inflammation [@doi:10.1038/ismej.2016.176].
Similarly, *Acetatifactor* is a bile-acid producing bacteria associated with a healthy gut, but limited evidence has associated it with decreased abundance in IBD [@doi:10.1002/hep.29857].
These species, as well as others that decreased in abundance in IBD, are strictly anaerobic (**Figure @fig:dda**), so these observed trends are consistent with a shift toward oxidative stress during disease that is intolerable for many gut microbes [@doi:10.1038/ismej.2013.80].

<!--*Gemmiger formicilis* had the most dominating set pieces decrease in abundance in UC, and was also decreased in CD (**Figure @fig:dda**).
*G. formicilis* is a strictly anaerobic bacteria that produces both formic acid and butyric acid [@doi:10.1099/00207713-25-2-202]. 
See 10.3390/cells10113243 for new association with Gemmiger formicilis being reduced in pouchitis.
-->

<!-- strict anaerobe citations:
*F. prausnitzii* is an obligate anaerobe  [@doi:10.1038/ismej.2016.176].
*F. prausnitzii* is extremely sensitive to oxygen, though it may be able to withstand oxygen exposure for up to 24 hours depending on the availability of metabolites for extracellular electron transfer [@doi:10.1038/ismej.2016.176].
*G. formicilis* is a strictly anaerobic bacteria that produces both formic acid and butyric acid [@doi:10.1099/00207713-25-2-202].
*Lachnospira eligens_A* is an obligate anaerobe that is unable to tolerate atmospheric oxygen for an hour [@doi:10.1186/s13073-017-0490-5]. 
-->

A substantial fraction of dominating set pieces were more abundant in CD than nonIBD in five metapangenome graphs (**Figure @fig:dda**).
These graphs represented sequences from species *R. gnavus*, *Enterocloster bolteae*, *Enterocloster sp005845215*, *Enterocloster clostridioformis*, and *Enterocloster clostridioformis_A*.
We posit that the increased abundance in some genomic segments amid the decrease in abundance of others represents strain switching that occurs in CD. 

<!--
Only the increase in the distinct accessory genes is detectable amongst the backdrop of general loss of gene abundance due to decrease in other strains.
-->

<!-- section will need to be updated based on graph annot results.  currently relies on R. gnavus prelim results -->
In support of this, when we annotated the differentially abundant pieces using PFAM orthologs, we found that in some cases pieces that were increased in abundance and pieces that were decreased in abundance were annotated with the same ortholog (average XX per graph, SD).
These genes likely represent the portion of the core genome shared by the strain(s) that is more abundant in CD and the strain(s) that is more abundant in nonIBD, but that is encoded by distinct sequences (PULL OUT MARKER GENES, MAKE FIGURE).

Many orthologs were only annotated among the pieces that were increased in abundance in CD. 
Among all five metapangenome graphs, XX orthologs were annotated...
XX pathways were enriched (NOS/ROS/abx res).
Enrichment of specific metabolic pathways is consistent with functional specialization of strains in different environmental niches [@doi:10.15252/msb.20177589].

| metapangenome graph species    | closest strain match                 | maximum containment |
|:--------------------------------:|:--------------------------------------:|:-------------:|
|*Enterocloster clostridioformis*| *Enterocloster clostridioformis* MSK.2.78| 0.71 |
|*Enterocloster bolteae*         | *[Clostridium] bolteae* 90A5             | 0.68 |
|*Ruminococcus_B gnavus*            | *[Ruminococcus] gnavus* RJX1122          | 0.66 |
|*Enterocloster clostridioformis_A* | *[Clostridium] clostridioforme* AGR2157  | 0.61 |
|*Enterocloster sp005845215*        | *Enterocloster clostridioformis* MSK.2.78| 0.50 |
Table: Maximum containment between sequences that were increased in abundance in CD and isolate genomes.
{#tbl:sim}

While dominating set differential abundance analysis identified genomic sequences that were more abundant in CD, the nature of short shotgun metagenomic sequencing reads precludes haplotype phasing or lineage resolution [@doi:10.1038/s41587-021-01130-z], meaning our results likely represent genomic variants from many distinct genomes that would not all naturally occur together in a single isolate genome.
Therefore, to identify isolate genomes that contain the genomic sequences that were more abundant in CD, we searched the GTDB rs202 database with the significantly differentially abundant sequences.
On average, the top matching isolate genome contained 63% of the sequences that were more abundant in CD (Table @tbl:sim).

One aerotolerant clade of *R. gnavus* was previously identified as being enriched in CD [@doi:10.1186/s13073-017-0490-5], and has been shown to produce a polysaccharide that induces the inflammatory cytokine TNF-alpha [@doi:10.1073/pnas.1904099116].
The three isolate genomes we identified as containing the highest amount of sequences that were increased in abundance in CD were among those that have been shown to induce TNF-alpha secretion (RJX1122, RJX1127, RJX1128) [@doi:10.1073/pnas.2007595118]. This suggests our method identified the same strain switch previously discovered to occur in IBD [@doi:10.1186/s13073-017-0490-5; @doi:10.1073/pnas.1904099116; @doi:10.1073/pnas.2007595118].
In further support of this, we found that 17 of the 23 genes in the operon that encodes the proteins responsible for producing the inflammatory polysaccharide were annotated in the dominating set pieces that were more abundant in CD.
These genes were encoded across 66 dominating set pieces, with multiple neighboring genes in the operon annotated in 6 of these dominating set pieces.

For two of the four *Enterocloster* species, the top matching isolate sequence was the same (*Enterocloster clostridioformis* MSK.2.78).
This points to overlap in the genomic sequences we identified as differentially abundant across these metapangenome graphs. 
Indeed, the average Jaccard similarity between the sequences that were increased in CD in the *Enterocloster* graphs was XXX, while the containment was XXX.
Given that a Jaccard similarity of 0.1 is required to recover at least 80% of a genome via assembly graph query, which approximately corresponds to an average nucleotide identity of 93% (CITE: TESSA), and that species boundaries in GTDB are drawn at 95% average nucleotide identity [@doi:10.1093/nar/gkab776], the metapangenome graphs likely store genomic sequences associated with both the query genome species and closely related species.
<!-- are marker genes more likely to overlap? -->
However, the metapangenome graphs presented here, as well as the differentially abundant sequences, contain both common and distinct nucleotide sequences, suggesting that multiple closely related *Enterocloster* species/genomes are associated with CD.
<!-- supp tab Jaccard similarity and containment for all metapangenome graphs and for all the entero increased abund seqs). -->
Taken together, our ability to recover a previously validated sub-species association with IBD (*R. gnavus*) suggests that the three new *Enterocloster* isolates we identified should be further investigated for their potential role in eliciting CD-like symptoms in the gut.

<!--
*C. bolteae* is a member of the normal gut microbiota but is an opportunistic pathogen that exploits compromised intestinal barriers [@doi:10.1186/s12864-016-3152-x].
It is associated with disturbance succession and has increased gene expression during gut dysbiosis [@doi:10.1101/gr.138198.112; @doi:10.1038/s41586-019-1237-9].
-->

## Genomic sequences that are differentially abundant in IBD are not exclusive to IBD

Given that genome sequences belonging to many species were differentially abundant in CD and UC, we next investigated whether there was a disease-specific microbiome in CD or UC -- i.e., whether there are sequences from a species that are only observed in IBD.
Using FracMinHash sketches from the differentially abundant sequences, we identified the differentially abundant sequences in each metagenome and compared their occurrence and distribution across diagnoses. 

In general, we found no evidence for disease-specific sequences among the 54 species we investigated.
Instead, we observed almost all sequences in at least some CD, UC, and nonIBD metagenomes (**Figure @fig:exclusive**, **Figure SX**).
Across all species, an average of 14.9% differentially abundant k-mers were unobserved in either CD, UC, or nonIBD.
These results in part explain heterogeneous study findings in previous IBD gut microbiome investigations.

![
**Most differentially abundant sequences occur in metagenomes of individuals diagnosed with CD, UC and non-IBD.** Upset plot of k-mers that were increased in abundance in CD and their occurrence in CD, UC, and nonIBD metagenomes.
](images/dda_kmers_shared_across_diagnosis_upset_increased.png){#fig:exclusive}

<!-- While we found no evidence of a general disease-specific metapangenome, we tested whether the biosynthetic cluster for the inflammatory polysaccharide produced by *F. gnavus* occurred in nonIBD as it had previously only been identified in CD [@doi:10.1073/pnas.1904099116].
An average of more than 100 reads mapped per gene in the cluster in 10 of 213 nonIBD metagenomes.
While more abundant in CD, this cluster is also identifiable within healthy human gut microbiomes, further supporting the lack of disease-specific pangenomes. -->

<!-- TR: only do this if we orpheum/panmer everything? -->
<!-- While the majority of genes were observed in CD, UC, and nonIBD, IBD metagenomes harbored fewer protein k-mers than nonIBD metagenomes (Table XX).
When we counted the number of protein k-mers observed within each metagenome, XX of 54 metapangenomes for CD and XX of 54 metapangenomes for UC had a significantly lower number of genes observed per metagenome than nonIBD (ANOVA p < 0.05, Tukey's HSD p < 0.05). 
Only *C. bolteae* in CD and *Romboutsia timonensis*, *Anaeromassilibacillus*, and *Actulibacter* in UC had a significantly more genes per sample within each pangenome.
Only *F. gnavus* showed no significant difference in the number of genes per sample across all groups. -->
