<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Taylor E. Reiter" />
  <meta name="author" content="Luiz Irber" />
  <meta name="author" content="Alicia A. Gingrich" />
  <meta name="author" content="Dylan Haynes" />
  <meta name="author" content="N. Tessa Pierce-Ward" />
  <meta name="author" content="Phillip T. Brooks" />
  <meta name="author" content="Yosuke Mizutani" />
  <meta name="author" content="Dominik Moritz" />
  <meta name="author" content="Felix Reidl" />
  <meta name="author" content="Amy D. Willis" />
  <meta name="author" content="Blair D. Sullivan" />
  <meta name="author" content="C. Titus Brown" />
  <meta name="dcterms.date" content="2022-06-30" />
  <meta name="keywords" content="compact de Bruijn Graph, k-mer, inflammatory bowel disease, dominating set differential abundance" />
  <title>Meta-analysis of metagenomes via machine learning and assembly graphs reveals strain switches in Crohn’s disease</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta name="dc.title" content="Meta-analysis of metagenomes via machine learning and assembly graphs reveals strain switches in Crohn&#39;s disease" />
  <meta name="citation_title" content="Meta-analysis of metagenomes via machine learning and assembly graphs reveals strain switches in Crohn&#39;s disease" />
  <meta property="og:title" content="Meta-analysis of metagenomes via machine learning and assembly graphs reveals strain switches in Crohn&#39;s disease" />
  <meta property="twitter:title" content="Meta-analysis of metagenomes via machine learning and assembly graphs reveals strain switches in Crohn&#39;s disease" />
  <meta name="dc.date" content="2022-06-30" />
  <meta name="citation_publication_date" content="2022-06-30" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Taylor E. Reiter" />
  <meta name="citation_author_institution" content="Department of Population Health and Reproduction, University of California, Davis" />
  <meta name="citation_author_orcid" content="0000-0002-7388-421X" />
  <meta name="twitter:creator" content="@ReiterTaylor" />
  <meta name="citation_author" content="Luiz Irber" />
  <meta name="citation_author_institution" content="Graduate Group in Computer Science, UC Davis" />
  <meta name="citation_author_institution" content="Department of Population Health and Reproduction, University of California, Davis" />
  <meta name="citation_author_orcid" content="0000-0003-4371-9659" />
  <meta name="twitter:creator" content="@luizirber" />
  <meta name="citation_author" content="Alicia A. Gingrich" />
  <meta name="citation_author_institution" content="Department of Surgery, University of California, Davis Medical Center" />
  <meta name="citation_author_orcid" content="0000-0002-7239-0154" />
  <meta name="citation_author" content="Dylan Haynes" />
  <meta name="citation_author_institution" content="School of Medicine, Oregon Health &amp; Science University" />
  <meta name="citation_author_orcid" content="0000-0001-8986-8196" />
  <meta name="citation_author" content="N. Tessa Pierce-Ward" />
  <meta name="citation_author_institution" content="Department of Population Health and Reproduction, University of California, Davis" />
  <meta name="citation_author_orcid" content="0000-0002-2942-5331" />
  <meta name="twitter:creator" content="@saltyscientist" />
  <meta name="citation_author" content="Phillip T. Brooks" />
  <meta name="citation_author_institution" content="Department of Population Health and Reproduction, University of California, Davis" />
  <meta name="citation_author_orcid" content="0000-0003-3987-244X" />
  <meta name="twitter:creator" content="@brooksph" />
  <meta name="citation_author" content="Yosuke Mizutani" />
  <meta name="citation_author_institution" content="School of Computing, University of Utah" />
  <meta name="citation_author_orcid" content="0000-0002-9847-4890" />
  <meta name="twitter:creator" content="@mogproject" />
  <meta name="citation_author" content="Dominik Moritz" />
  <meta name="citation_author_institution" content="Human-Computer Interaction Institute, Carnegie Mellon University" />
  <meta name="citation_author_orcid" content="0000-0002-3110-1053" />
  <meta name="twitter:creator" content="@domoritz" />
  <meta name="citation_author" content="Felix Reidl" />
  <meta name="citation_author_institution" content="Department of Computer Science and Information Systems, Birkbeck, University of London" />
  <meta name="citation_author_orcid" content="0000-0002-2354-3003" />
  <meta name="twitter:creator" content="@quantumgravitas" />
  <meta name="citation_author" content="Amy D. Willis" />
  <meta name="citation_author_institution" content="Department of Biostatistics, University of Washington" />
  <meta name="citation_author_orcid" content="0000-0002-2802-4317" />
  <meta name="twitter:creator" content="@AmyDWillis" />
  <meta name="citation_author" content="Blair D. Sullivan" />
  <meta name="citation_author_institution" content="School of Computing, University of Utah" />
  <meta name="citation_author_orcid" content="0000-0001-7720-6208" />
  <meta name="twitter:creator" content="@BlairDSullivan" />
  <meta name="citation_author" content="C. Titus Brown" />
  <meta name="citation_author_institution" content="Department of Population Health and Reproduction, University of California, Davis" />
  <meta name="citation_author_orcid" content="0000-0001-6001-2677" />
  <meta name="twitter:creator" content="@ctitusbrown" />
  <link rel="canonical" href="https://dib-lab.github.io/2021-paper-ibd/" />
  <meta property="og:url" content="https://dib-lab.github.io/2021-paper-ibd/" />
  <meta property="twitter:url" content="https://dib-lab.github.io/2021-paper-ibd/" />
  <meta name="citation_fulltext_html_url" content="https://dib-lab.github.io/2021-paper-ibd/" />
  <meta name="citation_pdf_url" content="https://dib-lab.github.io/2021-paper-ibd/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://dib-lab.github.io/2021-paper-ibd/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://dib-lab.github.io/2021-paper-ibd/v/3f5bcac8cd1a3047bd6129d5864bdf56076392d4/" />
  <meta name="manubot_html_url_versioned" content="https://dib-lab.github.io/2021-paper-ibd/v/3f5bcac8cd1a3047bd6129d5864bdf56076392d4/" />
  <meta name="manubot_pdf_url_versioned" content="https://dib-lab.github.io/2021-paper-ibd/v/3f5bcac8cd1a3047bd6129d5864bdf56076392d4/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Meta-analysis of metagenomes via machine learning and assembly graphs reveals strain switches in Crohn’s disease</h1>
</header>
<p><small><em>
This manuscript
(<a href="https://dib-lab.github.io/2021-paper-ibd/v/3f5bcac8cd1a3047bd6129d5864bdf56076392d4/">permalink</a>)
was automatically generated
from <a href="https://github.com/dib-lab/2021-paper-ibd/tree/3f5bcac8cd1a3047bd6129d5864bdf56076392d4">dib-lab/2021-paper-ibd@3f5bcac</a>
on June 30, 2022.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Taylor E. Reiter</strong><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-7388-421X">0000-0002-7388-421X</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/taylorreiter">taylorreiter</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/ReiterTaylor">ReiterTaylor</a><br>
<small>
Department of Population Health and Reproduction, University of California, Davis
· Funded by Grant GBMF4551 from the Gordon and Betty Moore Foundation to CTB; Grant R03OD030596 from the NIH Common Fund to CTB
</small></p></li>
<li><p><strong>Luiz Irber</strong><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-4371-9659">0000-0003-4371-9659</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/luizirber">luizirber</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/luizirber">luizirber</a><br>
<small>
Graduate Group in Computer Science, UC Davis; Department of Population Health and Reproduction, University of California, Davis
· Funded by Grant GBMF4551 from the Gordon and Betty Moore Foundation to CTB; Grant R01HG007513 from the NIH NHGRI to CTB
</small></p></li>
<li><p><strong>Alicia A. Gingrich</strong><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-7239-0154">0000-0002-7239-0154</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/alicia-gingrich">alicia-gingrich</a><br>
<small>
Department of Surgery, University of California, Davis Medical Center
</small></p></li>
<li><p><strong>Dylan Haynes</strong><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-8986-8196">0000-0001-8986-8196</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/dylan-haynes">dylan-haynes</a><br>
<small>
School of Medicine, Oregon Health &amp; Science University
</small></p></li>
<li><p><strong>N. Tessa Pierce-Ward</strong><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-2942-5331">0000-0002-2942-5331</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/bluegenes">bluegenes</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/saltyscientist">saltyscientist</a><br>
<small>
Department of Population Health and Reproduction, University of California, Davis
· Funded by Grant 1711984 from the NSF; Grant GBMF4551 from the Gordon and Betty Moore Foundation to CTB; Grant 2018911 from the NSF to CTB
</small></p></li>
<li><p><strong>Phillip T. Brooks</strong><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-3987-244X">0000-0003-3987-244X</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/brooksph">brooksph</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/brooksph">brooksph</a><br>
<small>
Department of Population Health and Reproduction, University of California, Davis
</small></p></li>
<li><p><strong>Yosuke Mizutani</strong><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-9847-4890">0000-0002-9847-4890</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/mogproject">mogproject</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/mogproject">mogproject</a><br>
<small>
School of Computing, University of Utah
</small></p></li>
<li><p><strong>Dominik Moritz</strong><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-3110-1053">0000-0002-3110-1053</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/domoritz">domoritz</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/domoritz">domoritz</a><br>
<small>
Human-Computer Interaction Institute, Carnegie Mellon University
</small></p></li>
<li><p><strong>Felix Reidl</strong><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-2354-3003">0000-0002-2354-3003</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/microgravitas">microgravitas</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/quantumgravitas">quantumgravitas</a><br>
<small>
Department of Computer Science and Information Systems, Birkbeck, University of London
</small></p></li>
<li><p><strong>Amy D. Willis</strong><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-2802-4317">0000-0002-2802-4317</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/adw96">adw96</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/AmyDWillis">AmyDWillis</a><br>
<small>
Department of Biostatistics, University of Washington
· Funded by Grant R35GM133420 from the NIH NIGMS
</small></p></li>
<li><p><strong>Blair D. Sullivan</strong><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-7720-6208">0000-0001-7720-6208</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/bdsullivan">bdsullivan</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/BlairDSullivan">BlairDSullivan</a><br>
<small>
School of Computing, University of Utah
· Funded by Grant GBMF4560 from the Gordon and Betty Moore Foundation
</small></p></li>
<li><p><strong>C. Titus Brown</strong><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-6001-2677">0000-0001-6001-2677</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/ctb">ctb</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/ctitusbrown">ctitusbrown</a><br>
<small>
Department of Population Health and Reproduction, University of California, Davis
· Funded by Grant GBMF4551 from the Gordon and Betty Moore Foundation; Grant R01HG007513 from the NIH NHGRI; Grant 2018911 from the NSF; Grant R03OD030596 from the NIH Common Fund
</small></p></li>
</ul>
<h2 class="page_break_before" id="abstract">Abstract</h2>
<p>Microbial strains have closely related genomes but may have different phenotypes in the same environment.
Shotgun metagenomic sequencing can capture the genomes of all strains present in a community but strain-resolved analysis from shotgun sequencing alone remains difficult.
We developed an approach to identify and interrogate strain-level differences in groups of metagenomes.
We use this approach to perform a meta-analysis of stool microbiomes from individuals with and without inflammatory bowel disease (IBD; Crohn’s disease, ulcerative colitis; n = 605), a disease for which there are not specific microbial biomarkers but some evidence that microbial strain variation may stratify by disease state.
We first developed a machine learning classifier based on compressed representations of complete metagenomes (FracMinHash sketches) and identified genomes that correlate with IBD subtype.
To rescue variation that may not have been present in the genomes, we then used assembly graph genome queries to recover strain variation for correlated genomes.
Lastly, we developed a novel differential abundance framework that works directly on the assembly graph to uncover all sequence variants correlated with IBD.
Using this approach, refer to this approach as dominating set differential abundance analysis and have implemented it in the <a href="https://github.com/spacegraphcats/spacegraphcats">spacegraphcats software package</a>.
We identified five bacterial strains that are associated with Crohn’s disease.
Our method captures variation within the entire sequencing data set, allowing for discovery of previously hidden disease associations.</p>
<h1 id="introduction">Introduction</h1>
<p>Sub-species groupings of microorganisms have functional differences that govern important genome-environment interactions across diverse ecosystems.
For example, ecotypes of <em>Escherichia coli</em> have different gene complements that allow each group to thrive in diverse environments like the gut, soil, and freshwater <span class="citation" data-cites="KY2dovBD">[<a href="#ref-KY2dovBD" role="doc-biblioref">1</a>]</span>.
Metagenomic sequencing data from communities of microorganisms contain information about specific strains present in a sample, but strain-resolved insights are lacking due to incomplete references or inability of current tools to retrieve such information <span class="citation" data-cites="nSWbl26Q">[<a href="#ref-nSWbl26Q" role="doc-biblioref">2</a>]</span>.
Here we use <em>strain</em> to refer to within-species variation that generates taxonomic grouping below the species level.</p>
<p>Inflammatory bowel disease (IBD) is a group of disorders that are characterized by chronic inflammation of the intestines which may in part be the result of host-mediated inflammatory responses to microorganisms <span class="citation" data-cites="9nrllDuS">[<a href="#ref-9nrllDuS" role="doc-biblioref">3</a>]</span>.
IBD classically manifests in three subtypes depending on clinical presentation, including Crohn’s disease (CD), which presents as discontinuous patches of inflammation throughout the gastrointestinal tract, ulcerative colitis (UC), which presents as continuous inflammation isolated to the colon and rectum, and undetermined subtype, which cannot be clinically or biologically distinguished as CD or UC.
Diagnosis can be clinically difficult, with ramifications associated with incorrect treatment resulting in unnecessary patient morbidity.
Detection of microbial signatures associated with IBD subtype may lead to improved diagnostic criteria and therapeutics that extend periods of remission.
However, such signatures have thus far remained elusive <span class="citation" data-cites="jxRtm2IY">[<a href="#ref-jxRtm2IY" role="doc-biblioref">4</a>]</span>.</p>
<p>The microbiome of CD and UC is heterogeneous, and studies that characterize the microbiome often produce conflicting results <span class="citation" data-cites="jxRtm2IY">[<a href="#ref-jxRtm2IY" role="doc-biblioref">4</a>]</span>.
This is likely in part driven by large inter- and intra-individual variation <span class="citation" data-cites="19Z0U0zL4">[<a href="#ref-19Z0U0zL4" role="doc-biblioref">5</a>]</span>, but is also attributable to non-standardized laboratory, sequencing, and analysis techniques used to profile the gut microbiome <span class="citation" data-cites="jxRtm2IY">[<a href="#ref-jxRtm2IY" role="doc-biblioref">4</a>]</span>.
Dysbiosis is frequently observed in IBD, particularly in CD <span class="citation" data-cites="1DalQm9EV VDWIC49v CwmUZE5B 1DAvhqb44 17MF0XTS9">[<a href="#ref-1DalQm9EV" role="doc-biblioref">6</a>,<a href="#ref-VDWIC49v" role="doc-biblioref">7</a>,<a href="#ref-CwmUZE5B" role="doc-biblioref">8</a>,<a href="#ref-1DAvhqb44" role="doc-biblioref">9</a>,<a href="#ref-17MF0XTS9" role="doc-biblioref">10</a>]</span>, however dysbiosis alone is not a signature of IBD <span class="citation" data-cites="19Z0U0zL4">[<a href="#ref-19Z0U0zL4" role="doc-biblioref">5</a>]</span>.
<em>Dysbiosis</em> is defined as a decrease in gut microbial diversity that results in an imbalance between protective and harmful microorganisms, leading to intestinal inflammation <span class="citation" data-cites="ilcqfnau">[<a href="#ref-ilcqfnau" role="doc-biblioref">11</a>]</span>.</p>
<p>Strain-level differences may account for some heterogeneity in IBD gut microbiome profiles.
A recent investigation of time-series gut microbiome metagenomes found that one clade of <em>Ruminococcus gnavus</em> is enriched in CD <span class="citation" data-cites="NznM4tWF">[<a href="#ref-NznM4tWF" role="doc-biblioref">12</a>]</span>.
Further, this clade produces an inflammatory polysaccharide <span class="citation" data-cites="17cWqPdre u0E0V8yd">[<a href="#ref-17cWqPdre" role="doc-biblioref">13</a>,<a href="#ref-u0E0V8yd" role="doc-biblioref">14</a>]</span>.
While this clade is enriched in CD, its enrichment was previously masked from computational discovery by concomitant decreases in other <em>Ruminococcus</em> species in IBD <span class="citation" data-cites="NznM4tWF">[<a href="#ref-NznM4tWF" role="doc-biblioref">12</a>]</span>, highlighting the need for strain-resolved analysis of metagenomic sequencing in the exploration of IBD gut microbiomes.</p>
<p>Given these features of the IBD gut microbiome, strain-resolved analysis may yield insights into the dynamics of these communities.
The two biggest obstacles to strain-level analysis of short read data are the lack of strain representation in databases together with the challenge of haplotype-level resolution in assembly and binning.
While long reads have made strides toward resolving the latter issue <span class="citation" data-cites="10nUzxyHX">[<a href="#ref-10nUzxyHX" role="doc-biblioref">15</a>]</span>, in habitats like the gut where communities are dominated by single strains of microbes <span class="citation" data-cites="yvK2iCjv">[<a href="#ref-yvK2iCjv" role="doc-biblioref">16</a>]</span> the largest barrier to strain-level analysis is the exclusion of data that does not match to reference databases.
New data analysis techniques are needed to make full use of strain level data.</p>
<p><em>K-mers</em>, words of length <em>k</em> in nucleotide sequences, have previously been used for annotation-free characterization of sequencing data <span class="citation" data-cites="1FQH2cgLx G7YuT52J U7VsiwaU">[<a href="#ref-1FQH2cgLx" role="doc-biblioref">17</a>,<a href="#ref-G7YuT52J" role="doc-biblioref">18</a>,<a href="#ref-U7VsiwaU" role="doc-biblioref">19</a>]</span>.
K-mers are suitable for strain-resolved metagenome analysis because their absence in reference databases does not preclude their analysis.
Moreover, k-mer analysis does not rely on marker genes which are largely conserved at the strain level, and k-mers are suitable for species- and strain-level classification <span class="citation" data-cites="mAjlUXL2 IadOz4ol">[<a href="#ref-mAjlUXL2" role="doc-biblioref">20</a>,<a href="#ref-IadOz4ol" role="doc-biblioref">21</a>]</span>.
Investigating all k-mers in metagenomes is more computationally intensive than reference-based approaches <span class="citation" data-cites="fFEMc6sb">[<a href="#ref-fFEMc6sb" role="doc-biblioref">22</a>]</span>, but data-reduction techniques like FracMinHash sketching make k-mer-based analysis scalable to large-scale sequence comparisons <span class="citation" data-cites="ART4gJKs cHw8gtst">[<a href="#ref-ART4gJKs" role="doc-biblioref">23</a>,<a href="#ref-cHw8gtst" role="doc-biblioref">24</a>]</span>.
FracMinHash sketching sacrifices the fine-scaled resolution of reference-based techniques but is representative of the full sequencing sample and can make use of all available genomes <span class="citation" data-cites="IadOz4ol">[<a href="#ref-IadOz4ol" role="doc-biblioref">21</a>]</span>, thus including strain-variable accessory elements that may be associated with diseases</p>
<p>Like sketches, assembly graphs also represent k-mers in a metagenome, but assembly graphs retain important sequencing context and can aggregate known functional and taxonomic annotations, recovering critical information lost through sketching approaches <span class="citation" data-cites="VsLophuT Qc2FKvHp">[<a href="#ref-VsLophuT" role="doc-biblioref">25</a>,<a href="#ref-Qc2FKvHp" role="doc-biblioref">26</a>]</span>.
While assembly graphs have been leveraged in metagenome analyses <span class="citation" data-cites="jroubkzG">[<a href="#ref-g2bc2Oy5" role="doc-biblioref">28</a>]</span>, their large size precludes analysis at scale.
The <em>spacegraphcats</em> tool is designed to tackle this issue, implementing algorithms that efficiently reduce the size of an assembly graph and enabling rapid querying and sequence retrieval <span class="citation" data-cites="VsLophuT">[<a href="#ref-VsLophuT" role="doc-biblioref">25</a>]</span>.
These algorithms center around dominating sets, which partition the graph into <em>pieces</em> by assigning every node to a graph-localized neighborhood <span class="citation" data-cites="VsLophuT">[<a href="#ref-VsLophuT" role="doc-biblioref">25</a>]</span>.
This simplified graph enables efficient queries: querying with a sequence that overlaps any k-mer in a compact de Bruijn graph (cDBG) node returns all k-mers (or all reads containing those k-mers) from the graph neighborhood.
Genome queries often recover sequences not in reference databases or <em>de novo</em> assemblies, which disproportionately include sequences from both low coverage regions and highly variable portions of the graph (e.g. sequencing reads that neither assemble nor bin) <span class="citation" data-cites="VsLophuT">[<a href="#ref-VsLophuT" role="doc-biblioref">25</a>]</span>.
When a query has a containment index between 10<sup>-2</sup> and 10<sup>-3</sup> with the assembly graph, 20-40% of a target genome sequence is recovered from a metagenome query, and for containment indices above 10<sup>-1</sup> this increases to &gt;80% <span class="citation" data-cites="VsLophuT">[<a href="#ref-VsLophuT" role="doc-biblioref">25</a>]</span>.
Containment index is calculated by comparing the relative size of the intersection to the union between k-mers in a query and k-mers in a metagenome <span class="citation" data-cites="s8orJ7KE">[<a href="#ref-s8orJ7KE" role="doc-biblioref">29</a>]</span>.</p>
<p>Here, we develop k-mer- and assembly graph-based techniques to perform a meta-analysis of stool microbiome metagenomes from individuals with (CD, UC) and without (nonIBD) IBD <span class="citation" data-cites="19Z0U0zL4 CwmUZE5B NznM4tWF Xi4Nccqm dSDu60g3 17MF0XTS9">[<a href="#ref-19Z0U0zL4" role="doc-biblioref">5</a>,<a href="#ref-CwmUZE5B" role="doc-biblioref">8</a>,<a href="#ref-17MF0XTS9" role="doc-biblioref">10</a>,<a href="#ref-NznM4tWF" role="doc-biblioref">12</a>,<a href="#ref-Xi4Nccqm" role="doc-biblioref">30</a>,<a href="#ref-dSDu60g3" role="doc-biblioref">31</a>]</span>.
Using these approaches, we detect a consistent signature of IBD subtype in fecal microbiome metagenomes.
We identify a small set of k-mers that are predictive of UC and CD, and find that these k-mers originate from a core set of microbial genomes.
We find that a stochastic loss of diversity in this core set of microbial genomes is a hallmark of CD, and to a lesser extent, UC, as has been previously demonstrated <span class="citation" data-cites="jxRtm2IY">[<a href="#ref-jxRtm2IY" role="doc-biblioref">4</a>]</span>.
While reduced diversity is responsible for the majority of disease signatures, we also find signatures of strains present in the disease state.
Sequences associated with these strains occurred more frequently in IBD metagenomes but are present in low abundance in nonIBD metagenomes as well.
Our approach provides a solution for strain-level analysis of short read metagenomic data sets, and our findings provide future avenues for research into IBD therapeutics.</p>
<h1 id="results">Results</h1>
<p>We developed a computational approach to resolve sub-species level differences between groups of short read shotgun metagenomes (<strong>Figure <a href="#fig:overview">1</a></strong>).
While our pipeline relies on many published algorithms, we developed two new approaches that, when combined with existing tools, generated insights into microbial sequences associated with IBD subtype.
After consistent pre-processing, we used FracMinHash sketching to produce subsampled k-mer abundance profiles of metagenomes that reflected the sequence diversity in a sample <span class="citation" data-cites="ART4gJKs IadOz4ol">[<a href="#ref-IadOz4ol" role="doc-biblioref">21</a>,<a href="#ref-ART4gJKs" role="doc-biblioref">23</a>]</span>, and used these profiles to perform metagenome-wide k-mer association with IBD subtype.
We refer to FracMinHash sketches as <em>sketches</em> or k-mer abundance profiles, and for simplicity, continue referring to the sub-sampled k-mers in a sketch as <em>k-mers</em>.
Retaining only k-mers associated with IBD, we used a minimum set cover approach to identify the genomes that best encompassed these k-mers <span class="citation" data-cites="IadOz4ol">[<a href="#ref-IadOz4ol" role="doc-biblioref">21</a>]</span>.</p>
<p>Next, we developed an approach to perform differential abundance analysis directly on assembly graphs in order to recover all sequences that were differentially abundant in each IBD subtype when compared to nonIBD.
Using the genomes identified by our k-mer association analysis, we first performed assembly graph genome queries to recover all sequences associated with a given species within a metagenome.
For each genome query, we combined these sequences into a single assembly graph, which we refer to as a <em>metapangenome graph</em>.
We estimated the abundance of each piece in this graph within each metagenome, and used these abundances to perform differential abundance analysis.</p>
<div id="fig:overview" class="fignos">
<figure>
<img src="images/ibd_overview_steps.png" alt="Figure 1: Overview of the metagenome analysis technique. Steps that are outlined in grey were developed in this paper. Step 1: Using quality controlled sequencing reads from many metagenomes, we decomposed reads into k-mers and subsample these k-mers using FracMinHash, thereby selecting k-mers that evenly represent the sequence diversity within a sample. We then identified interesting k-mers using random forests, and associated these k-mers with genomes in reference databases. Step 2: For each metagenome, we constructed a compact de Bruijn assembly graph (cDBG) that contains all k-mers from a metagenome. We used dominating sets to carve the graph into pieces. We queried this graph with genomes associated with interesting k-mers identified in Step 1, recovering sequence diversity nearby in the assembly graph. We refer to these sequences as genome query neighborhoods. Step 2 is the workflow published in [25]. Step 3: We combined genome query neighborhoods for a single genome from all metagenomes. We constructed a cDBG from these sequences, and used a dominating set with a large radius to carve the graph into large pieces. Here, we diagram construction of r=2 dominating set pieces, but in practice we used r=10. We estimated the abundance of k-mers in each metagenome for each dominating set piece, and used these abundances to perform differential abundance analysis." /><figcaption aria-hidden="true"><span>Figure 1:</span> <strong>Overview of the metagenome analysis technique.</strong> Steps that are outlined in grey were developed in this paper. <strong>Step 1:</strong> Using quality controlled sequencing reads from many metagenomes, we decomposed reads into k-mers and subsample these k-mers using FracMinHash, thereby selecting k-mers that evenly represent the sequence diversity within a sample. We then identified interesting k-mers using random forests, and associated these k-mers with genomes in reference databases. <strong>Step 2:</strong> For each metagenome, we constructed a compact de Bruijn assembly graph (cDBG) that contains all k-mers from a metagenome. We used dominating sets to carve the graph into pieces. We queried this graph with genomes associated with interesting k-mers identified in Step 1, recovering sequence diversity nearby in the assembly graph. We refer to these sequences as genome query neighborhoods. Step 2 is the workflow published in <span class="citation" data-cites="VsLophuT">[<a href="#ref-VsLophuT" role="doc-biblioref">25</a>]</span>. <strong>Step 3:</strong> We combined genome query neighborhoods for a single genome from all metagenomes. We constructed a cDBG from these sequences, and used a dominating set with a large radius to carve the graph into large pieces. Here, we diagram construction of <em>r</em>=2 dominating set pieces, but in practice we used <em>r</em>=10. We estimated the abundance of k-mers in each metagenome for each dominating set piece, and used these abundances to perform differential abundance analysis.</figcaption>
</figure>
</div>
<p>We applied this approach to the analysis of IBD gut microbiomes via meta-analysis.
Meta-analyses have recently shown success in improving the power to detect microbial signatures of colorectal cancer <span class="citation" data-cites="n90P8bOS 19itPpFkZ 17XxJx4ep">[<a href="#ref-n90P8bOS" role="doc-biblioref">32</a>,<a href="#ref-19itPpFkZ" role="doc-biblioref">33</a>,<a href="#ref-17XxJx4ep" role="doc-biblioref">34</a>]</span>.
To this end, we identified studies that performed metagenomic sequencing of individuals with CD, UC, or nonIBD and combined these to perform a meta-analysis (<strong>Table <a href="#tbl:cohorts">1</a></strong>, <strong>Table S1</strong>).
All studies profiled fecal gut microbiomes via Illumina shotgun metagenome sequencing.
Individuals were from five distinct countries and seven cohorts (<strong>Table <a href="#tbl:cohorts">1</a></strong>).
In many studies, samples were taken in time series to profile disease progression or individual response to treatment.
In these cases we included only the first sample in the time series so organized interventions would not skew our results.
In addition, many of the nonIBD samples, particularly those from the iHMP, profiled sick individuals that were not diagnosed with IBD, meaning some of these samples are not healthy controls.</p>
<div id="tbl:cohorts" class="tablenos">
<table id="tbl:cohorts">
<caption><span>Table 1:</span> Six IBD shotgun metagenome sequencing cohorts used in this meta-cohort analysis.
</caption>
<colgroup>
<col style="width: 15%" />
<col style="width: 17%" />
<col style="width: 16%" />
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 12%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Cohort</th>
<th style="text-align: left;">Name</th>
<th style="text-align: left;">Country</th>
<th style="text-align: right;">Total</th>
<th style="text-align: right;">CD</th>
<th style="text-align: right;">UC</th>
<th style="text-align: right;">nonIBD</th>
<th style="text-align: left;">Reference</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">iHMP</td>
<td style="text-align: left;">IBDMDB</td>
<td style="text-align: left;">USA</td>
<td style="text-align: right;">106</td>
<td style="text-align: right;">50</td>
<td style="text-align: right;">30</td>
<td style="text-align: right;">26</td>
<td style="text-align: left;"><span class="citation" data-cites="19Z0U0zL4">[<a href="#ref-19Z0U0zL4" role="doc-biblioref">5</a>]</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">PRJEB2054</td>
<td style="text-align: left;">MetaHIT</td>
<td style="text-align: left;">Denmark, Spain</td>
<td style="text-align: right;">124</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">99</td>
<td style="text-align: left;"><span class="citation" data-cites="17MF0XTS9">[<a href="#ref-17MF0XTS9" role="doc-biblioref">10</a>]</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">SRP057027</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Canada, USA</td>
<td style="text-align: right;">112</td>
<td style="text-align: right;">87</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">25</td>
<td style="text-align: left;"><span class="citation" data-cites="CwmUZE5B">[<a href="#ref-CwmUZE5B" role="doc-biblioref">8</a>]</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">PRJNA385949</td>
<td style="text-align: left;">PRISM, STiNKi</td>
<td style="text-align: left;">USA</td>
<td style="text-align: right;">17</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">3</td>
<td style="text-align: left;"><span class="citation" data-cites="NznM4tWF">[<a href="#ref-NznM4tWF" role="doc-biblioref">12</a>]</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">PRJNA400072</td>
<td style="text-align: left;">PRISM, LLDeep, and NLIBD</td>
<td style="text-align: left;">USA, Netherlands</td>
<td style="text-align: right;">218</td>
<td style="text-align: right;">87</td>
<td style="text-align: right;">76</td>
<td style="text-align: right;">55</td>
<td style="text-align: left;"><span class="citation" data-cites="Xi4Nccqm">[<a href="#ref-Xi4Nccqm" role="doc-biblioref">30</a>]</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">PRJNA237362</td>
<td style="text-align: left;">RISK</td>
<td style="text-align: left;">North America</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">5</td>
<td style="text-align: left;"><span class="citation" data-cites="dSDu60g3">[<a href="#ref-dSDu60g3" role="doc-biblioref">31</a>]</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">Total</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">605</td>
<td style="text-align: right;">260</td>
<td style="text-align: right;">132</td>
<td style="text-align: right;">213</td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
</div>
<h2 id="k-mers-are-weakly-predictive-of-ibd-subtype">K-mers are weakly predictive of IBD subtype</h2>
<p>We first sought an approach to compare many metagenomes without relying on reference databases, <em>de novo</em> assembly, or annotations.
We reasoned that FracMinHash sketches randomly subsample k-mers to allow comparisons, which may provide an unbiased approach to quickly compare across many metagenomes.
In total, we profiled 7,376,151 subsampled k-mers across all samples in all cohorts, representing approximately 14 billion distinct k-mers in the original samples.</p>
<p>We detected variation correlated with IBD diagnosis in k-mer profiles of gut metagenomes from different cohorts.
We calculated a pairwise distance matrix using angular distance between k-mer abundance profiles to assess sample diversity.
We performed principle coordinate analysis and PERMANOVA with this distance matrix (<strong>Figure <a href="#fig:kmers">2</a> A, B</strong>), using the variables study accession, diagnosis, library size, and number of k-mers observed in a sample (<strong>Figure <a href="#fig:kmers">2</a> B</strong>).
Study accounted for highest variation, emphasizing that technical artifacts or cohort diversity can introduce strong signals that may influence heterogeneity in results across IBD microbiome studies but that can be mitigated through meta-analysis <span class="citation" data-cites="n90P8bOS">[<a href="#ref-n90P8bOS" role="doc-biblioref">32</a>]</span>.
The number of k-mers observed in a sample accounted for the second highest variation, possibly reflecting reduced diversity in stool metagenomes of CD and UC patients (reviewed in <span class="citation" data-cites="aKDEtlHU">[<a href="#ref-aKDEtlHU" role="doc-biblioref">35</a>]</span>).
Diagnosis accounted for a substantial amount of variation as well, indicating that there is a small but detectable signal of IBD subtype in stool metagenomes.</p>
<!--
|Variable          |Jaccard distance | Angular distance|
|:-----------------|:---------------:|:---------------:|
| Number of k-mers | 9.9%           | 6.2% |
| Study accession  | 6.6%           | 13.5% |
| Diagnosis        | 6.2%           | 3.3% |
| Library size     | 0.009%        | 0.01% |
Table: Results from PERMANOVA performed on Jaccard and Angular distance matrices. Number of k-mers refers to the number of k-mers in a signature, while library size refers to the number of raw reads per sample. All test were significant at p < .001.
{#tbl:permanova}
-->
<p>To evaluate whether the variation captured by diagnosis is predictive of IBD subtype, we built random forest classifiers to predict CD, UC, or nonIBD subtype.
To assess whether disease signatures generalize across study populations, we used a leave-one-study-out cross-validation approach where we built and optimized a classifier using five cohorts and validated on the sixth.
We built each model six times to hone in on cross-study and cross-model signal.
Given the high-dimensional structure of this data set (e.g. many more k-mers than metagenomes), we first used the vita method of variable selection to narrow the set of predictive k-mers in the training set <span class="citation" data-cites="gePakmgr y3cwAdm4">[<a href="#ref-gePakmgr" role="doc-biblioref">36</a>,<a href="#ref-y3cwAdm4" role="doc-biblioref">37</a>]</span>.
Variable selection reduced the number of k-mers used in each model by two orders of magnitude, from 7,376,151 to 28,684-41,701 (mean = 35,673.1, sd = 4090.3) (<strong>Figure <a href="#fig:kmers">2</a> C</strong>).</p>
<p>Using this reduced set of k-mers, we optimized each random forests classifier on the training set, producing 36 optimized models.
We validated each model on the left-out study.
The accuracy on the validation studies ranged from 49%-77% (<strong>Figure <a href="#fig:kmers">2</a> D</strong>), outperforming a previously published model built on metagenomic data alone <span class="citation" data-cites="Xi4Nccqm">[<a href="#ref-Xi4Nccqm" role="doc-biblioref">30</a>]</span>.</p>
<div id="fig:kmers" class="fignos">
<figure>
<img src="images/kmers.png" alt="Figure 2: Long nucleotide k-mers retain information about IBD subtype classification. A. Principal coordinate analysis of distance matrices obtained from comparing FracMinHash sketches with abundances and B. PERMANOVA results that explain the variance in the principal coordinate analysis. Number of k-mers refers to the number of k-mers in a sketch, while library size refers to the number of raw reads per sample. All tests were significant at p &lt; .001. C. Box plots indicating the number of k-mers used to build each random forests model. Variable selection using the vita method reduced the number of k-mers used to build each model. D. Heatmap indicating accuracy of each model on the left-out validation study. Model performance varied by validation study, but models predicted IBD subtype better than chance (1/3)." /><figcaption aria-hidden="true"><span>Figure 2:</span> <strong>Long nucleotide k-mers retain information about IBD subtype classification.</strong> <strong>A.</strong> Principal coordinate analysis of distance matrices obtained from comparing FracMinHash sketches with abundances and <strong>B.</strong> PERMANOVA results that explain the variance in the principal coordinate analysis. Number of k-mers refers to the number of k-mers in a sketch, while library size refers to the number of raw reads per sample. All tests were significant at p &lt; .001. <strong>C.</strong> Box plots indicating the number of k-mers used to build each random forests model. Variable selection using the vita method reduced the number of k-mers used to build each model. <strong>D.</strong> Heatmap indicating accuracy of each model on the left-out validation study. Model performance varied by validation study, but models predicted IBD subtype better than chance (1/3).</figcaption>
</figure>
</div>
<p>To understand which genomes were responsible for disease signatures detected by our models, we anchored k-mers in the models against genomes in reference databases using sourmash gather <span class="citation" data-cites="IadOz4ol">[<a href="#ref-IadOz4ol" role="doc-biblioref">21</a>]</span>.
Sourmash gather determines the minimum set of genomes in a database necessary to cover all of the k-mers in a query <span class="citation" data-cites="IadOz4ol">[<a href="#ref-IadOz4ol" role="doc-biblioref">21</a>]</span>.
We used the GTDB rs202 representatives database, which contains bacterial and archaeal genomes, and the GenBank viral, fungal, and protozoa databases.
We found that a substantial fraction of genomes were shared between models, indicating there is a consistent biological signal captured among classifiers: of 3,889 total genomes detected across all classifiers, 360 genomes were shared between all classifiers (<strong>Figure <a href="#fig:metacoder">3</a></strong>, <strong>Figure <a href="#fig:tree">S1</a></strong>, <strong>Table S2</strong>).
The presence of shared k-mers between classifiers indicates that there is a consistent biological signal in metagenomes for IBD subtype between cohorts.</p>
<p>K-mers that anchored to these shared genomes represented 65% of all k-mers used to build the optimized classifiers, but accounted for an outsize proportion of variable importance in the optimized classifiers.
After normalizing variable importance across classifiers, 74% of the total variable importance was held by these k-mers.
These k-mers contribute a large fraction of predictive power for classification of IBD subtype, and the genomes in which they are found represent a microbial core that contains predictive power in IBD subtype classification.</p>
<p>Given that 360 genomes anchored the majority of k-mers and variable importance across all models, we were curious whether a smaller number of genomes could still retain the majority of variable importance.
Limiting genomes to those that could hold at least 1% of the normalized variable importance, we found that 54 genomes accounted for 50% of the variable importance (<strong>Figure <a href="#fig:metacoder">3</a></strong>, <strong>Figure <a href="#fig:tree">S1</a></strong>, <strong>Figure <a href="#fig:varimp">S2</a></strong>, <strong>Table S2</strong>).
We assumed these genomes represent the strongest candidates for discriminating IBD subtype and focused on them for the remainder of our analyses.</p>
<div id="fig:metacoder" class="fignos">
<figure>
<img src="images/metacoder_by_varimp.png" alt="Figure 3: Tree of bacterial species that were predictive of IBD subtype in all models. Nodes are summarized to the genus level. All taxa up to the class level are labelled. Taxa that could account for at least 1% of the normalized variable importance across random forests models are colored and labelled. Node size and node color reflect potential normalized variable importance attributable to each taxonomic lineage with larger node sizes and darker color representing larger variable importance; while normalized variable importance across models sums to one, some sequences are shared across genomes making the total potential variable importance across all genomes larger than one." /><figcaption aria-hidden="true"><span>Figure 3:</span> <strong>Tree of bacterial species that were predictive of IBD subtype in all models.</strong>
Nodes are summarized to the genus level.
All taxa up to the class level are labelled.
Taxa that could account for at least 1% of the normalized variable importance across random forests models are colored and labelled.
Node size and node color reflect potential normalized variable importance attributable to each taxonomic lineage with larger node sizes and darker color representing larger variable importance; while normalized variable importance across models sums to one, some sequences are shared across genomes making the total potential variable importance across all genomes larger than one.</figcaption>
</figure>
</div>
<h2 id="genome-queries-into-metagenome-assembly-graphs-recover-neighborhoods-of-sequence-variation-and-establish-species-umbrellas">Genome queries into metagenome assembly graphs recover neighborhoods of sequence variation and establish species umbrellas</h2>
<p>While we were able to identify the majority of k-mers that were important for predicting IBD subtype, 26% of k-mers remained unannotated.
We hypothesized that these k-mers represented strain variable sequences not in reference databases but belonging to species represented by annotated k-mers.
To investigate this hypothesis, we performed genome queries on assembly graphs of each metagenome using the 54 candidate genomes that discriminated IBD subtype (<strong>Figure <a href="#fig:overview">1</a></strong>).
Assembly graph genome queries recover sequences in a metagenome that match the query, as well as those that are nearby in the assembly graph, making queries akin to but more general than read mapping against reference genomes (<strong>Figure <a href="#fig:overview">1</a></strong>) <span class="citation" data-cites="VsLophuT">[<a href="#ref-VsLophuT" role="doc-biblioref">25</a>]</span>.
The resulting genome query neighborhood represents a species-level umbrella that contains sequence variation from the metagenome associated with a query.</p>
<p>After performing genome queries, we re-anchored k-mers against the resulting query neighborhoods as well as the databases used previously.
We observed that the fraction of unassigned k-mers decreased from 26% to 8% (<strong>Figure <a href="#fig:alluvial">4</a></strong>), supporting our hypothesis that many of these k-mers are sequence variants belonging to species identified in k-mers important for predicting IBD subtype.
We further observed that many other k-mers previously anchored by other genomes were reassigned to the genome query neighborhoods (<strong>Figure <a href="#fig:alluvial">4</a></strong>).
This suggests that the genome queries create species umbrellas that represent sequence variation for the query genome itself as well as other closely related genomes that occur within a metagenome.</p>
<div id="fig:alluvial" class="fignos">
<figure>
<img src="images/alluvial_percent_of_hashes.png" alt="Figure 4: Alluvial plot depicting the set of genomes that anchored k-mers that were important for predicting IBD subtype. The blocks on the left represent the breakdown of k-mer assignations from greedy exact matching against databases alone, while the blocks on the right represent k-mer assignations after metagenome assembly graph queries." /><figcaption aria-hidden="true"><span>Figure 4:</span> <strong>Alluvial plot depicting the set of genomes that anchored k-mers that were important for predicting IBD subtype.</strong>
The blocks on the left represent the breakdown of k-mer assignations from greedy exact matching against databases alone, while the blocks on the right represent k-mer assignations after metagenome assembly graph queries.</figcaption>
</figure>
</div>
<!--
I think part of why this might occur is because a lot of the sequences that were important for predicting IBD subtype, some of which were not in reference databases, encoded marker genes, including 16s and 23s. But the reason I know this is from multifasta queries. I would really rather not get in to the whole 16s/23s marker gene stuff, as it now seems tangential to the dominating set differential abundance (e.g., I'm trying to minimize the random forests component, bc while it's cool from a discovery perspective, all I use it for now it to pick out candidate genomes to analyze more closely). Also, I don't want to go into all of this stuff bc a good 1/2 of the results were generated from the old catlases and I have no intention of regenerating all of those results. So. It feels good knowing that some of the stuff gets scooped in bc 16s/23s should organize stuff close together in graph land, but i think I can safely hide that insight from the reads and not feel too silly about it.
-->
<h2 id="ibd-gut-microbiomes-have-decreased-diversity-in-strict-anaerobes-that-is-punctuated-by-strain-switches-for-some-facultative-anaerobes">IBD gut microbiomes have decreased diversity in strict anaerobes that is punctuated by strain switches for some facultative anaerobes</h2>
<p>After recovering all sequences in metagenomes in the neighborhoods of the species that discriminate IBD subtypes, we next sought to determine the specific genome segments that were differentially abundant in IBD.
Differential abundance analysis is a common step in metagenome analysis, however it is typically applied to gene counts <span class="citation" data-cites="nKH9jczu 1FH3KNlym">[<a href="#ref-nKH9jczu" role="doc-biblioref">38</a>,<a href="#ref-1FH3KNlym" role="doc-biblioref">39</a>]</span>, which requires assembly or mapping prior to abundance estimation.
<!--TR: Also commonly applied to species counts. Should this be added? -->
To avoid assembly or mapping and the accompanying loss of reads <span class="citation" data-cites="1CjZRQMrc">[<a href="#ref-1CjZRQMrc" role="doc-biblioref">40</a>]</span>, we developed an abundance estimation approach that works directly on the assembly graphs, enabling differential abundance analysis from the assembly graph.
Our abundance estimation approach was based on <em>r</em>-dominating sets, an algorithm introduced in <span class="citation" data-cites="VsLophuT">[<a href="#ref-VsLophuT" role="doc-biblioref">25</a>]</span> that efficiently computes the dominating nodes in a cDBG so that every node is no more that distance <em>r</em> from a dominator.
The dominating set is used to carve the graph into pieces, each of which contains one dominating node.
Here, we first build a species-level assembly graph that contains neighborhood sequences for a given genome across all metagenomes, which we call a <em>metapangenome graph</em>.
We then partition the graph into pieces using a large radius (<em>r</em> = 10).
The large radius carves the graph into pieces that average 103 k-mers in size.
We next estimated the abundance of each piece within each metagenome using average k-mer abundance.
We also annotated the graph pieces using using k-mer overlap between genes of known function and graph pieces.
Using this information, we performed dominating set differential abundance analysis using corncob <span class="citation" data-cites="UIA9vjDO">[<a href="#ref-UIA9vjDO" role="doc-biblioref">41</a>]</span>, a statistical package that tests for differential relative abundance in the presence of variable sequencing depth and excessive zeroes for unobserved observations, conditions which occur in abundances from dominating sets.
We tested differential abundance at the 5% significance level after correcting for multiple testing (see methods).</p>
<p>We applied this method for each of our genome queries, building 54 metapangenome graphs and performing dominating set differential abundance analysis on each.
We tested for differential abundance in pieces that occurred in at least 100 metagenomes, since we sought differences that characterized the majority of our samples within a group.
Note that corncob fits a model for each dominating set piece and therefore does not require abundance information for all pieces <span class="citation" data-cites="UIA9vjDO">[<a href="#ref-UIA9vjDO" role="doc-biblioref">41</a>]</span>.
On average, this condition was met in 6.4% of dominating set pieces.
Focusing on pieces that occurred in many metagenomes increased the average piece size to 1088 k-mers, which is similar to the average bacterial gene length of approximately 1000 base pairs <span class="citation" data-cites="17f69WRJu">[<a href="#ref-17f69WRJu" role="doc-biblioref">42</a>]</span> and should enable biologically meaningful comparisons across groups.</p>
<div id="fig:dda" class="fignos">
<figure>
<img src="images/gtdb_and_corncob_res.png" alt="Figure 5: Dominating set differential abundance analysis revealed genome segments that were significantly different in CD and UC compared to nonIBD. Results are organized by GTDB taxonomy, with a tree representing the 54 species and colored by family on the far left. The percent of dominating set pieces tested is labelled in grey, and the percent of significantly differentially abundant pieces are colored by increased (blue) or decreased (brown) abundance." /><figcaption aria-hidden="true"><span>Figure 5:</span> <strong>Dominating set differential abundance analysis revealed genome segments that were significantly different in CD and UC compared to nonIBD.</strong> Results are organized by GTDB taxonomy, with a tree representing the 54 species and colored by family on the far left. The percent of dominating set pieces tested is labelled in grey, and the percent of significantly differentially abundant pieces are colored by increased (blue) or decreased (brown) abundance.</figcaption>
</figure>
</div>
<p>We found that the majority of species decreased in abundance in CD, and to a lesser extent, UC (<strong>Figure <a href="#fig:dda">5</a></strong>).
Many of these species are generally regarded as beneficial bacteria.
For example, nine of the 54 genomes we investigated were <em>Faecalibacterium prausnitzii</em>, the phylogroups of which are separated in the GTDB taxonomy but combined into a single species in the NCBI taxonomy.
<em>F. prausnitzii</em> is a key butyrate producer in the gut and plays a crucial role in reducing intestinal inflammation <span class="citation" data-cites="gmUTraZV">[<a href="#ref-gmUTraZV" role="doc-biblioref">43</a>]</span>.
Similarly, <em>Acetatifactor</em> is a bile-acid producing bacteria associated with a healthy gut, but limited evidence has associated it with decreased abundance in IBD <span class="citation" data-cites="JSiu3OlL">[<a href="#ref-JSiu3OlL" role="doc-biblioref">44</a>]</span>.
These species, as well as others that decreased in abundance in IBD, are strictly anaerobic (<strong>Figure <a href="#fig:dda">5</a></strong>), so these observed trends are consistent with a shift toward oxidative stress during disease that is intolerable for many gut microbes <span class="citation" data-cites="1HrlmihRx">[<a href="#ref-1HrlmihRx" role="doc-biblioref">45</a>]</span>.</p>
<!--*Gemmiger formicilis* had the most dominating set pieces decrease in abundance in UC, and was also decreased in CD (**Figure @fig:dda**).
*G. formicilis* is a strictly anaerobic bacteria that produces both formic acid and butyric acid [@doi:10.1099/00207713-25-2-202]. 
See 10.3390/cells10113243 for new association with Gemmiger formicilis being reduced in pouchitis.
-->
<!-- strict anaerobe citations:
*F. prausnitzii* is an obligate anaerobe  [@doi:10.1038/ismej.2016.176].
*F. prausnitzii* is extremely sensitive to oxygen, though it may be able to withstand oxygen exposure for up to 24 hours depending on the availability of metabolites for extracellular electron transfer [@doi:10.1038/ismej.2016.176].
*G. formicilis* is a strictly anaerobic bacteria that produces both formic acid and butyric acid [@doi:10.1099/00207713-25-2-202].
*Lachnospira eligens_A* is an obligate anaerobe that is unable to tolerate atmospheric oxygen for an hour [@doi:10.1186/s13073-017-0490-5]. 
-->
<p>A substantial fraction of dominating set pieces were more abundant in CD than nonIBD in five metapangenome graphs (<strong>Figure <a href="#fig:dda">5</a></strong>).
These graphs represented sequences from species <em>R. gnavus</em>, <em>Enterocloster bolteae</em>, <em>Enterocloster sp005845215</em>, <em>Enterocloster clostridioformis</em>, and <em>Enterocloster clostridioformis_A</em>.
We posit that the increased abundance in some genomic segments amid the decrease in abundance of others represents strain switching that occurs in CD.</p>
<!--
Only the increase in the distinct accessory genes is detectable amongst the backdrop of general loss of gene abundance due to decrease in other strains.
-->
<p>In support of this, when we annotated the differentially abundant pieces using KEGG orthologs, we found that in some cases pieces that were increased in abundance and pieces that were decreased in abundance were annotated with the same ortholog (mean = 1453.8, sd = 727.2 pieces representing mean = 64, sd = 23.8 orthologs per graph, <strong>Figure <a href="#fig:enrichment">S3</a></strong>, <strong>Table S3</strong>).
These genes likely represent the portion of the core genome shared by the strain(s) that is more abundant in CD and the strain(s) that is more abundant in nonIBD, but that is encoded by distinct sequences.
Some shared annotations encoded single copy marker genes <span class="citation" data-cites="zNvJhZsz">[<a href="#ref-zNvJhZsz" role="doc-biblioref">46</a>]</span>.
To confirm that multiple strains of the same species were represented by these sequences, we queried with these marker gene sequences, extracted the reads associated with those graph pieces, and mapped the reads back to the marker gene sequence.
We then selected reads that aligned to the same portion of the marker gene and contained single nucleotide variants, and BLASTed those reads against the NCBI nr database.
For the subset of reads that we tested, we confirmed that different strains of the same species were the best matches.
This demonstrates that mulitple strains of the same species were present in each species graph, and that these strains were differentially abundant in CD compared to nonIBD.</p>
<p>In contrast to the annotations that were identified among sequences that were increased and decreased in abundance, many orthologs were only annotated among the pieces that were increased in abundance in CD (mean = 1193.4, SD = 155.7, <strong>Table S4</strong>).
Many of the same pathways were enriched among these orthologs across the five species, including those associated with oxidative stress response (cysteine and methionine metabolism, the pentose phosphate pathway) (<strong>Figure <a href="#fig:enrichment">S3</a></strong>).
The oxidative branch of the pentose phosphate pathway regenerates NADPH, while cysteine is a precursor for the antioxidant glutathione.
Given this, we investigated the presence of reactive oxygen species-scavenging orthologs <span class="citation" data-cites="C0kIERQa">[<a href="#ref-C0kIERQa" role="doc-biblioref">47</a>]</span>; sequences encoding superoxide dismutase (K04565), thioredoxin reductase (K00384), and peroxiredoxin (K03386) were increased in abundance in CD for all five species.
Additionally, many enriched pathways were associated with virulence (quorum sensing, flagellar assembly, bacterial chemotaxis, vancomycin resistance).
Enrichment of specific metabolic pathways is consistent with functional specialization of strains in different environmental niches <span class="citation" data-cites="5qxIIksl">[<a href="#ref-5qxIIksl" role="doc-biblioref">48</a>]</span>.</p>
<div id="tbl:sim" class="tablenos">
<table id="tbl:sim">
<caption><span>Table 2:</span> Maximum containment between sequences that were increased in abundance in CD and isolate genomes.
</caption>
<colgroup>
<col style="width: 38%" />
<col style="width: 44%" />
<col style="width: 16%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Metapangenome graph species</th>
<th style="text-align: left;">Closest strain match</th>
<th style="text-align: right;">Maximum containment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><em>Enterocloster clostridioformis</em></td>
<td style="text-align: left;"><em>Enterocloster clostridioformis</em> MSK.2.78</td>
<td style="text-align: right;">0.71</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>Enterocloster bolteae</em></td>
<td style="text-align: left;"><em>[Clostridium] bolteae</em> 90A5</td>
<td style="text-align: right;">0.68</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>Ruminococcus_B gnavus</em></td>
<td style="text-align: left;"><em>[Ruminococcus] gnavus</em> RJX1122</td>
<td style="text-align: right;">0.66</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>Enterocloster clostridioformis_A</em></td>
<td style="text-align: left;"><em>[Clostridium] clostridioforme</em> AGR2157</td>
<td style="text-align: right;">0.61</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>Enterocloster sp005845215</em></td>
<td style="text-align: left;"><em>Enterocloster clostridioformis</em> MSK.2.78</td>
<td style="text-align: right;">0.50</td>
</tr>
</tbody>
</table>
</div>
<p>While dominating set differential abundance analysis identified genomic sequences that were more abundant in CD, the nature of short shotgun metagenomic sequencing reads precludes haplotype phasing or lineage resolution <span class="citation" data-cites="10nUzxyHX">[<a href="#ref-10nUzxyHX" role="doc-biblioref">15</a>]</span>, meaning our results likely represent genomic variants from many distinct genomes that would not all naturally occur together in a single isolate genome.
Therefore, to identify isolate genomes that contain the genomic sequences that were more abundant in CD, we searched the GTDB rs202 database with the significantly differentially abundant sequences.
On average, the top matching isolate genome contained 63% of the sequences that were more abundant in CD (<strong>Table <a href="#tbl:sim">2</a></strong>).</p>
<p>One aerotolerant clade of <em>R. gnavus</em> was previously identified as being enriched in CD <span class="citation" data-cites="NznM4tWF">[<a href="#ref-NznM4tWF" role="doc-biblioref">12</a>]</span>, and has been shown to produce a polysaccharide that induces the inflammatory cytokine TNF-alpha <span class="citation" data-cites="17cWqPdre">[<a href="#ref-17cWqPdre" role="doc-biblioref">13</a>]</span>.
The three isolate genomes we identified as containing the highest amount of sequences that were increased in abundance in CD were among those that have been shown to induce TNF-alpha secretion (RJX1122, RJX1127, RJX1128) <span class="citation" data-cites="u0E0V8yd">[<a href="#ref-u0E0V8yd" role="doc-biblioref">14</a>]</span>. This suggests our method identified the same strain switch previously discovered to occur in IBD <span class="citation" data-cites="NznM4tWF 17cWqPdre u0E0V8yd">[<a href="#ref-NznM4tWF" role="doc-biblioref">12</a>,<a href="#ref-17cWqPdre" role="doc-biblioref">13</a>,<a href="#ref-u0E0V8yd" role="doc-biblioref">14</a>]</span>.
In further support of this, we found that 17 of the 23 genes in the operon that encodes the proteins responsible for producing the inflammatory polysaccharide were annotated in the dominating set pieces that were more abundant in CD.
These genes were encoded across 66 dominating set pieces, with multiple neighboring genes in the operon annotated in 6 of these dominating set pieces.</p>
<p>For two of the four <em>Enterocloster</em> species, the top matching isolate sequence was the same (<em>Enterocloster clostridioformis</em> MSK.2.78).
This points to overlap in the genomic sequences we identified as differentially abundant across these metapangenome graphs.
Indeed, the average Jaccard similarity between the sequences that were increased in CD in the <em>Enterocloster</em> graphs was 0.53, while the average max containment was 0.74.
Given that a Jaccard similarity of 0.1 is required to recover at least 80% of a genome via assembly graph query, which approximately corresponds to an average nucleotide identity of 93% <span class="citation" data-cites="63pA9pO7">[<a href="#ref-63pA9pO7" role="doc-biblioref">49</a>]</span>, and that species boundaries in GTDB are drawn at 95% average nucleotide identity <span class="citation" data-cites="vFApiGX2">[<a href="#ref-vFApiGX2" role="doc-biblioref">50</a>]</span>, the metapangenome graphs likely store genomic sequences associated with both the query genome species and closely related species.
<!-- are marker genes more likely to overlap? -->
However, the metapangenome graphs presented here, as well as the differentially abundant sequences, contain both common and distinct nucleotide sequences, suggesting that multiple closely related <em>Enterocloster</em> species/genomes are associated with CD.
<!-- supp tab Jaccard similarity and containment for all metapangenome graphs and for all the entero increased abund seqs). -->
Taken together, our ability to recover a previously validated sub-species association with IBD (<em>R. gnavus</em>) suggests that the three new <em>Enterocloster</em> isolates we identified should be further investigated for their potential role in eliciting CD-like symptoms in the gut.</p>
<!--
*C. bolteae* is a member of the normal gut microbiota but is an opportunistic pathogen that exploits compromised intestinal barriers [@doi:10.1186/s12864-016-3152-x].
It is associated with disturbance succession and has increased gene expression during gut dysbiosis [@doi:10.1101/gr.138198.112; @doi:10.1038/s41586-019-1237-9].
-->
<h2 id="genomic-sequences-that-are-differentially-abundant-in-ibd-are-not-exclusive-to-ibd">Genomic sequences that are differentially abundant in IBD are not exclusive to IBD</h2>
<p>Since genome sequences belonging to many species were differentially abundant from nonIBD in CD and UC, we next investigated whether there was a disease-specific microbiome in CD or UC – i.e., whether there are sequences from a species that were only observed in IBD.
Using FracMinHash sketches from the differentially abundant sequences, we identified the differentially abundant sequences in each metagenome and compared their occurrence and distribution across diagnoses.</p>
<p>In general, we found no evidence for disease-specific sequences among the 54 species we investigated.
Using FracMinHash sketches of the differentially abundant sequences for each species, we counted the number of k-mers that were observed in different sets of diagnoses.
We observed almost all sequences in at least some CD, UC, and nonIBD metagenomes (<strong>Figure <a href="#fig:exclusive">6</a></strong>, <strong>Figure <a href="#fig:exclusive2">S4</a></strong>).
Across all species, an average of 14.9% differentially abundant k-mers were unobserved in either CD, UC, or nonIBD.
These results in part explain heterogeneous study findings in previous IBD gut microbiome investigations.</p>
<div id="fig:exclusive" class="fignos">
<figure>
<img src="images/dda_kmers_shared_across_diagnosis_upset_increased.png" alt="Figure 6: Most differentially abundant sequences occur in metagenomes of individuals diagnosed with CD, UC and non-IBD. Upset plot of k-mers that were increased in abundance in CD and their occurrence in CD, UC, and nonIBD metagenomes. The bottom half of the plot highlights which diagnoses are included in each set, while the bar chart in the top half of the plot shows the number of k-mers that were observed in that set. The bar chart is colored by the metapangenome species graph in which the sequence was differentially abundant." /><figcaption aria-hidden="true"><span>Figure 6:</span> <strong>Most differentially abundant sequences occur in metagenomes of individuals diagnosed with CD, UC and non-IBD.</strong> Upset plot of k-mers that were increased in abundance in CD and their occurrence in CD, UC, and nonIBD metagenomes.
The bottom half of the plot highlights which diagnoses are included in each set, while the bar chart in the top half of the plot shows the number of k-mers that were observed in that set.
The bar chart is colored by the metapangenome species graph in which the sequence was differentially abundant.</figcaption>
</figure>
</div>
<h1 id="discussion">Discussion</h1>
<p>In this paper, we present an assembly-free metagenome analysis framework for group association discovery that is minimally reliant on reference databases.
Our approach uses k-mers to discover genomes associated with groups of metagenomes, and then recovers sequence variation from those genomes and closely related genomes in the metagenomes.
These sequences are organized in a “metapangenome graph” which is then used to perform differential abundance analysis to discover specific genomic sequences that differ between groups.</p>
<p>We applied this method to perform a meta-analysis of fecal gut microbiome metagenomes from individuals with CD, UC, and nonIBD and uncovered cross-study microbial signatures of IBD subtype.
The underlying etiology of IBD remains poorly understood with inconsistent microbiological findings produced from different studies <span class="citation" data-cites="jxRtm2IY">[<a href="#ref-jxRtm2IY" role="doc-biblioref">4</a>]</span>.
The signatures we identified demonstrate consistent loss of diversity of specific microorganisms, particularly in CD.
Among the background of generalized loss of diversity, we observed that some genomic sequences increased in abundance while others decreased in abundance for five species in CD.
This pattern is consistent with strain switching, where one strain is more abundant in CD and another is more abundant in nonIBD.
For one species we identified, <em>R. gnavus</em>, this strain switching behavior was previously discovered via isolate sequencing and metagenome mapping <span class="citation" data-cites="NznM4tWF">[<a href="#ref-NznM4tWF" role="doc-biblioref">12</a>]</span>.
Our recovery of this pattern demonstrates the utility of our approach for discovering sub-species level associations from metagenomic sequences alone, and opens the door for additional discovery.
Indeed, we identified four additional species where strain switching occurred in CD.</p>
<p>While our approach identified genomic sequences that were more abundant in CD than nonIBD, the nature of short read sequences precludes haplotype or lineage resolution directly from the metagenomic data analyzed here.
To circumvent this issue, we identified isolate genomes that encoded all of the genomic sequences that were more abundant in CD.
These isolate genomes represent candidate organisms for further research into the microbial component of CD pathophysiology.
As high fidelity long read sequencing of microbiomes becomes increasingly common <span class="citation" data-cites="10nUzxyHX">[<a href="#ref-10nUzxyHX" role="doc-biblioref">15</a>]</span>, long reads can be integrated into the approaches introduced here, enabling lineage-resolved association detection directly from sequencing data.</p>
<p>While we found conserved signatures in IBD subtype, we found no evidence for disease-specific sequences within the gut microbiome.
The observation that almost all differentially abundant sequences for a given species occur in CD, UC, and nonIBD suggests the presence of ecotypes – subspecies that are adapted to different environments – rather than pathotypes – subspecies associated with a specific disease.
These patterns in part explain the inconsistent results generated in IBD subtype characterization, where no consistent microbiological signal has emerged in human gut microbiomes other than loss of diversity <span class="citation" data-cites="jxRtm2IY">[<a href="#ref-jxRtm2IY" role="doc-biblioref">4</a>]</span>.</p>
<!-- I think this is a good point to include, but it doesn't really fit...does it fit well enough to include here as is? Would a different transition be nice? -->
<p>Our models consistently performed the most poorly on the iHMP cohort.
The iHMP tracked the emergence and diagnosis of IBD through time series profiling of emergent cases <span class="citation" data-cites="19Z0U0zL4">[<a href="#ref-19Z0U0zL4" role="doc-biblioref">5</a>]</span>.
We selected the first sample in each time series for this analysis, and given the relatively poor performance of these models, this may suggest that disease onset is a distinct biological process.
However, the inclusion of the iHMP cohort in this analysis ensured that not all nonIBD samples were healthy controls and some fraction were symptomatic cases that did not result in an IBD diagnosis <span class="citation" data-cites="19Z0U0zL4">[<a href="#ref-19Z0U0zL4" role="doc-biblioref">5</a>]</span>.</p>
<p>While we apply our pipeline to IBD classification, it is extensible to other large meta cohorts of metagenomic sequencing data.
This method may be particularly suitable for diseases such as colorectal cancer, where a recent meta-analysis using a marker gene approach was successful in classifying colorectal samples from healthy controls <span class="citation" data-cites="n90P8bOS">[<a href="#ref-n90P8bOS" role="doc-biblioref">32</a>]</span>.
Beyond classification of disease, our method may bring strain-level resolution and generate hypotheses for further research.
This may be particularly useful in the context of tumor microbiomes as previous research has demonstrated that strain-specific <em>Helicobacter pylori</em> and human papillomaviruses are risk factors for or directly transmit certain cancers <span class="citation" data-cites="ytQVjYqv">[<a href="#ref-ytQVjYqv" role="doc-biblioref">51</a>]</span>.
Strain-resolved methods may further this area of research.</p>
<p>The methods we used to perform the k-mer association analysis are modular and may be improved by substituting parts of the pipeline with different approaches.
For example, we used abundances from long nucleotide k-mers (<em>k</em> = 31) – which capture species-level sequence similarity <span class="citation" data-cites="mAjlUXL2">[<a href="#ref-mAjlUXL2" role="doc-biblioref">20</a>]</span> – as our features.
K-mers constructed from protein or other reduced alphabets may improve accuracy, as we would expect more shared sequence content between metagenomes as well as a better representation of functional content <span class="citation" data-cites="T2ylGIH0">[<a href="#ref-T2ylGIH0" role="doc-biblioref">52</a>]</span>.
While this may improve classification accuracy, switching to reduced alphabet k-mers may not be desirable in the context of strain-specific differences which may be obscured by these degenerate representations.
Similarly, while we used random forests to to perform k-mer association analysis, other machine learning or statistical techniques may improve classification accuracy.
These approaches remain areas of future research.</p>
<p>The first part of the pipeline is disconnectable from the second part of the pipeline – that is, the discovery of discriminatory genomes between groups is not a prerequisite for dominating set differential abundance analysis as query genomes could be selected arbitrarily.
Therefore, the assembly graph differential abundance approach presented here could be applied to metagenomes for samples originating from diverse environments.
The requirements for the application of dominating set differential abundance are threefold.
First, there must be sufficient samples for comparison (e.g., a minimum of three cases and three controls, with the typical caveats for small sample sizes <span class="citation" data-cites="CbLVMC4X">[<a href="#ref-CbLVMC4X" role="doc-biblioref">53</a>]</span>).
Second, we must have a genome with which to query the graph.
And third, we must have sufficient compute resources to run spacegraphcats <span class="citation" data-cites="VsLophuT">[<a href="#ref-VsLophuT" role="doc-biblioref">25</a>]</span>.
These requirements make the application of dominating set differential abundance analysis available to metagenomes from diverse environments, not just the well-studied human gut microbiome.</p>
<!--
+ assembly graphs are rising in popularity as they accommodate the representation of pangenomic variants present outside of reference genomes. 
+ assembly graphs can be built with a combination of long and short reads or from long or short reads alone, so dominating set differential abundance analysis is a flexible method to compare groups of metagenomes that will accommodate the field as metagenome sequencing continues to improve
+ the framework is theoretically extensible to metatranscriptomes, but this remains a point of future research
-->
<p>While we present an initial pipeline that enables differential abundance analysis directly on assembly graphs, we identified several areas where our approach could be improved.
First, implementing approaches to better control graph piece size would be beneficial.
We heuristically selected <em>r</em> = 10 to build the metapangenome graphs because this radius produced dominating set pieces with approximately the same number of k-mers as the average bacterial gene for pieces that were present in many metagenomes.
However, the radius needed to meet this condition may change depending on the diversity of the sequencing reads used to build the metapangenome graph.
Diversity increases with the complexity of the sequenced community, sequencing depth, and the number of communities observed.
Algorithms that either automatically select a radius that achieves a user-specified average piece size, or that produce more consistent piece sizes independent of diversity of the sequencing data would provide finer control of the graph structure and subsequent sequence comparisons.
Further, switching from a cDBG to a de Bruijn graph as the base spacegraphcats graph structure could lead to more consistent piece sizes; cDBGs have variable node sizes because they combine nodes without branching paths, while every node in a de Bruijn graph contains one k-mer.
Second, improving RAM efficiency at high radii would enable more diversity to be represented in individual graphs.
In order to build the metapangenome graphs, we first hard-trimmed the input sequences to remove low abundance k-mers, thereby decreasing the RAM needed to construct each graph.
Algorithmic changes that improve RAM efficiency at high radii would obviate the need for hard trimming, and increase the amount of diversity that could be represented in a single graph.
Similarly, improved performance would allow dominating set differential abundance analysis to be performed directly on groups of metagenomes without the need to first identify species of interest via genome queries.</p>
<h1 id="methods">Methods</h1>
<p>All code associated with our analyses is available at <a href="www.github.com/dib-lab/2020-ibd/">github.com/dib-lab/2020-ibd</a> (DOI: 10.5281/zenodo.6783208).
An example repository for dominating set differential abundance analysis is available at <a href="https://github.com/dib-lab/2022-dominating-set-differential-abundance-example/">github.com/dib-lab/2022-dominating-set-differential-abundance-example</a> (DOI: 10.5281/zenodo.6783363).</p>
<h2 id="ibd-metagenome-data-acquisition-and-processing">IBD metagenome data acquisition and processing</h2>
<p>We searched the NCBI Sequence Read Archive and BioProject databases for shotgun metagenome studies that sequenced fecal samples from humans with Crohn’s disease, ulcerative colitis, and healthy controls.
We included studies sequenced on Illumina platforms with paired-end chemistries and with sample libraries that contained greater than one million reads.
For time series intervention cohorts, we selected the first time point to ensure all metagenomes came from treatment-naive subjects.</p>
<p>We downloaded metagenomic FASTQ files from the European Nucleotide Archive using the “fastq_ftp” link and concatenated fast files annotated as the same library into single files.
We also downloaded iHMP samples from idbmdb.org.
We used Trimmomatic (version 0.39) to adapter trim reads using all default Trimmomatic paired-end adapter sequences (<code>ILLUMINACLIP:{inputs/adapters.fa}:2:0:15</code>) and lightly quality-trimmed the reads (<code>MINLEN:31 LEADING:2 TRAILING:2 SLIDINGWINDOW:4:2</code>) <span class="citation" data-cites="l4tjRGIB">[<a href="#ref-l4tjRGIB" role="doc-biblioref">54</a>]</span>.
We then removed human DNA using BBMap and a masked version of hg19 <span class="citation" data-cites="Q2EhhAmn">[<a href="#ref-Q2EhhAmn" role="doc-biblioref">55</a>]</span>.
Next, we trimmed low-abundance k-mers from sequences with high coverage using khmer’s <code>trim-low-abund.py</code> <span class="citation" data-cites="G2CzfHHF">[<a href="#ref-G2CzfHHF" role="doc-biblioref">56</a>]</span>.</p>
<p>Using these trimmed reads, we generated FracMinHash signatures for each library using sourmash (k-size 31, scaled 2000, abundance tracking on) <span class="citation" data-cites="ExvjTLL3">[<a href="#ref-ExvjTLL3" role="doc-biblioref">57</a>]</span>.
FracMinHash sketching produces compressed representations of k-mers in a metagenome while retaining the sequence diversity in a sample <span class="citation" data-cites="ART4gJKs IadOz4ol">[<a href="#ref-IadOz4ol" role="doc-biblioref">21</a>,<a href="#ref-ART4gJKs" role="doc-biblioref">23</a>]</span>.
This approach creates a consistent set of k-mers across samples by retaining the same k-mers when the same k-mers were observed.
This enables comparisons between metagenomes.
We refer to FracMinHash sketches as <em>signatures</em>, and to each sub-sampled k-mer in a signature as a <em>k-mer</em>.
At a scaled value of 2000, an average of one k-mer will be detected in each 2000 base pair window, and 99.8% of 10,000 base pair windows will have at least one k-mer representative.
We selected a k-mer size of 31 because of its species-level specificity <span class="citation" data-cites="mAjlUXL2">[<a href="#ref-mAjlUXL2" role="doc-biblioref">20</a>]</span>.
We retained all k-mers that were present in multiple samples.</p>
<h2 id="principle-coordinate-analysis">Principle Coordinate Analysis</h2>
<p>We used Jaccard distance and angular similarity as implemented in <code>sourmash compare</code> to pairwise compare FracMinHash signatures.
We then used the <code>dist()</code> function in base R to compute distance matrices.
We used the <code>cmdscale()</code> function to perform principle coordinate analysis <span class="citation" data-cites="gpnF30mH">[<a href="#ref-gpnF30mH" role="doc-biblioref">58</a>]</span>.
We used ggplot2 and ggMarginal to visualize the principle coordinate analysis <span class="citation" data-cites="EWll2pyF">[<a href="#ref-EWll2pyF" role="doc-biblioref">59</a>]</span>.
To test for sources of variation in these distance matrices, we performed PERMANOVA using the <code>adonis</code> function in the R vegan package <span class="citation" data-cites="NGto4o3z">[<a href="#ref-NGto4o3z" role="doc-biblioref">60</a>]</span>.
The PERMANOVA was modeled as <code>~ diagnosis + study accession + library size + number of k-mers</code>.</p>
<h2 id="random-forests-classifiers">Random forests classifiers</h2>
<!-- We built random forests classifiers to predict CD, UC, and non-IBD status using scaled MinHash signatures (k-mer models), marker genes in the shared 41 genomes (marker gene models), signatures from reads that were detected as marker genes (k-mer models of marker genes), and marker genes in the full metagenome (full marker gene models). -->
<p>We built random forests classifiers to predict CD, UC, and non-IBD status using FracMinHash signatures.
We transformed sourmash signatures into a k-mer (hash) abundance table where each metagenome was a sample, each k-mer was a feature, and abundances were recorded for each k-mer for each sample.
We normalized abundances by dividing by the total number of k-mers in each FracMinHash signature.
We then used a leave-one-study-out validation approach where we trained six models, each of which was trained on five studies and validated on the sixth.
We built each model six times, each time using a different random seed.
To build each model, we first performed vita variable selection on the training set as implemented in the Pomona and ranger packages <span class="citation" data-cites="y3cwAdm4 NZfzsGtQ">[<a href="#ref-y3cwAdm4" role="doc-biblioref">37</a>,<a href="#ref-NZfzsGtQ" role="doc-biblioref">61</a>]</span>.
Vita variable selection reduces the number of variables (e.g. k-mers) to a smaller set of predictive variables through selection of variables with high cross-validated permutation variable importance <span class="citation" data-cites="gePakmgr">[<a href="#ref-gePakmgr" role="doc-biblioref">36</a>]</span>.
It is based on permutation of variable importance, where p-values for variable importance are calculated against a null distribution that is built from variables that are estimated as non-important <span class="citation" data-cites="gePakmgr">[<a href="#ref-gePakmgr" role="doc-biblioref">36</a>]</span>.
This approach retains important variables that are correlated <span class="citation" data-cites="gePakmgr 18PJ2rKUg">[<a href="#ref-gePakmgr" role="doc-biblioref">36</a>,<a href="#ref-18PJ2rKUg" role="doc-biblioref">62</a>]</span>, which is desirable in omics-settings where correlated features are often involved in a coordinated biological response, e.g. part of the same operon, pathways, or genome <span class="citation" data-cites="uLmc3mM sI5BS4Sx">[<a href="#ref-uLmc3mM" role="doc-biblioref">63</a>,<a href="#ref-sI5BS4Sx" role="doc-biblioref">64</a>]</span>.
Using this smaller set of k-mers, we then built an optimized random forests model using tuneRanger <span class="citation" data-cites="204SJWYj">[<a href="#ref-204SJWYj" role="doc-biblioref">65</a>]</span>.
We evaluated each validation set using the optimal model, and extracted variable importance measures for each k-mer for subsequent analysis.
To make variable importance measures comparable across models, we normalized importance to 1 by dividing variable importance by the total number of k-mers in a model and the total number of models.</p>
<!-- For the marker gene models, we generated marker gene abundances for 14 ribosomal marker genes and 16S rRNA using singleM [@woodcroft2018singlem].
We then followed the same model building procedure as the k-mer models. -->
<h2 id="anchoring-predictive-k-mers-to-genomes">Anchoring predictive k-mers to genomes</h2>
<p>We used sourmash <code>gather</code> with parameters <code>k 31</code> and <code>--scaled 2000</code> to anchor predictive k-mers to known genomes <span class="citation" data-cites="ExvjTLL3">[<a href="#ref-ExvjTLL3" role="doc-biblioref">57</a>]</span>.
Sourmash <code>gather</code> searches a database of known k-mers for matches with a query <span class="citation" data-cites="IadOz4ol">[<a href="#ref-IadOz4ol" role="doc-biblioref">21</a>]</span>.
We used the sourmash GTDB rs202 representatives data base (https://osf.io/w4bcm/download).
To calculate the cumulative variable importance attributable to a single genome, we used an iterative winner-takes-all approach.
The genome with the largest fraction of predictive k-mers won the variable importance for all k-mers contained within its genome.
These k-mers were then removed, and we repeated the process for the genome with the next largest fraction of predictive k-mers.
To genomes that were predictive in all models, we took the union of predictive genomes from the 36 models.
We filtered this set of genomes to contain only those genomes with a cumulative normalized variable importance greater than 1%.</p>
<h2 id="r-dominating-sets"><em>r</em>-dominating sets</h2>
<p>The original spacegraphcats publication defined the dominating set as a set of nodes in the compact de Bruijn graph (cDBG) such that every node is a distance-1 neighbor of a node in the dominating set <span class="citation" data-cites="VsLophuT">[<a href="#ref-VsLophuT" role="doc-biblioref">25</a>]</span>.
However, the algorithms as implemented allow this distance to be flexible and tunable <span class="citation" data-cites="VsLophuT">[<a href="#ref-VsLophuT" role="doc-biblioref">25</a>]</span>.
We refer to the largest distance that any node may be from a member of the dominating set as the <em>radius</em>, <em>r</em>.
Increasing the radius increases the average piece size while reducing the total number of pieces in the graph.</p>
<h2 id="genome-neighborhood-queries-with-spacegraphcats">Genome neighborhood queries with spacegraphcats</h2>
<p>To recover sequence variation associated with genomes that were correlated with IBD subtype, we used spacegraphcats <code>search</code> to retrieve k-mers in the compact de Bruijn graph neighborhood of each genomes (<em>r</em> = 1, <em>k</em> = 31) <span class="citation" data-cites="VsLophuT">[<a href="#ref-VsLophuT" role="doc-biblioref">25</a>]</span>.
We then used spacegraphcats <code>extract_reads</code> to retrieve the reads and <code>extract_contigs</code> to retrieve unitigs in the cDBG that contained those k-mers, respectively.
<!-- These reads were used to generate marker gene abundances for the 41 shared genomes for the marker gene random forests models. --></p>
<h2 id="construction-of-the-metapangenome-graph">Construction of the metapangenome graph</h2>
<p>After retrieving genome neighborhood sequences from each metagenome, we combined these sequences to build a single metapangenome graph (<em>r</em> = 10, <em>k</em> = 31).
We increased the radius of the metapangenome graph to produce larger level 1 dominating set pieces and to overcome highly articulated cDBGs resulting from an abundance of sequencing data.
While working with single-species metapangenome graphs from many metagenomes reduced the graph size compared working with complete metapangenome graphs, we performed two preprocessing steps prior to the metapangenome graph generation.
We combined all genome query neighborhood reads and performed digital normalization and then truncated reads at k-mer that was not present in the data set at least 4 times.
These are heuristic steps that we believe are unlikely to remove biologically important sequences.</p>
<h2 id="annotating-the-metapangenome-graph">Annotating the metapangenome graph</h2>
<p>We implemented an approach to annotate dominating set pieces in spacegraphcats assembly graph.
This approach is implemented in spacegraphcats as <code>multifasta_queries</code>.
This approach relies on k-mer overlap between sequences in a reference multifasta file and nodes in the cDBG and is executed in a two-step approach.
First, <code>search.index_cdbg_by_multifasta.py</code> identifies all cDBG nodes that match to the k-mers in a FASTA sequence and promotes those annotations to all cDBG records in the dominating set piece.
Then, <code>search/extract_cdbg_by_multifasta.py</code> extracts and summarizes information about these annotations and outputs it to a CSV file.</p>
<p>We applied this annotation approach to the metapangenome graphs for species that were more abundant in CD.
To generate a reference multifasta gene file to transfer annotations from, we first downloaded all genomes of the species represented in the metapangenome graph and in the GTDB rs202 database.
We annotated open reading frames (ORFs) in these genomes using bakta <span class="citation" data-cites="wS8KrahX">[<a href="#ref-wS8KrahX" role="doc-biblioref">66</a>]</span>, combined and clustered predicted ORFs using cdhit-est <span class="citation" data-cites="vXPdBFEN">[<a href="#ref-vXPdBFEN" role="doc-biblioref">67</a>]</span>, and performed ortholog annotation using eggnog <span class="citation" data-cites="MqG4Noz7">[<a href="#ref-MqG4Noz7" role="doc-biblioref">68</a>]</span>.</p>
<h2 id="calculating-abundances-metagenome-abundances-of-dominating-set-nodes-in-the-metapangenome-graph">Calculating abundances metagenome abundances of dominating set nodes in the metapangenome graph</h2>
<p>We implemented an approach to calculate k-mer abundances for each graph piece in the level 1 dominating set.
This approach is implemented in spacegraphcats in <code>search/count_dominator_abundance.py</code>.
Using a spacegraphcats assembly graph and a set of reads from a metagenome, the abundance of each dominating set piece is calculated by summing the abundance of every k-mer in that piece within the metagenome.</p>
<p>We applied this abundance estimation approach to each metapangenome graph, estimating the abundance of each dominating set piece within each metagenome.</p>
<h2 id="performing-dominating-set-differential-abundance-analysis">Performing dominating set differential abundance analysis</h2>
<p>We used Corncob to perform dominating set differential abundance analysis <span class="citation" data-cites="UIA9vjDO">[<a href="#ref-UIA9vjDO" role="doc-biblioref">41</a>]</span>.
Corncob tests for differential relative abundance in the presence of variable sequencing depth and excessive zeroes for unobserved observations, conditions which occur in abundances from dominating sets <span class="citation" data-cites="UIA9vjDO">[<a href="#ref-UIA9vjDO" role="doc-biblioref">41</a>]</span>.
To focus on the most common sequencing variants and to reduce runtimes, we first filtered to dominating set pieces that were present in at least 100 (16.5%) metagenomes; corncob fits a model to each dominating set piece, so it does not require abundance information for all pieces.
We performed differential abundance testing using the <code>bbdml()</code> function using a likelihood ratio test with <code>formula = ~ study_accession + diagnosis</code> and <code>formula_null = ~study_accession</code>.
We estimated the number of k-mers in the quality controlled metagenome reads using ntcard and used this as the denominator.
We performed Bonferroni p value correction and used a significance cut off of 0.05.</p>
<p>To analyze the results of dominating set differential abundance analysis, we combined the significantly differentially abundant piece information with the results from the mulitfasta query annotations, and with the ortholog annotations for the multifasta query genomes.
We focused our analysis on KEGG orthologs.
When a single gene was annotated by eggnog with multiple KEGG orthologs, we selected the first match.
We identified the set of KEGG orthologs that were annotated among pieces that increased in abundance and pieces decreased in abundance in CD compared to nonIBD.
To identify single copy marker genes within this set, we used the marker gene sequences with an average copy of one in <span class="citation" data-cites="zNvJhZsz">[<a href="#ref-zNvJhZsz" role="doc-biblioref">46</a>]</span>.
We then selected at least one marker gene for each species, focusing on <code>rpl</code> sequences that were only annotated on two differentially abundant pieces in the graph (<em>Enterocloster clostridioformis</em>, <em>rplT</em>; <em>Ruminococcus_B gnavus</em>, <em>rplQ</em>; <em>Enterocloster clostridioformis_A</em>, <em>rplO</em>, <em>rplC</em>; <em>Enterocloster sp005845215</em>, <em>rplO</em>; <em>Enterocloster bolteae</em>, <em>rplC</em>).
We queried with the marker gene sequence that was used to perform the multifasta annotation and extracted the reads associated with that graph piece using <code>spacegraphcats extract_reads</code>.
We then mapped the extracted reads back to the query sequence using <code>bwa mem</code> <span class="citation" data-cites="SiisKKpX">[<a href="#ref-SiisKKpX" role="doc-biblioref">69</a>]</span>.
We visualized alignments in the Integrative Genomics Viewer <span class="citation" data-cites="F2D6lhmD">[<a href="#ref-F2D6lhmD" role="doc-biblioref">70</a>]</span> and selected reads that overlapped the same coordinates in the reference gene but that had different complements of single nucleotide polymorphisms.
We BLASTed these reads using blastn against the NCBI nr database and found the best strain-level matches.</p>
<p>We next identified KEGG orthologs that were only annotated in the either the pieces that were increased or decreased in abundance in CD compared to nonIBD.
We performed KEGG enrichment analysis using clusterProfiler <code>enricher</code> <span class="citation" data-cites="ScJScmR9">[<a href="#ref-ScJScmR9" role="doc-biblioref">71</a>]</span>, using TERM2GENE as all KEGG orthologs with pathway mappings and with argument <code>maxGSSize = 500</code>.
We considered pathways enriched to be enriched which had adjusted p values &lt; 0 .05.
Lastly, we searched for the presence of KEGG orthologs that quench reactive oxygen species using orthologs defined in <span class="citation" data-cites="C0kIERQa">[<a href="#ref-C0kIERQa" role="doc-biblioref">47</a>]</span>.</p>
<h2 id="searching-for-isolates-that-contained-differentially-abundant-genomic-sequences">Searching for isolates that contained differentially abundant genomic sequences</h2>
<p>To identify isolate genomes that contained sequences that were in CD, we searched the GTDB rs202 database.
We generated FracMinHash signatures (k = 31, scaled = 2000) of differentially abundant sequences using sourmash <code>sketch</code>.
We searched GTDB rs202 using sourmash <code>search</code>, using parameter <code>--max-containment</code>.
We filtered results to only include isolate genome sequences (e.g., removed metagenome-assembled genomes) and selected the top match as the best match.</p>
<h2 id="searching-for-metagenomes-that-contained-differentially-abundant-genome-sequences">Searching for metagenomes that contained differentially abundant genome sequences</h2>
<p>We intersected FracMinHash signatures (<em>k</em> = 31, scaled = 2000) of differentially abundant sequences and query neighborhoods for each genome query, producing hashes that were differentially abundant and observed within each metagenome.
We combined these hashes across diagnosis conditions (CD, UC, and nonIBD) and used the complexUpset R package to visualize the intersection size across conditions.</p>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-KY2dovBD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline"><strong>Genome sequencing of environmental &lt;i&gt;Escherichia coli&lt;/i&gt; expands understanding of the ecology and speciation of the model bacterial species</strong> <div class="csl-block">Chengwei Luo, Seth T Walk, David M Gordon, Michael Feldgarden, James M Tiedje, Konstantinos T Konstantinidis</div> <em>Proceedings of the National Academy of Sciences</em> (2011-04-26) <a href="https://doi.org/cvhvmz">https://doi.org/cvhvmz</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1073/pnas.1015622108">10.1073/pnas.1015622108</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21482770">21482770</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3084108">PMC3084108</a></div></div>
</div>
<div id="ref-nSWbl26Q" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline"><strong>Multiple levels of the unknown in microbiome research</strong> <div class="csl-block">Andrew Maltez Thomas, Nicola Segata</div> <em>BMC Biology</em> (2019-12) <a href="https://doi.org/gnm4t7">https://doi.org/gnm4t7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12915-019-0667-z">10.1186/s12915-019-0667-z</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31189463">31189463</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6560723">PMC6560723</a></div></div>
</div>
<div id="ref-9nrllDuS" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline"><strong>The Microbiome in Inflammatory Bowel Disease: Current Status and the Future Ahead</strong> <div class="csl-block">Aleksandar D Kostic, Ramnik J Xavier, Dirk Gevers</div> <em>Gastroenterology</em> (2014-05) <a href="https://doi.org/f2rggj">https://doi.org/f2rggj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1053/j.gastro.2014.02.009">10.1053/j.gastro.2014.02.009</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24560869">24560869</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4034132">PMC4034132</a></div></div>
</div>
<div id="ref-jxRtm2IY" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline"><strong>Integrating omics for a better understanding of Inflammatory Bowel Disease: a step towards personalized medicine</strong> <div class="csl-block">Manoj Kumar, Mathieu Garand, Souhaila Al Khodor</div> <em>Journal of Translational Medicine</em> (2019-12) <a href="https://doi.org/gnm4t8">https://doi.org/gnm4t8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12967-019-02174-1">10.1186/s12967-019-02174-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31836022">31836022</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6909475">PMC6909475</a></div></div>
</div>
<div id="ref-19Z0U0zL4" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline"><strong>Multi-omics of the gut microbial ecosystem in inflammatory bowel diseases</strong> <div class="csl-block">IBDMDB Investigators, Jason Lloyd-Price, Cesar Arze, Ashwin N Ananthakrishnan, Melanie Schirmer, Julian Avila-Pacheco, Tiffany W Poon, Elizabeth Andrews, Nadim J Ajami, Kevin S Bonham, … Curtis Huttenhower</div> <em>Nature</em> (2019-05) <a href="https://doi.org/ggd6wc">https://doi.org/ggd6wc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41586-019-1237-9">10.1038/s41586-019-1237-9</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31142855">31142855</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6650278">PMC6650278</a></div></div>
</div>
<div id="ref-1DalQm9EV" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline"><strong>Dysbiosis of fecal microbiota in Crohnʼs disease patients as revealed by a custom phylogenetic microarray:</strong> <div class="csl-block">Seungha Kang, Stuart E Denman, Mark Morrison, Zhongtang Yu, Joel Dore, Marion Leclerc, Chris S McSweeney</div> <em>Inflammatory Bowel Diseases</em> (2010-12) <a href="https://doi.org/ckh8bd">https://doi.org/ckh8bd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/ibd.21319">10.1002/ibd.21319</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/20848492">20848492</a></div></div>
</div>
<div id="ref-VDWIC49v" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline"><strong>A decrease of the butyrate-producing species &lt;i&gt;Roseburia hominis&lt;/i&gt; and &lt;i&gt;Faecalibacterium prausnitzii&lt;/i&gt; defines dysbiosis in patients with ulcerative colitis</strong> <div class="csl-block">Kathleen Machiels, Marie Joossens, João Sabino, Vicky De Preter, Ingrid Arijs, Venessa Eeckhaut, Vera Ballet, Karolien Claes, Filip Van Immerseel, Kristin Verbeke, … Séverine Vermeire</div> <em>Gut</em> (2014-08) <a href="https://doi.org/f59nf3">https://doi.org/f59nf3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1136/gutjnl-2013-304833">10.1136/gutjnl-2013-304833</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24021287">24021287</a></div></div>
</div>
<div id="ref-CwmUZE5B" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline"><strong>Inflammation, Antibiotics, and Diet as Environmental Stressors of the Gut Microbiome in Pediatric Crohn’s Disease</strong> <div class="csl-block">James D Lewis, Eric Z Chen, Robert N Baldassano, Anthony R Otley, Anne M Griffiths, Dale Lee, Kyle Bittinger, Aubrey Bailey, Elliot S Friedman, Christian Hoffmann, … Frederic D Bushman</div> <em>Cell Host &amp; Microbe</em> (2015-10) <a href="https://doi.org/f7zp6n">https://doi.org/f7zp6n</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.chom.2015.09.008">10.1016/j.chom.2015.09.008</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26468751">26468751</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4633303">PMC4633303</a></div></div>
</div>
<div id="ref-1DAvhqb44" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline"><strong>Genetic risk, dysbiosis, and treatment stratification using host genome and gut microbiome in inflammatory bowel disease</strong> <div class="csl-block">Ahmed Moustafa, Weizhong Li, Ericka L Anderson, Emily HM Wong, Parambir S Dulai, William J Sandborn, William Biggs, Shibu Yooseph, Marcus B Jones, Craig J Venter, … Brigid S Boland</div> <em>Clinical and Translational Gastroenterology</em> (2018-01) <a href="https://doi.org/gctt4v">https://doi.org/gctt4v</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ctg.2017.58">10.1038/ctg.2017.58</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29345635">29345635</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5795019">PMC5795019</a></div></div>
</div>
<div id="ref-17MF0XTS9" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline"><strong>A human gut microbial gene catalogue established by metagenomic sequencing</strong> <div class="csl-block">MetaHIT Consortium, Junjie Qin, Ruiqiang Li, Jeroen Raes, Manimozhiyan Arumugam, Kristoffer Solvsten Burgdorf, Chaysavanh Manichanh, Trine Nielsen, Nicolas Pons, Florence Levenez, … Jun Wang</div> <em>Nature</em> (2010-03) <a href="https://doi.org/dpw2s3">https://doi.org/dpw2s3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nature08821">10.1038/nature08821</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/20203603">20203603</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3779803">PMC3779803</a></div></div>
</div>
<div id="ref-ilcqfnau" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline"><strong>Mechanisms and consequences of intestinal dysbiosis</strong> <div class="csl-block">GAdrienne Weiss, Thierry Hennet</div> <em>Cellular and Molecular Life Sciences</em> (2017-08) <a href="https://doi.org/gj9fxf">https://doi.org/gj9fxf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/s00018-017-2509-x">10.1007/s00018-017-2509-x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28352996">28352996</a></div></div>
</div>
<div id="ref-NznM4tWF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline"><strong>A novel Ruminococcus gnavus clade enriched in inflammatory bowel disease patients</strong> <div class="csl-block">Andrew Brantley Hall, Moran Yassour, Jenny Sauk, Ashley Garner, Xiaofang Jiang, Timothy Arthur, Georgia K Lagoudas, Tommi Vatanen, Nadine Fornelos, Robin Wilson, … Curtis Huttenhower</div> <em>Genome Medicine</em> (2017-12) <a href="https://doi.org/gnm4t9">https://doi.org/gnm4t9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13073-017-0490-5">10.1186/s13073-017-0490-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29183332">29183332</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5704459">PMC5704459</a></div></div>
</div>
<div id="ref-17cWqPdre" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline"><strong>&lt;i&gt;Ruminococcus gnavus&lt;/i&gt; , a member of the human gut microbiome associated with Crohn’s disease, produces an inflammatory polysaccharide</strong> <div class="csl-block">Matthew T Henke, Douglas J Kenny, Chelsi D Cassilly, Hera Vlamakis, Ramnik J Xavier, Jon Clardy</div> <em>Proceedings of the National Academy of Sciences</em> (2019-06-25) <a href="https://doi.org/ghzzmg">https://doi.org/ghzzmg</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1073/pnas.1904099116">10.1073/pnas.1904099116</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31182571">31182571</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6601261">PMC6601261</a></div></div>
</div>
<div id="ref-u0E0V8yd" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline"><strong>Capsular polysaccharide correlates with immune response to the human gut microbe &lt;i&gt;Ruminococcus gnavus&lt;/i&gt;</strong> <div class="csl-block">Matthew T Henke, Eric M Brown, Chelsi D Cassilly, Hera Vlamakis, Ramnik J Xavier, Jon Clardy</div> <em>Proceedings of the National Academy of Sciences</em> (2021-05-18) <a href="https://doi.org/gnq78g">https://doi.org/gnq78g</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1073/pnas.2007595118">10.1073/pnas.2007595118</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33972416">33972416</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8157926">PMC8157926</a></div></div>
</div>
<div id="ref-10nUzxyHX" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline"><strong>Generating lineage-resolved, complete metagenome-assembled genomes from complex microbial communities</strong> <div class="csl-block">Derek M Bickhart, Mikhail Kolmogorov, Elizabeth Tseng, Daniel M Portik, Anton Korobeynikov, Ivan Tolstoganov, Gherman Uritskiy, Ivan Liachko, Shawn T Sullivan, Sung Bong Shin, … Timothy PL Smith</div> <em>Nature Biotechnology</em> (2022-05) <a href="https://doi.org/gn3fkx">https://doi.org/gn3fkx</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41587-021-01130-z">10.1038/s41587-021-01130-z</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34980911">34980911</a></div></div>
</div>
<div id="ref-yvK2iCjv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline"><strong>Genomic variation landscape of the human gut microbiome</strong> <div class="csl-block">Siegfried Schloissnig, Manimozhiyan Arumugam, Shinichi Sunagawa, Makedonka Mitreva, Julien Tap, Ana Zhu, Alison Waller, Daniel R Mende, Jens Roat Kultima, John Martin, … Peer Bork</div> <em>Nature</em> (2013-01) <a href="https://doi.org/j5d">https://doi.org/j5d</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nature11711">10.1038/nature11711</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23222524">23222524</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3536929">PMC3536929</a></div></div>
</div>
<div id="ref-1FQH2cgLx" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline"><strong>Genome-wide association study identifies vitamin B &lt;sub&gt;5&lt;/sub&gt; biosynthesis as a host specificity factor in &lt;i&gt;Campylobacter&lt;/i&gt;</strong> <div class="csl-block">Samuel K Sheppard, Xavier Didelot, Guillaume Meric, Alicia Torralbo, Keith A Jolley, David J Kelly, Stephen D Bentley, Martin CJ Maiden, Julian Parkhill, Daniel Falush</div> <em>Proceedings of the National Academy of Sciences</em> (2013-07-16) <a href="https://doi.org/f4562b">https://doi.org/f4562b</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1073/pnas.1305559110">10.1073/pnas.1305559110</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23818615">23818615</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3718156">PMC3718156</a></div></div>
</div>
<div id="ref-G7YuT52J" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline"><strong>Assessment of k-mer spectrum applicability for metagenomic dissimilarity analysis</strong> <div class="csl-block">Veronika B Dubinkina, Dmitry S Ischenko, Vladimir I Ulyantsev, Alexander V Tyakht, Dmitry G Alexeev</div> <em>BMC Bioinformatics</em> (2016-12) <a href="https://doi.org/gk7ks3">https://doi.org/gk7ks3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12859-015-0875-7">10.1186/s12859-015-0875-7</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26774270">26774270</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4715287">PMC4715287</a></div></div>
</div>
<div id="ref-U7VsiwaU" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline"><strong>Kevlar: A Mapping-Free Framework for Accurate Discovery of De Novo Variants</strong> <div class="csl-block">Daniel S Standage, CTitus Brown, Fereydoun Hormozdiari</div> <em>iScience</em> (2019-08) <a href="https://doi.org/ghfc63">https://doi.org/ghfc63</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.isci.2019.07.032">10.1016/j.isci.2019.07.032</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31377530">31377530</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6682328">PMC6682328</a></div></div>
</div>
<div id="ref-mAjlUXL2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline"><strong>MetaPalette: a &lt;i&gt;k&lt;/i&gt; -mer Painting Approach for Metagenomic Taxonomic Profiling and Quantification of Novel Strain Variation</strong> <div class="csl-block">David Koslicki, Daniel Falush</div> <em>mSystems</em> (2016-06-28) <a href="https://doi.org/gg3gbd">https://doi.org/gg3gbd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1128/msystems.00020-16">10.1128/msystems.00020-16</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27822531">27822531</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5069763">PMC5069763</a></div></div>
</div>
<div id="ref-IadOz4ol" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline"><strong>Lightweight compositional analysis of metagenomes with FracMinHash and minimum metagenome covers</strong> <div class="csl-block">Luiz Irber, Phillip T Brooks, Taylor Reiter, NTessa Pierce-Ward, Mahmudur Rahman Hera, David Koslicki, CTitus Brown</div> <em>Bioinformatics</em> (2022-01-12) <a href="https://doi.org/gn34zt">https://doi.org/gn34zt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2022.01.11.475838">10.1101/2022.01.11.475838</a></div></div>
</div>
<div id="ref-fFEMc6sb" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline"><strong>Multiple comparative metagenomics using multiset &lt;i&gt;k&lt;/i&gt; -mer counting</strong> <div class="csl-block">Gaëtan Benoit, Pierre Peterlongo, Mahendra Mariadassou, Erwan Drezen, Sophie Schbath, Dominique Lavenier, Claire Lemaitre</div> <em>PeerJ Computer Science</em> (2016-11-14) <a href="https://doi.org/gnm4vb">https://doi.org/gnm4vb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.7717/peerj-cs.94">10.7717/peerj-cs.94</a></div></div>
</div>
<div id="ref-ART4gJKs" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline"><strong>Large-scale sequence comparisons with sourmash</strong> <div class="csl-block">NTessa Pierce, Luiz Irber, Taylor Reiter, Phillip Brooks, CTitus Brown</div> <em>F1000Research</em> (2019-07-04) <a href="https://doi.org/gf9v84">https://doi.org/gf9v84</a> <div class="csl-block">DOI: <a href="https://doi.org/10.12688/f1000research.19675.1">10.12688/f1000research.19675.1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31508216">31508216</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6720031">PMC6720031</a></div></div>
</div>
<div id="ref-cHw8gtst" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline"><strong>When the levee breaks: a practical guide to sketching algorithms for processing the flood of genomic data</strong> <div class="csl-block">Will PM Rowe</div> <em>Genome Biology</em> (2019-12) <a href="https://doi.org/gf8bfj">https://doi.org/gf8bfj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-019-1809-x">10.1186/s13059-019-1809-x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31519212">31519212</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6744645">PMC6744645</a></div></div>
</div>
<div id="ref-VsLophuT" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline"><strong>Exploring neighborhoods in large metagenome assembly graphs using spacegraphcats reveals hidden sequence diversity</strong> <div class="csl-block">CTitus Brown, Dominik Moritz, Michael P O’Brien, Felix Reidl, Taylor Reiter, Blair D Sullivan</div> <em>Genome Biology</em> (2020-12) <a href="https://doi.org/d4bb">https://doi.org/d4bb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-020-02066-4">10.1186/s13059-020-02066-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32631445">32631445</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7336657">PMC7336657</a></div></div>
</div>
<div id="ref-Qc2FKvHp" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline"><strong>A fast and agnostic method for bacterial genome-wide association studies: Bridging the gap between k-mers and genetic events</strong> <div class="csl-block">Magali Jaillard, Leandro Lima, Maud Tournoud, Pierre Mahé, Alex van Belkum, Vincent Lacroix, Laurent Jacob</div> <em>PLOS Genetics</em> (2018-11-12) <a href="https://doi.org/gjjs4c">https://doi.org/gjjs4c</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pgen.1007758">10.1371/journal.pgen.1007758</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30419019">30419019</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6258240">PMC6258240</a></div></div>
</div>
<div id="ref-jroubkzG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline"><strong>Genome-resolved metagenomics identifies genetic mobility, metabolic interactions, and unexpected diversity in perchlorate-reducing communities</strong> <div class="csl-block">Tyler P Barnum, Israel A Figueroa, Charlotte I Carlström, Lauren N Lucas, Anna L Engelbrektson, John D Coates</div> <em>The ISME Journal</em> (2018-06) <a href="https://doi.org/gdms93">https://doi.org/gdms93</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41396-018-0081-5">10.1038/s41396-018-0081-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29476141">29476141</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5955982">PMC5955982</a></div></div>
</div>
<div id="ref-g2bc2Oy5" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline"><strong>MetaCherchant: analyzing genomic context of antibiotic resistance genes in gut microbiota</strong> <div class="csl-block">Evgenii I Olekhnovich, Artem T Vasilyev, Vladimir I Ulyantsev, Elena S Kostryukova, Alexander V Tyakht</div> <em>Bioinformatics</em> (2018-02-01) <a href="https://doi.org/gcg2gv">https://doi.org/gcg2gv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btx681">10.1093/bioinformatics/btx681</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29092015">29092015</a></div></div>
</div>
<div id="ref-s8orJ7KE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline"><strong>Improving MinHash via the containment index with applications to metagenomic analysis</strong> <div class="csl-block">David Koslicki, Hooman Zabeti</div> <em>Applied Mathematics and Computation</em> (2019-08) <a href="https://doi.org/ghtqrv">https://doi.org/ghtqrv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.amc.2019.02.018">10.1016/j.amc.2019.02.018</a></div></div>
</div>
<div id="ref-Xi4Nccqm" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline"><strong>Gut microbiome structure and metabolic activity in inflammatory bowel disease</strong> <div class="csl-block">Eric A Franzosa, Alexandra Sirota-Madi, Julian Avila-Pacheco, Nadine Fornelos, Henry J Haiser, Stefan Reinker, Tommi Vatanen, ABrantley Hall, Himel Mallick, Lauren J McIver, … Ramnik J Xavier</div> <em>Nature Microbiology</em> (2019-02) <a href="https://doi.org/gf9727">https://doi.org/gf9727</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41564-018-0306-4">10.1038/s41564-018-0306-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30531976">30531976</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6342642">PMC6342642</a></div></div>
</div>
<div id="ref-dSDu60g3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline"><strong>The Treatment-Naive Microbiome in New-Onset Crohn’s Disease</strong> <div class="csl-block">Dirk Gevers, Subra Kugathasan, Lee A Denson, Yoshiki Vázquez-Baeza, Will Van Treuren, Boyu Ren, Emma Schwager, Dan Knights, Se Jin Song, Moran Yassour, … Ramnik J Xavier</div> <em>Cell Host &amp; Microbe</em> (2014-03) <a href="https://doi.org/f5vq7x">https://doi.org/f5vq7x</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.chom.2014.02.005">10.1016/j.chom.2014.02.005</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24629344">24629344</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4059512">PMC4059512</a></div></div>
</div>
<div id="ref-n90P8bOS" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline"><strong>Meta-analysis of fecal metagenomes reveals global microbial signatures that are specific for colorectal cancer</strong> <div class="csl-block">Jakob Wirbel, Paul Theodor Pyl, Ece Kartal, Konrad Zych, Alireza Kashani, Alessio Milanese, Jonas S Fleck, Anita Y Voigt, Albert Palleja, Ruby Ponnudurai, … Georg Zeller</div> <em>Nature Medicine</em> (2019-04) <a href="https://doi.org/gfxrv9">https://doi.org/gfxrv9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41591-019-0406-6">10.1038/s41591-019-0406-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30936547">30936547</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7984229">PMC7984229</a></div></div>
</div>
<div id="ref-19itPpFkZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline"><strong>Metagenomic analysis of colorectal cancer datasets identifies cross-cohort microbial diagnostic signatures and a link with choline degradation</strong> <div class="csl-block">Andrew Maltez Thomas, Paolo Manghi, Francesco Asnicar, Edoardo Pasolli, Federica Armanini, Moreno Zolfo, Francesco Beghini, Serena Manara, Nicolai Karcher, Chiara Pozzi, … Nicola Segata</div> <em>Nature Medicine</em> (2019-04) <a href="https://doi.org/gfxrv6">https://doi.org/gfxrv6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41591-019-0405-7">10.1038/s41591-019-0405-7</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30936548">30936548</a></div></div>
</div>
<div id="ref-17XxJx4ep" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline"><strong>A Metagenomic Meta-analysis Reveals Functional Signatures of Health and Disease in the Human Gut Microbiome</strong> <div class="csl-block">Courtney R Armour, Stephen Nayfach, Katherine S Pollard, Thomas J Sharpton</div> <em>mSystems</em> (2019-08-27) <a href="https://doi.org/ggrn32">https://doi.org/ggrn32</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1128/msystems.00332-18">10.1128/msystems.00332-18</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31098399">31098399</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6517693">PMC6517693</a></div></div>
</div>
<div id="ref-aKDEtlHU" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline"><strong>Microbial genes and pathways in inflammatory bowel disease</strong> <div class="csl-block">Melanie Schirmer, Ashley Garner, Hera Vlamakis, Ramnik J Xavier</div> <em>Nature Reviews Microbiology</em> (2019-08) <a href="https://doi.org/gf8tk6">https://doi.org/gf8tk6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41579-019-0213-6">10.1038/s41579-019-0213-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31249397">31249397</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6759048">PMC6759048</a></div></div>
</div>
<div id="ref-gePakmgr" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline"><strong>A computationally fast variable importance test for random forests for high-dimensional data</strong> <div class="csl-block">Silke Janitza, Ender Celik, Anne-Laure Boulesteix</div> <em>Advances in Data Analysis and Classification</em> (2018-12) <a href="https://doi.org/gdj8zn">https://doi.org/gdj8zn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/s11634-016-0276-4">10.1007/s11634-016-0276-4</a></div></div>
</div>
<div id="ref-y3cwAdm4" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline"><strong>Evaluation of variable selection methods for random forests and omics data sets</strong> <div class="csl-block">Frauke Degenhardt, Stephan Seifert, Silke Szymczak</div> <em>Briefings in Bioinformatics</em> (2019-03-25) <a href="https://doi.org/gdz6nz">https://doi.org/gdz6nz</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bib/bbx124">10.1093/bib/bbx124</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29045534">29045534</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6433899">PMC6433899</a></div></div>
</div>
<div id="ref-nKH9jczu" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline"><strong>Comparison of normalization methods for the analysis of metagenomic gene abundance data</strong> <div class="csl-block">Mariana Buongermino Pereira, Mikael Wallroth, Viktor Jonsson, Erik Kristiansson</div> <em>BMC Genomics</em> (2018-12) <a href="https://doi.org/gdmzhp">https://doi.org/gdmzhp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12864-018-4637-6">10.1186/s12864-018-4637-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29678163">29678163</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5910605">PMC5910605</a></div></div>
</div>
<div id="ref-1FH3KNlym" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline"><strong>Statistical evaluation of methods for identification of differentially abundant genes in comparative metagenomics</strong> <div class="csl-block">Viktor Jonsson, Tobias Österlund, Olle Nerman, Erik Kristiansson</div> <em>BMC Genomics</em> (2016-12) <a href="https://doi.org/f3p6xv">https://doi.org/f3p6xv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12864-016-2386-y">10.1186/s12864-016-2386-y</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26810311">26810311</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4727335">PMC4727335</a></div></div>
</div>
<div id="ref-1CjZRQMrc" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline"><strong>Exploring neighborhoods in large metagenome assembly graphs using spacegraphcats reveals hidden sequence diversity</strong> <div class="csl-block">CTitus Brown, Dominik Moritz, Michael P O’Brien, Felix Reidl, Taylor Reiter, Blair D Sullivan</div> <em>Genome Biology</em> (2020-12) <a href="https://doi.org/d4bb">https://doi.org/d4bb</a> <div class="csl-block">DOI: <a href="https://doi.org/doi.org/10.1186/s13059-020-02066-4">doi.org/10.1186/s13059-020-02066-4</a></div></div>
</div>
<div id="ref-UIA9vjDO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">41. </div><div class="csl-right-inline"><strong>Modeling microbial abundances and dysbiosis with beta-binomial regression</strong> <div class="csl-block">Bryan D Martin, Daniela Witten, Amy D Willis</div> <em>The Annals of Applied Statistics</em> (2020-03-01) <a href="https://doi.org/gg6825">https://doi.org/gg6825</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1214/19-aoas1283">10.1214/19-aoas1283</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32983313">32983313</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7514055">PMC7514055</a></div></div>
</div>
<div id="ref-17f69WRJu" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline"><strong>MetaGeneAnnotator: Detecting Species-Specific Patterns of Ribosomal Binding Site for Precise Gene Prediction in Anonymous Prokaryotic and Phage Genomes</strong> <div class="csl-block">H Noguchi, T Taniguchi, T Itoh</div> <em>DNA Research</em> (2008-10-17) <a href="https://doi.org/frz7bm">https://doi.org/frz7bm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/dnares/dsn027">10.1093/dnares/dsn027</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/18940874">18940874</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2608843">PMC2608843</a></div></div>
</div>
<div id="ref-gmUTraZV" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline"><strong>Faecalibacterium prausnitzii: from microbiology to diagnostics and prognostics</strong> <div class="csl-block">Mireia Lopez-Siles, Sylvia H Duncan, LJesús Garcia-Gil, Margarita Martinez-Medina</div> <em>The ISME Journal</em> (2017-04) <a href="https://doi.org/f9kfz3">https://doi.org/f9kfz3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ismej.2016.176">10.1038/ismej.2016.176</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28045459">28045459</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5364359">PMC5364359</a></div></div>
</div>
<div id="ref-JSiu3OlL" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">44. </div><div class="csl-right-inline"><strong>Intestine farnesoid X receptor agonist and the gut microbiota activate G‐protein bile acid receptor‐1 signaling to improve metabolism</strong> <div class="csl-block">Preeti Pathak, Cen Xie, Robert G Nichols, Jessica M Ferrell, Shannon Boehme, Kristopher W Krausz, Andrew D Patterson, Frank J Gonzalez, John YL Chiang</div> <em>Hepatology</em> (2018-10) <a href="https://doi.org/gkx66p">https://doi.org/gkx66p</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/hep.29857">10.1002/hep.29857</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29486523">29486523</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6111007">PMC6111007</a></div></div>
</div>
<div id="ref-1HrlmihRx" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">45. </div><div class="csl-right-inline"><strong>Dysbiosis in inflammatory bowel diseases: the oxygen hypothesis</strong> <div class="csl-block">Lionel Rigottier-Gois</div> <em>The ISME Journal</em> (2013-07) <a href="https://doi.org/f43frf">https://doi.org/f43frf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ismej.2013.80">10.1038/ismej.2013.80</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23677008">23677008</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3695303">PMC3695303</a></div></div>
</div>
<div id="ref-zNvJhZsz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">46. </div><div class="csl-right-inline"><strong>MUSiCC: a marker genes based framework for metagenomic normalization and accurate profiling of gene abundances in the microbiome</strong> <div class="csl-block">Ohad Manor, Elhanan Borenstein</div> <em>Genome Biology</em> (2015-12) <a href="https://doi.org/gp79v6">https://doi.org/gp79v6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-015-0610-8">10.1186/s13059-015-0610-8</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25885687">25885687</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4391136">PMC4391136</a></div></div>
</div>
<div id="ref-C0kIERQa" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">47. </div><div class="csl-right-inline"><strong>Distribution of reactive oxygen species defense mechanisms across domain bacteria</strong> <div class="csl-block">Lisa A Johnson, Laura A Hug</div> <em>Free Radical Biology and Medicine</em> (2019-08) <a href="https://doi.org/gf2s5j">https://doi.org/gf2s5j</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.freeradbiomed.2019.03.032">10.1016/j.freeradbiomed.2019.03.032</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30930298">30930298</a></div></div>
</div>
<div id="ref-5qxIIksl" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">48. </div><div class="csl-right-inline"><strong>Subspecies in the global human gut microbiome</strong> <div class="csl-block">Paul I Costea, Luis Pedro Coelho, Shinichi Sunagawa, Robin Munch, Jaime Huerta‐Cepas, Kristoffer Forslund, Falk Hildebrand, Almagul Kushugulova, Georg Zeller, Peer Bork</div> <em>Molecular Systems Biology</em> (2017-12) <a href="https://doi.org/gcpk4k">https://doi.org/gcpk4k</a> <div class="csl-block">DOI: <a href="https://doi.org/10.15252/msb.20177589">10.15252/msb.20177589</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29242367">29242367</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5740502">PMC5740502</a></div></div>
</div>
<div id="ref-63pA9pO7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">49. </div><div class="csl-right-inline"><strong>Protein k-mer analyses for assembly- and alignment-free sequence analysis</strong> <div class="csl-block">NTessa Pierce-Ward, Taylor E Reiter, CTitus Brown</div> <em>Manubot</em> (2022-06-22) <a href="https://bluegenes.github.io/2022-paper-protein-kmers/">https://bluegenes.github.io/2022-paper-protein-kmers/</a></div>
</div>
<div id="ref-vFApiGX2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">50. </div><div class="csl-right-inline"><strong>GTDB: an ongoing census of bacterial and archaeal diversity through a phylogenetically consistent, rank normalized and complete genome-based taxonomy</strong> <div class="csl-block">Donovan H Parks, Maria Chuvochina, Christian Rinke, Aaron J Mussig, Pierre-Alain Chaumeil, Philip Hugenholtz</div> <em>Nucleic Acids Research</em> (2022-01-07) <a href="https://doi.org/gm97d8">https://doi.org/gm97d8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkab776">10.1093/nar/gkab776</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34520557">34520557</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8728215">PMC8728215</a></div></div>
</div>
<div id="ref-ytQVjYqv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">51. </div><div class="csl-right-inline"><strong>New Insights Into the Cancer–Microbiome–Immune Axis: Decrypting a Decade of Discoveries</strong> <div class="csl-block">Tejeshwar Jain, Prateek Sharma, Abhi C Are, Selwyn M Vickers, Vikas Dudeja</div> <em>Frontiers in Immunology</em> (2021-02-23) <a href="https://doi.org/gmjrdg">https://doi.org/gmjrdg</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fimmu.2021.622064">10.3389/fimmu.2021.622064</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33708214">33708214</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7940198">PMC7940198</a></div></div>
</div>
<div id="ref-T2ylGIH0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">52. </div><div class="csl-right-inline"><strong>Protein k-mers enable assembly-free microbial metapangenomics</strong> <div class="csl-block">Taylor E Reiter, NTessa Pierce-Ward, Luiz Irber, Olga Borisovna Botvinnik, CTitus Brown</div> <em>Bioinformatics</em> (2022-06-27) <a href="https://doi.org/gqfjbh">https://doi.org/gqfjbh</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2022.06.27.497795">10.1101/2022.06.27.497795</a></div></div>
</div>
<div id="ref-CbLVMC4X" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">53. </div><div class="csl-right-inline"><strong>How many biological replicates are needed in an RNA-seq experiment and which differential expression tool should you use?</strong> <div class="csl-block">Nicholas J Schurch, Pietá Schofield, Marek Gierliński, Christian Cole, Alexander Sherstnev, Vijender Singh, Nicola Wrobel, Karim Gharbi, Gordon G Simpson, Tom Owen-Hughes, … Geoffrey J Barton</div> <em>RNA</em> (2016-06) <a href="https://doi.org/f8mrmk">https://doi.org/f8mrmk</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1261/rna.053959.115">10.1261/rna.053959.115</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27022035">27022035</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4878611">PMC4878611</a></div></div>
</div>
<div id="ref-l4tjRGIB" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">54. </div><div class="csl-right-inline"><strong>Trimmomatic: a flexible trimmer for Illumina sequence data</strong> <div class="csl-block">Anthony M Bolger, Marc Lohse, Bjoern Usadel</div> <em>Bioinformatics</em> (2014-08-01) <a href="https://doi.org/f6cj5w">https://doi.org/f6cj5w</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btu170">10.1093/bioinformatics/btu170</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24695404">24695404</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4103590">PMC4103590</a></div></div>
</div>
<div id="ref-Q2EhhAmn" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">55. </div><div class="csl-right-inline"><strong>Introducing RemoveHuman: Human Contaminant Removal</strong> <div class="csl-block">SEQanswers</div> <a href="https://www.seqanswers.com/forum/bioinformatics/bioinformatics-aa/37175-introducing-removehuman-human-contaminant-removal">https://www.seqanswers.com/forum/bioinformatics/bioinformatics-aa/37175-introducing-removehuman-human-contaminant-removal</a></div>
</div>
<div id="ref-G2CzfHHF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">56. </div><div class="csl-right-inline"><strong>The khmer software package: enabling efficient nucleotide sequence analysis</strong> <div class="csl-block">Michael R Crusoe, Hussien F Alameldin, Sherine Awad, Elmar Boucher, Adam Caldwell, Reed Cartwright, Amanda Charbonneau, Bede Constantinides, Greg Edvenson, Scott Fay, … CTitus Brown</div> <em>F1000Research</em> (2015-09-25) <a href="https://doi.org/9qp">https://doi.org/9qp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.12688/f1000research.6924.1">10.12688/f1000research.6924.1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26535114">26535114</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4608353">PMC4608353</a></div></div>
</div>
<div id="ref-ExvjTLL3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">57. </div><div class="csl-right-inline"><strong>sourmash: a library for MinHash sketching of DNA</strong> <div class="csl-block">C Titus Brown, Luiz Irber</div> <em>The Journal of Open Source Software</em> (2016-09-14) <a href="https://doi.org/ghdrk5">https://doi.org/ghdrk5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.21105/joss.00027">10.21105/joss.00027</a></div></div>
</div>
<div id="ref-gpnF30mH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">58. </div><div class="csl-right-inline"><strong>Some Distance Properties of Latent Root and Vector Methods Used in Multivariate Analysis</strong> <div class="csl-block">JC Gower</div> <em>Biometrika</em> (1966-12) <a href="https://doi.org/ch3msp">https://doi.org/ch3msp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.2307/2333639">10.2307/2333639</a></div></div>
</div>
<div id="ref-EWll2pyF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">59. </div><div class="csl-right-inline"><strong>Welcome to the Tidyverse</strong> <div class="csl-block">Hadley Wickham, Mara Averick, Jennifer Bryan, Winston Chang, Lucy McGowan, Romain François, Garrett Grolemund, Alex Hayes, Lionel Henry, Jim Hester, … Hiroaki Yutani</div> <em>Journal of Open Source Software</em> (2019-11-21) <a href="https://doi.org/ggddkj">https://doi.org/ggddkj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.21105/joss.01686">10.21105/joss.01686</a></div></div>
</div>
<div id="ref-NGto4o3z" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">60. </div><div class="csl-right-inline"><strong>vegan: Community Ecology Package</strong> <div class="csl-block">Jari Oksanen, Gavin L Simpson, FGuillaume Blanchet, Roeland Kindt, Pierre Legendre, Peter R Minchin, RB O'Hara, Peter Solymos, MHenry H Stevens, Eduard Szoecs, … James Weedon</div> (2022-04-17) <a href="https://CRAN.R-project.org/package=vegan">https://CRAN.R-project.org/package=vegan</a></div>
</div>
<div id="ref-NZfzsGtQ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">61. </div><div class="csl-right-inline"><strong>&lt;b&gt;ranger&lt;/b&gt; : A Fast Implementation of Random Forests for High Dimensional Data in &lt;i&gt;C++&lt;/i&gt; and &lt;i&gt;R&lt;/i&gt;</strong> <div class="csl-block">Marvin N Wright, Andreas Ziegler</div> <em>Journal of Statistical Software</em> (2017) <a href="https://doi.org/b8q3">https://doi.org/b8q3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.18637/jss.v077.i01">10.18637/jss.v077.i01</a></div></div>
</div>
<div id="ref-18PJ2rKUg" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">62. </div><div class="csl-right-inline"><strong>Surrogate minimal depth as an importance measure for variables in random forests</strong> <div class="csl-block">Stephan Seifert, Sven Gundlach, Silke Szymczak</div> <em>Bioinformatics</em> (2019-10-01) <a href="https://doi.org/gmmrnk">https://doi.org/gmmrnk</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btz149">10.1093/bioinformatics/btz149</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30824905">30824905</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6761946">PMC6761946</a></div></div>
</div>
<div id="ref-uLmc3mM" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">63. </div><div class="csl-right-inline"><strong>A Gene-Coexpression Network for Global Discovery of Conserved Genetic Modules</strong> <div class="csl-block">Joshua M Stuart, Eran Segal, Daphne Koller, Stuart K Kim</div> <em>Science</em> (2003-10-10) <a href="https://doi.org/bkfpd8">https://doi.org/bkfpd8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/science.1087447">10.1126/science.1087447</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/12934013">12934013</a></div></div>
</div>
<div id="ref-sI5BS4Sx" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">64. </div><div class="csl-right-inline"><strong>Co-expression pattern from DNA microarray experiments as a tool for operon prediction</strong> <div class="csl-block">C Sabatti</div> <em>Nucleic Acids Research</em> (2002-07-01) <a href="https://doi.org/d8zkdv">https://doi.org/d8zkdv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkf388">10.1093/nar/gkf388</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/12087173">12087173</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC117043">PMC117043</a></div></div>
</div>
<div id="ref-204SJWYj" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">65. </div><div class="csl-right-inline"><strong>Hyperparameters and tuning strategies for random forest</strong> <div class="csl-block">Philipp Probst, Marvin N Wright, Anne‐Laure Boulesteix</div> <em>WIREs Data Mining and Knowledge Discovery</em> (2019-05) <a href="https://doi.org/gf3sz2">https://doi.org/gf3sz2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/widm.1301">10.1002/widm.1301</a></div></div>
</div>
<div id="ref-wS8KrahX" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">66. </div><div class="csl-right-inline"><strong>Bakta: rapid and standardized annotation of bacterial genomes via alignment-free sequence identification</strong> <div class="csl-block">Oliver Schwengers, Lukas Jelonek, Marius Alfred Dieckmann, Sebastian Beyvers, Jochen Blom, Alexander Goesmann</div> <em>Microbial Genomics</em> (2021-11-05) <a href="https://doi.org/gnfrj7">https://doi.org/gnfrj7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1099/mgen.0.000685">10.1099/mgen.0.000685</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34739369">34739369</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8743544">PMC8743544</a></div></div>
</div>
<div id="ref-vXPdBFEN" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">67. </div><div class="csl-right-inline"><strong>Cd-hit: a fast program for clustering and comparing large sets of protein or nucleotide sequences</strong> <div class="csl-block">W Li, A Godzik</div> <em>Bioinformatics</em> (2006-07-01) <a href="https://doi.org/ct8g72">https://doi.org/ct8g72</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btl158">10.1093/bioinformatics/btl158</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/16731699">16731699</a></div></div>
</div>
<div id="ref-MqG4Noz7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">68. </div><div class="csl-right-inline"><strong>eggNOG 5.0: a hierarchical, functionally and phylogenetically annotated orthology resource based on 5090 organisms and 2502 viruses</strong> <div class="csl-block">Jaime Huerta-Cepas, Damian Szklarczyk, Davide Heller, Ana Hernández-Plaza, Sofia K Forslund, Helen Cook, Daniel R Mende, Ivica Letunic, Thomas Rattei, Lars J Jensen, … Peer Bork</div> <em>Nucleic Acids Research</em> (2019-01-08) <a href="https://doi.org/gg8bdg">https://doi.org/gg8bdg</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gky1085">10.1093/nar/gky1085</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30418610">30418610</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6324079">PMC6324079</a></div></div>
</div>
<div id="ref-SiisKKpX" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">69. </div><div class="csl-right-inline"><strong>Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM</strong> <div class="csl-block">Heng Li</div> <em>arXiv</em> (2013) <a href="https://doi.org/gpswnw">https://doi.org/gpswnw</a> <div class="csl-block">DOI: <a href="https://doi.org/10.48550/arxiv.1303.3997">10.48550/arxiv.1303.3997</a></div></div>
</div>
<div id="ref-F2D6lhmD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">70. </div><div class="csl-right-inline"><strong>Integrative Genomics Viewer (IGV): high-performance genomics data visualization and exploration</strong> <div class="csl-block">H Thorvaldsdottir, JT Robinson, JP Mesirov</div> <em>Briefings in Bioinformatics</em> (2013-03-01) <a href="https://doi.org/f4sc43">https://doi.org/f4sc43</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bib/bbs017">10.1093/bib/bbs017</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22517427">22517427</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3603213">PMC3603213</a></div></div>
</div>
<div id="ref-ScJScmR9" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">71. </div><div class="csl-right-inline"><strong>clusterProfiler: an R Package for Comparing Biological Themes Among Gene Clusters</strong> <div class="csl-block">Guangchuang Yu, Li-Gen Wang, Yanyan Han, Qing-Yu He</div> <em>OMICS: A Journal of Integrative Biology</em> (2012-05) <a href="https://doi.org/gdf33f">https://doi.org/gdf33f</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1089/omi.2011.0118">10.1089/omi.2011.0118</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22455463">22455463</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3339379">PMC3339379</a></div></div>
</div>
</div>
<h1 id="supplementary-information">Supplementary information</h1>
<div id="fig:tree" class="fignos">
<figure>
<img src="images/genome360.png" data-tag="S1" alt="Figure S1: Phylogenetic tree of 360 bacterial species that were predictive of IBD subtype in all models. Tree was built from the GTDB rs202 tree with all tips except those represented by the 360 genomes removed. Tree tips are labelled by genomes that anchored at least 1% of the normalized variable importance. The inner ring annotates the rank of the genomes, with the genome holding the most normalized variable importance across models ranked as 1. The outer ring is the species name within the GTDB database." /><figcaption aria-hidden="true"><span>Figure S1:</span> <strong>Phylogenetic tree of 360 bacterial species that were predictive of IBD subtype in all models.</strong> Tree was built from the GTDB rs202 tree with all tips except those represented by the 360 genomes removed. Tree tips are labelled by genomes that anchored at least 1% of the normalized variable importance. The inner ring annotates the rank of the genomes, with the genome holding the most normalized variable importance across models ranked as 1. The outer ring is the species name within the GTDB database.</figcaption>
</figure>
</div>
<div id="fig:varimp" class="fignos">
<figure>
<img src="images/varimp54.png" data-tag="S2" alt="Figure S2: Fifty-four genomes are important across models and anchor the majority of variable importance.. The bottom panel depicts a heat map of the scale variable importance contributed by k-mers that anchored to each of the top 54 genomes that were important for predicting IBD subtype. Models are labelled by the validation study and by the random seed used to build the model.Rank corresponds to the genome that anchored the most variable importance. Rank:species can be decoded using the tree in Figure S1. The top panels depict bar charts that correspond to the minimum (lower) or maximum (upper) variable importance a genome could anchor. The minimum variable importance was estimated following the sourmash gather algorithm, where each important k-mer was assigned to only one genome, and the genome it was assigned to was determined by a greedy winner-takes-all approach. Therefore, in the minimum bar chart, variable importance attributable to a k-mer was only summed once per k-mer, even if that k-mer occurred in multiple genomes. The maximum variable importance was estimated by allowing k-mers to be anchored to multiple genomes, so all k-mers were assigned to all possible genomes even if that meant a k-mer was assigned multiple times." /><figcaption aria-hidden="true"><span>Figure S2:</span> <strong>Fifty-four genomes are important across models and anchor the majority of variable importance.</strong>.
The bottom panel depicts a heat map of the scale variable importance contributed by k-mers that anchored to each of the top 54 genomes that were important for predicting IBD subtype.
Models are labelled by the validation study and by the random seed used to build the model.Rank corresponds to the genome that anchored the most variable importance.
Rank:species can be decoded using the tree in <strong>Figure <a href="#fig:tree">S1</a></strong>. The top panels depict bar charts that correspond to the minimum (lower) or maximum (upper) variable importance a genome could anchor.
The minimum variable importance was estimated following the sourmash gather algorithm, where each important k-mer was assigned to only one genome, and the genome it was assigned to was determined by a greedy winner-takes-all approach.
Therefore, in the minimum bar chart, variable importance attributable to a k-mer was only summed once per k-mer, even if that k-mer occurred in multiple genomes.
The maximum variable importance was estimated by allowing k-mers to be anchored to multiple genomes, so all k-mers were assigned to all possible genomes even if that meant a k-mer was assigned multiple times.</figcaption>
</figure>
</div>
<div id="fig:enrichment" class="fignos">
<figure>
<img src="images/sig_ccs_enrichment.png" data-tag="S3" style="height:9in" alt="Figure S3: Pathways that were enriched among sets of differentially abundant sequences in CD compared to nonIBD. The x axis represents the number of orthologs identified in the pathway, while the y axis annotates the pathway. Top: Some dominating set pieces that significantly increased in abundance were annotated as the same KEGG orthologs as dominating set pieces that were significantly decreased in abundance. Many of these pathways encode core functions. Middle: KEGG pathway enrichment from KEGG ortholog annotations that were only observed in dominating set pieces that were significantly increased in abundance in CD. Bottom: KEGG pathway enrichment from KEGG ortholog annotations that were only observed in dominating set pieces that were significantly decreased in abundance in CD." /><figcaption aria-hidden="true"><span>Figure S3:</span> <strong>Pathways that were enriched among sets of differentially abundant sequences in CD compared to nonIBD</strong>. The x axis represents the number of orthologs identified in the pathway, while the y axis annotates the pathway. Top: Some dominating set pieces that significantly increased in abundance were annotated as the same KEGG orthologs as dominating set pieces that were significantly decreased in abundance. Many of these pathways encode core functions. Middle: KEGG pathway enrichment from KEGG ortholog annotations that were only observed in dominating set pieces that were significantly increased in abundance in CD. Bottom: KEGG pathway enrichment from KEGG ortholog annotations that were only observed in dominating set pieces that were significantly decreased in abundance in CD.</figcaption>
</figure>
</div>
<div id="fig:exclusive2" class="fignos">
<figure>
<img src="images/dda_kmers_shared_across_diagnosis_upset_decreased_supplemental.png" data-tag="S4" alt="Figure S4: Most differentially abundant sequences occur in metagenomes of individuals diagnosed with CD, UC and non-IBD. Upset plot of k-mers that were decreased in abundance in CD and their occurrence in CD, UC, and nonIBD metagenomes. The bottom half of the plot highlights which diagnoses are included in each set, while the bar chart in the top half of the plot shows the number of k-mers that were observed in that set. The bar chart is colored by the metapangenome species graph in which the sequence was differentially abundant." /><figcaption aria-hidden="true"><span>Figure S4:</span> <strong>Most differentially abundant sequences occur in metagenomes of individuals diagnosed with CD, UC and non-IBD.</strong> Upset plot of k-mers that were decreased in abundance in CD and their occurrence in CD, UC, and nonIBD metagenomes.
The bottom half of the plot highlights which diagnoses are included in each set, while the bar chart in the top half of the plot shows the number of k-mers that were observed in that set.
The bar chart is colored by the metapangenome species graph in which the sequence was differentially abundant.</figcaption>
</figure>
</div>
<!-- default theme -->

<style>
  /* import google fonts */
  @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
  @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

  /* -------------------------------------------------- */
  /* global */
  /* -------------------------------------------------- */

  /* all elements */
  * {
    /* force sans-serif font unless specified otherwise */
    font-family: "Open Sans", "Helvetica", sans-serif;

    /* prevent text inflation on some mobile browsers */
    -webkit-text-size-adjust: none !important;
    -moz-text-size-adjust: none !important;
    -o-text-size-adjust: none !important;
    text-size-adjust: none !important;
  }

  @media only screen {
    /* "page" element */
    body {
      position: relative;
      box-sizing: border-box;
      font-size: 12pt;
      line-height: 1.5;
      max-width: 8.5in;
      margin: 20px auto;
      padding: 40px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* "page" element */
    body {
      padding: 20px;
      margin: 0;
      border-radius: 0;
      border: none;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
      background: none;
    }
  }

  /* -------------------------------------------------- */
  /* headings */
  /* -------------------------------------------------- */

  /* all headings */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 20px 0;
    padding: 0;
    font-weight: bold;
  }

  /* biggest heading */
  h1 {
    margin: 40px 0;
    text-align: center;
  }

  /* second biggest heading */
  h2 {
    margin-top: 30px;
    padding-bottom: 5px;
    border-bottom: solid 1px #bdbdbd;
  }

  /* heading font sizes */
  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.35em;
  }
  h4 {
    font-size: 1.25em;
  }
  h5 {
    font-size: 1.15em;
  }
  h6 {
    font-size: 1em;
  }

  /* -------------------------------------------------- */
  /* manuscript header */
  /* -------------------------------------------------- */

  /* manuscript title */
  header > h1 {
    margin: 0;
  }

  /* manuscript title caption text (ie "automatically generated on") */
  header + p {
    text-align: center;
    margin-top: 10px;
  }

  /* -------------------------------------------------- */
  /* text elements */
  /* -------------------------------------------------- */

  /* links */
  a {
    color: #2196f3;
    overflow-wrap: break-word;
  }

  /* superscripts and subscripts */
  sub,
  sup {
    /* prevent from affecting line height */
    line-height: 0;
  }

  /* unordered and ordered lists*/
  ul,
  ol {
    padding-left: 20px;
  }

  /* class for styling text semibold */
  .semibold {
    font-weight: 600;
  }

  /* class for styling elements horizontally left aligned */
  .left {
    display: block;
    text-align: left;
    margin-left: auto;
    margin-right: 0;
    justify-content: left;
  }

  /* class for styling elements horizontally centered */
  .center {
    display: block;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
  }

  /* class for styling elements horizontally right aligned */
  .right {
    display: block;
    text-align: right;
    margin-left: 0;
    margin-right: auto;
    justify-content: right;
  }

  /* -------------------------------------------------- */
  /* section elements */
  /* -------------------------------------------------- */

  /* horizontal divider line */
  hr {
    border: none;
    height: 1px;
    background: #bdbdbd;
  }

  /* paragraphs, horizontal dividers, figures, tables, code */
  p,
  hr,
  figure,
  table,
  pre {
    /* treat all as "paragraphs", with consistent vertical margins */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* figures */
  /* -------------------------------------------------- */

  /* figure */
  figure {
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure caption */
  figcaption {
    padding: 0;
    padding-top: 10px;
  }

  /* figure image element */
  figure > img,
  figure > svg {
    max-width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure auto-number */
  img + figcaption > span:first-of-type,
  svg + figcaption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* tables */
  /* -------------------------------------------------- */

  /* table */
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* table cells */
  th,
  td {
    border: solid 1px #bdbdbd;
    padding: 10px;
    /* squash table if too wide for page by forcing line breaks */
    overflow-wrap: break-word;
    word-break: break-word;
  }

  /* header row and even rows */
  th,
  tr:nth-child(2n) {
    background-color: #fafafa;
  }

  /* odd rows */
  tr:nth-child(2n + 1) {
    background-color: #ffffff;
  }

  /* table caption */
  caption {
    text-align: left;
    padding: 0;
    padding-bottom: 10px;
  }

  /* table auto-number */
  table > caption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* code */
  /* -------------------------------------------------- */

  /* multi-line code block */
  pre {
    padding: 10px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
    break-inside: avoid;
    text-align: left;
  }

  /* inline code, ie code within normal text */
  :not(pre) > code {
    padding: 0 4px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
  }

  /* code text */
  /* apply all children, to reach syntax highlighting sub-elements */
  code,
  code * {
    /* force monospace font */
    font-family: "Source Code Pro", "Courier New", monospace;
  }

  /* -------------------------------------------------- */
  /* quotes */
  /* -------------------------------------------------- */

  /* quoted text */
  blockquote {
    margin: 0;
    padding: 0;
    border-left: 4px solid #bdbdbd;
    padding-left: 16px;
    break-inside: avoid;
  }

  /* -------------------------------------------------- */
  /* banners */
  /* -------------------------------------------------- */

  /* info banners */
  .banner {
    box-sizing: border-box;
    display: block;
    position: relative;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px;
    text-align: center;
  }

  /* paragraph in banner */
  .banner > p {
    margin: 0;
  }

  /* -------------------------------------------------- */
  /* highlight colors */
  /* -------------------------------------------------- */

  .white {
    background: #ffffff;
  }
  .lightgrey {
    background: #eeeeee;
  }
  .grey {
    background: #757575;
  }
  .darkgrey {
    background: #424242;
  }
  .black {
    background: #000000;
  }
  .lightred {
    background: #ffcdd2;
  }
  .lightyellow {
    background: #ffecb3;
  }
  .lightgreen {
    background: #dcedc8;
  }
  .lightblue {
    background: #e3f2fd;
  }
  .lightpurple {
    background: #f3e5f5;
  }
  .red {
    background: #f44336;
  }
  .orange {
    background: #ff9800;
  }
  .yellow {
    background: #ffeb3b;
  }
  .green {
    background: #4caf50;
  }
  .blue {
    background: #2196f3;
  }
  .purple {
    background: #9c27b0;
  }
  .white,
  .lightgrey,
  .lightred,
  .lightyellow,
  .lightgreen,
  .lightblue,
  .lightpurple,
  .orange,
  .yellow,
  .white a,
  .lightgrey a,
  .lightred a,
  .lightyellow a,
  .lightgreen a,
  .lightblue a,
  .lightpurple a,
  .orange a,
  .yellow a {
    color: #000000;
  }
  .grey,
  .darkgrey,
  .black,
  .red,
  .green,
  .blue,
  .purple,
  .grey a,
  .darkgrey a,
  .black a,
  .red a,
  .green a,
  .blue a,
  .purple a {
    color: #ffffff;
  }

  /* -------------------------------------------------- */
  /* buttons */
  /* -------------------------------------------------- */

  /* class for styling links like buttons */
  .button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 5px;
    padding: 10px 20px;
    font-size: 0.75em;
    font-weight: 600;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 1px;
    background: none;
    color: #2196f3;
    border: solid 1px #bdbdbd;
    border-radius: 5px;
  }

  /* buttons when hovered */
  .button:hover:not([disabled]),
  .icon_button:hover:not([disabled]) {
    cursor: pointer;
    background: #f5f5f5;
  }

  /* buttons when disabled */
  .button[disabled],
  .icon_button[disabled] {
    opacity: 0.35;
    pointer-events: none;
  }

  /* class for styling buttons containg only single icon */
  .icon_button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0;
    padding: 0;
    background: none;
    border-radius: 5px;
    border: none;
    width: 20px;
    height: 20px;
    min-width: 20px;
    min-height: 20px;
  }

  /* icon button inner svg image */
  .icon_button > svg {
    height: 16px;
  }

  /* -------------------------------------------------- */
  /* icons */
  /* -------------------------------------------------- */

  /* class for styling icons inline with text */
  .inline_icon {
    height: 1em;
    position: relative;
    top: 0.125em;
  }

  /* -------------------------------------------------- */
  /* references */
  /* -------------------------------------------------- */

  .csl-entry {
    margin-top: 15px;
    margin-bottom: 15px;
  }

  /* -------------------------------------------------- */
  /* print control */
  /* -------------------------------------------------- */

  @media print {
    @page {
      /* suggested printing margin */
      margin: 0.5in;
    }

    /* document and "page" elements */
    html,
    body {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
    }

    /* "page" element */
    body {
      font-size: 11pt !important;
      line-height: 1.35;
    }

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: 15px 0;
    }

    /* figures and tables */
    figure,
    table {
      font-size: 0.85em;
    }

    /* table cells */
    th,
    td {
      padding: 5px;
    }

    /* shrink font awesome icons */
    i.fas,
    i.fab,
    i.far,
    i.fal {
      transform: scale(0.85);
    }

    /* decrease banner margins */
    .banner {
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 15px;
    }

    /* class for centering an element vertically on its own page */
    .page_center {
      margin: auto;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      vertical-align: middle;
      break-before: page;
      break-after: page;
    }

    /* always insert a page break before the element */
    .page_break_before {
      break-before: page;
    }

    /* always insert a page break after the element */
    .page_break_after {
      break-after: page;
    }

    /* avoid page break before the element */
    .page_break_before_avoid {
      break-before: avoid;
    }

    /* avoid page break after the element */
    .page_break_after_avoid {
      break-after: avoid;
    }

    /* avoid page break inside the element */
    .page_break_inside_avoid {
      break-inside: avoid;
    }
  }

  /* -------------------------------------------------- */
  /* override pandoc css quirks */
  /* -------------------------------------------------- */

  .sourceCode {
    /* prevent unsightly overflow in wide code blocks */
    overflow: auto !important;
  }

  div.sourceCode {
    /* prevent background fill on top-most code block  container */
    background: none !important;
  }

  .sourceCode * {
    /* force consistent line spacing */
    line-height: 1.5 !important;
  }

  div.sourceCode {
    /* style code block margins same as <pre> element */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* tablenos */
  /* -------------------------------------------------- */

  /* tablenos wrapper */
  .tablenos {
    width: 100%;
    margin: 20px 0;
  }

  .tablenos > table {
    /* move margins from table to table_wrapper to allow margin collapsing */
    margin: 0;
  }

  @media only screen {
    /* tablenos wrapper */
    .tablenos {
      /* show scrollbar on tables if necessary to prevent overflow */
      overflow-x: auto !important;
    }

    .tablenos th,
    .tablenos td {
      overflow-wrap: unset !important;
      word-break: unset !important;
    }

    /* table in wrapper */
    .tablenos table,
    .tablenos table * {
      /* don't break table words */
      overflow-wrap: normal !important;
    }
  }
</style>
<!-- 
    Plugin Core

    Functions needed for and shared across all first-party plugins.
-->

<script>
  // get element that is target of hash (from link element or url)
  function getHashTarget(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector(`[id="${id}"]`);
    if (!target) return;

    // if figure or table, modify target to get expected element
    if (id.indexOf("fig:") === 0) target = target.querySelector("figure");
    if (id.indexOf("tbl:") === 0) target = target.querySelector("table");

    return target;
  }

  // get position/dimensions of element or viewport
  function getRectInView(element) {
    let rect = {};
    rect.left = 0;
    rect.top = 0;
    rect.right = document.documentElement.clientWidth;
    rect.bottom = document.documentElement.clientHeight;
    let style = {};

    if (element instanceof HTMLElement) {
      rect = element.getBoundingClientRect();
      style = window.getComputedStyle(element);
    }

    const margin = {};
    margin.left = parseFloat(style.marginLeftWidth) || 0;
    margin.top = parseFloat(style.marginTopWidth) || 0;
    margin.right = parseFloat(style.marginRightWidth) || 0;
    margin.bottom = parseFloat(style.marginBottomWidth) || 0;

    const border = {};
    border.left = parseFloat(style.borderLeftWidth) || 0;
    border.top = parseFloat(style.borderTopWidth) || 0;
    border.right = parseFloat(style.borderRightWidth) || 0;
    border.bottom = parseFloat(style.borderBottomWidth) || 0;

    const newRect = {};
    newRect.left = rect.left + margin.left + border.left;
    newRect.top = rect.top + margin.top + border.top;
    newRect.right = rect.right + margin.right + border.right;
    newRect.bottom = rect.bottom + margin.bottom + border.bottom;
    newRect.width = newRect.right - newRect.left;
    newRect.height = newRect.bottom - newRect.top;

    return newRect;
  }

  // get position of element relative to page
  function getRectInPage(element) {
    const rect = getRectInView(element);
    const body = getRectInView(document.body);

    const newRect = {};
    newRect.left = rect.left - body.left;
    newRect.top = rect.top - body.top;
    newRect.right = rect.right - body.left;
    newRect.bottom = rect.bottom - body.top;
    newRect.width = rect.width;
    newRect.height = rect.height;

    return newRect;
  }

  // get closest element before specified element that matches query
  function firstBefore(element, query) {
    while (element && element !== document.body && !element.matches(query))
      element = element.previousElementSibling || element.parentNode;

    return element;
  }

  // check if element is part of collapsed heading
  function isCollapsed(element) {
    while (element && element !== document.body) {
      if (element.dataset.collapsed === "true") return true;
      element = element.parentNode;
    }
    return false;
  }

  // expand any collapsed parent containers of element if necessary
  function expandElement(element) {
    if (isCollapsed(element)) {
      // accordion plugin
      const heading = firstBefore(element, "h2");
      if (heading) heading.click();
      // details/summary HTML element
      const summary = firstBefore(element, "summary");
      if (summary) summary.click();
    }
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);

    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // get list of elements after a start element up to element matching query
  function nextUntil(element, query, exclude) {
    const elements = [];
    while (((element = element.nextElementSibling), element)) {
      if (element.matches(query)) break;
      if (!element.matches(exclude)) elements.push(element);
    }
    return elements;
  }
</script>
<!--
  Accordion Plugin

  Allows sections of content under h2 headings to be collapsible.
-->

<script type="module">
  // whether to always start expanded ('false'), always start collapsed
  // ('true'), or start collapsed when screen small ('auto')
  const startCollapsed = "auto";

  // start script
  function start() {
    // run through each <h2> heading
    const headings = document.querySelectorAll("h2");
    for (const heading of headings) {
      addArrow(heading);

      // start expanded/collapsed based on option
      if (
        startCollapsed === "true" ||
        (startCollapsed === "auto" && isSmallScreen()) ||
        heading.dataset.collapsed === "true"
      )
        collapseHeading(heading);
      else expandElement(heading);
    }

    // attach hash change listener to window
    window.addEventListener("hashchange", onHashChange);
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) goToElement(target);
  }

  // add arrow to heading
  function addArrow(heading) {
    // add arrow button
    const arrow = document.createElement("button");
    arrow.innerHTML = document.querySelector(".icon_angle_down").innerHTML;
    arrow.classList.add("icon_button", "accordion_arrow");
    heading.insertBefore(arrow, heading.firstChild);

    // attach click listener to heading and button
    heading.addEventListener("click", onHeadingClick);
    arrow.addEventListener("click", onArrowClick);
  }

  // determine if on mobile-like device with small screen
  function isSmallScreen() {
    return Math.min(window.innerWidth, window.innerHeight) < 480;
  }

  // when <h2> heading is clicked
  function onHeadingClick(event) {
    // only collapse if <h2> itself is target of click (eg, user did
    // not click on anchor within <h2>)
    if (event.target === this) toggleCollapse(this);
  }

  // when arrow button is clicked
  function onArrowClick() {
    toggleCollapse(this.parentNode);
  }

  // collapse section if expanded, expand if collapsed
  function toggleCollapse(heading) {
    if (heading.dataset.collapsed === "false") collapseHeading(heading);
    else expandElement(heading);
  }

  // elements to exclude from collapse, such as table of contents panel,
  // hypothesis panel, etc
  const exclude = "#toc_panel, div.annotator-frame, #lightbox_overlay";

  // collapse section
  function collapseHeading(heading) {
    heading.setAttribute("data-collapsed", "true");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "true");
  }

  // expand section
  function expandElement(heading) {
    heading.setAttribute("data-collapsed", "false");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "false");
  }

  // get list of elements between this <h2> and next <h2> or <h1>
  // ("children" of the <h2> section)
  function getChildren(heading) {
    return nextUntil(heading, "h2, h1", exclude);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
  <!-- modified from: https://fontawesome.com/icons/angle-down -->
  <svg width="16" height="16" viewBox="0 0 448 512">
    <path
      fill="currentColor"
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* accordion arrow button */
    .accordion_arrow {
      margin-right: 10px;
    }

    /* arrow icon when <h2> data-collapsed attribute true */
    h2[data-collapsed="true"] > .accordion_arrow > svg {
      transform: rotate(-90deg);
    }

    /* all elements (except <h2>'s) when data-collapsed attribute true */
    *:not(h2)[data-collapsed="true"] {
      display: none;
    }

    /* accordion arrow button when hovered and <h2>'s when hovered */
    .accordion_arrow:hover,
    h2[data-collapsed="true"]:hover,
    h2[data-collapsed="false"]:hover {
      cursor: pointer;
    }
  }

  /* always hide accordion arrow button on print */
  @media only print {
    .accordion_arrow {
      display: none;
    }
  }
</style>
<!--
  Anchors Plugin

  Adds an anchor next to each of a certain type of element that provides a
  human-readable url to that specific item/position in the document (e.g.
  "manuscript.html#abstract"). It also makes it such that scrolling out of view
  of a target removes its identifier from the url.
-->

<script type="module">
  // which types of elements to add anchors next to, in "document.querySelector"
  // format
  const typesQuery =
    'h1, h2, h3, div[id^="fig:"], div[id^="tbl:"], span[id^="eq:"]';

  // start script
  function start() {
    // add anchor to each element of specified types
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) addAnchor(element);

    // attach scroll listener to window
    window.addEventListener("scroll", onScroll);
  }

  // when window is scrolled
  function onScroll() {
    // if url has hash and user has scrolled out of view of hash
    // target, remove hash from url
    const tolerance = 100;
    const target = getHashTarget();
    if (target) {
      if (
        target.getBoundingClientRect().top > window.innerHeight + tolerance ||
        target.getBoundingClientRect().bottom < 0 - tolerance
      )
        history.pushState(null, null, " ");
    }
  }

  // add anchor to element
  function addAnchor(element) {
    let addTo; // element to add anchor button to

    // if figure or table, modify withId and addTo to get expected
    // elements
    if (element.id.indexOf("fig:") === 0) {
      addTo = element.querySelector("figcaption");
    } else if (element.id.indexOf("tbl:") === 0) {
      addTo = element.querySelector("caption");
    } else if (element.id.indexOf("eq:") === 0) {
      addTo = element.querySelector(".eqnos-number");
    }

    addTo = addTo || element;
    const id = element.id || null;

    // do not add anchor if element doesn't have assigned id.
    // id is generated by pandoc and is assumed to be unique and
    // human-readable
    if (!id) return;

    // create anchor button
    const anchor = document.createElement("a");
    anchor.innerHTML = document.querySelector(".icon_link").innerHTML;
    anchor.title = "Link to this part of the document";
    anchor.classList.add("icon_button", "anchor");
    anchor.dataset.ignore = "true";
    anchor.href = "#" + id;
    addTo.appendChild(anchor);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- link icon -->

<template class="icon_link">
  <!-- modified from: https://fontawesome.com/icons/link -->
  <svg width="16" height="16" viewBox="0 0 512 512">
    <path
      fill="currentColor"
      d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* anchor button */
    .anchor {
      opacity: 0;
      margin-left: 5px;
    }

    /* anchor buttons within <h2>'s */
    h2 .anchor {
      margin-left: 10px;
    }

    /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
    *:hover > .anchor,
    .anchor:hover,
    .anchor:focus {
      opacity: 1;
    }

    /* anchor button when hovered */
    .anchor:hover {
      cursor: pointer;
    }
  }

  /* always show anchor button on devices with no mouse/hover ability */
  @media (hover: none) {
    .anchor {
      opacity: 1;
    }
  }

  /* always hide anchor button on print */
  @media only print {
    .anchor {
      display: none;
    }
  }
</style>
<!-- 
    Attributes Plugin

    Allows arbitrary HTML attributes to be attached to (almost) any element.
    Place an HTML comment inside or next to the desired element with the content:
    $attribute="value"
-->

<script type="module">
  // start script
  function start() {
    // get list of comments in document
    const comments = findComments();

    for (const comment of comments)
      if (comment.parentElement)
        addAttributes(comment.parentElement, comment.nodeValue.trim());
  }

  // add html attributes to specified element based on string of
  // html attributes and values
  function addAttributes(element, text) {
    // regex's for finding attribute/value pairs in the format of
    // attribute="value" or attribute='value
    const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
    const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

    // loop through attribute/value pairs
    let match;
    while ((match = text.match(regex2) || text.match(regex1))) {
      // get attribute and value from regex capture groups
      let attribute = match[1];
      let value = match[2];

      // remove from string
      text = text.substring(match.index + match[0].length);

      if (!attribute || !value) break;

      // set attribute of parent element
      try {
        element.setAttribute(attribute, value);
      } catch (error) {
        console.log(error);
      }

      // special case for colspan
      if (attribute === "colspan") removeTableCells(element, value);
    }
  }

  // get list of comment elements in document
  function findComments() {
    const comments = [];

    // iterate over comment nodes in document
    function acceptNode(node) {
      return NodeFilter.FILTER_ACCEPT;
    }
    const iterator = document.createNodeIterator(
      document.body,
      NodeFilter.SHOW_COMMENT,
      acceptNode
    );
    let node;
    while ((node = iterator.nextNode())) comments.push(node);

    return comments;
  }

  // remove certain number of cells after specified cell
  function removeTableCells(cell, number) {
    number = parseInt(number);
    if (!number) return;

    // remove elements
    for (; number > 1; number--) {
      if (cell.nextElementSibling) cell.nextElementSibling.remove();
    }
  }

  // start script on DOMContentLoaded instead of load to ensure this plugins
  // runs before other plugins
  window.addEventListener("DOMContentLoaded", start);
</script>
<!--
  Jump to First Plugin

  Adds a button next to each reference entry, figure, and table that jumps the
  page to the first occurrence of a link to that item in the manuscript.
-->

<script type="module">
  // whether to add buttons next to reference entries
  const references = "true";
  // whether to add buttons next to figures
  const figures = "true";
  // whether to add buttons next to tables
  const tables = "true";

  // start script
  function start() {
    if (references !== "false")
      makeButtons(`div[id^="ref-"]`, ".csl-left-margin", "reference");
    if (figures !== "false")
      makeButtons(`div[id^="fig:"]`, "figcaption", "figure");
    if (tables !== "false") makeButtons(`div[id^="tbl:"]`, "caption", "table");
  }

  // when jump button clicked
  function onButtonClick() {
    const first = getFirstOccurrence(this.dataset.id);
    if (!first) return;

    // update url hash so navigating "back" in history will return user to button
    window.location.hash = this.dataset.id;
    // scroll to link
    const timeout = function () {
      goToElement(first, window.innerHeight * 0.5);
    };
    window.setTimeout(timeout, 0);
  }

  // get first occurrence of link to item in document
  function getFirstOccurrence(id) {
    let query = "a";
    query += '[href="#' + id + '"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelector(query);
  }

  // add button next to each reference entry, figure, or table
  function makeButtons(query, containerQuery, subject) {
    const elements = document.querySelectorAll(query);
    for (const element of elements) {
      const id = element.id;
      const buttonContainer = element.querySelector(containerQuery);
      const first = getFirstOccurrence(id);

      // if can't find link to reference or place to put button, ignore
      if (!first || !buttonContainer) continue;

      // make jump button
      let button = document.createElement("button");
      button.classList.add("icon_button", "jump_arrow");
      button.title = `Jump to the first occurrence of this ${subject} in the document`;
      const icon = document.querySelector(".icon_angle_double_up");
      button.innerHTML = icon.innerHTML;
      button.dataset.id = id;
      button.dataset.ignore = "true";
      button.addEventListener("click", onButtonClick);
      buttonContainer.prepend(button);
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
  <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
  <svg width="16" height="16" viewBox="0 0 320 512">
    <path
      fill="currentColor"
      d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* jump button */
    .jump_arrow {
      position: relative;
      top: 0.125em;
      margin-right: 5px;
    }
  }

  /* always hide jump button on print */
  @media only print {
    .jump_arrow {
      display: none;
    }
  }
</style>
<!-- 
    Lightbox Plugin

    Makes it such that when a user clicks on an image, the image fills the
    screen and the user can pan/drag/zoom the image and navigate between other
    images in the document.
-->

<script type="module">
  // list of possible zoom/scale factors
  const zooms =
    "0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8";
  // whether to fit image to view ('fit'), display at 100% and shrink if
  // necessary ('shrink'), or always display at 100% ('100')
  const defaultZoom = "fit";
  // whether to zoom in/out toward center of view ('true') or mouse ('false')
  const centerZoom = "false";

  // start script
  function start() {
    // run through each <img> element
    const imgs = document.querySelectorAll("figure > img");
    let count = 1;
    for (const img of imgs) {
      img.classList.add("lightbox_document_img");
      img.dataset.number = count;
      img.dataset.total = imgs.length;
      img.addEventListener("click", openLightbox);
      count++;
    }

    // attach mouse and key listeners to window
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("keyup", onKeyUp);
  }

  // when mouse is moved anywhere in window
  function onWindowMouseMove(event) {
    window.mouseX = event.clientX;
    window.mouseY = event.clientY;
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("lightbox_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("lightbox_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeLightbox();
        break;
    }
  }

  // open lightbox
  function openLightbox() {
    const lightbox = makeLightbox(this);
    if (!lightbox) return;

    blurBody(lightbox);
    document.body.appendChild(lightbox);
  }

  // make lightbox
  function makeLightbox(img) {
    // delete lightbox if it exists, start fresh
    closeLightbox();

    // create screen overlay containing lightbox
    const overlay = document.createElement("div");
    overlay.id = "lightbox_overlay";

    // create image info boxes
    const numberInfo = document.createElement("div");
    const zoomInfo = document.createElement("div");
    numberInfo.id = "lightbox_number_info";
    zoomInfo.id = "lightbox_zoom_info";

    // create container for image
    const imageContainer = document.createElement("div");
    imageContainer.id = "lightbox_image_container";
    const lightboxImg = makeLightboxImg(
      img,
      imageContainer,
      numberInfo,
      zoomInfo
    );
    imageContainer.appendChild(lightboxImg);

    // create bottom container for caption and navigation buttons
    const bottomContainer = document.createElement("div");
    bottomContainer.id = "lightbox_bottom_container";
    const caption = makeCaption(img);
    const prevButton = makePrevButton(img);
    const nextButton = makeNextButton(img);
    bottomContainer.appendChild(prevButton);
    bottomContainer.appendChild(caption);
    bottomContainer.appendChild(nextButton);

    // attach top middle and bottom to overlay
    overlay.appendChild(numberInfo);
    overlay.appendChild(zoomInfo);
    overlay.appendChild(imageContainer);
    overlay.appendChild(bottomContainer);

    return overlay;
  }

  // make <img> object that is intuitively draggable and zoomable
  function makeLightboxImg(sourceImg, container, numberInfoBox, zoomInfoBox) {
    // create copy of source <img>
    const img = sourceImg.cloneNode(true);
    img.classList.remove("lightbox_document_img");
    img.removeAttribute("id");
    img.removeAttribute("width");
    img.removeAttribute("height");
    img.style.position = "unset";
    img.style.margin = "0";
    img.style.padding = "0";
    img.style.width = "";
    img.style.height = "";
    img.style.minWidth = "";
    img.style.minHeight = "";
    img.style.maxWidth = "";
    img.style.maxHeight = "";
    img.id = "lightbox_img";

    // build sorted list of zoomSteps
    const zoomSteps = zooms.split(/[^0-9.]/).map((step) => parseFloat(step));
    zoomSteps.sort((a, b) => a - b);

    // <img> object property variables
    let zoom = 1;
    let translateX = 0;
    let translateY = 0;
    let clickMouseX = undefined;
    let clickMouseY = undefined;
    let clickTranslateX = undefined;
    let clickTranslateY = undefined;

    updateNumberInfo();

    // update image numbers displayed in info box
    function updateNumberInfo() {
      numberInfoBox.innerHTML =
        sourceImg.dataset.number + " of " + sourceImg.dataset.total;
    }

    // update zoom displayed in info box
    function updateZoomInfo() {
      let zoomInfo = zoom * 100;
      if (!Number.isInteger(zoomInfo)) zoomInfo = zoomInfo.toFixed(2);
      zoomInfoBox.innerHTML = zoomInfo + "%";
    }

    // move to closest zoom step above current zoom
    const zoomIn = function () {
      for (const zoomStep of zoomSteps) {
        if (zoomStep > zoom) {
          zoom = zoomStep;
          break;
        }
      }
      updateTransform();
    };

    // move to closest zoom step above current zoom
    const zoomOut = function () {
      zoomSteps.reverse();
      for (const zoomStep of zoomSteps) {
        if (zoomStep < zoom) {
          zoom = zoomStep;
          break;
        }
      }
      zoomSteps.reverse();

      updateTransform();
    };

    // update display of <img> based on scale/translate properties
    const updateTransform = function () {
      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      // get new width/height after scale
      const rect = img.getBoundingClientRect();
      // limit translate
      translateX = Math.max(translateX, -rect.width / 2);
      translateX = Math.min(translateX, rect.width / 2);
      translateY = Math.max(translateY, -rect.height / 2);
      translateY = Math.min(translateY, rect.height / 2);

      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      updateZoomInfo();
    };

    // fit <img> to container
    const fit = function () {
      // no x/y offset, 100% zoom by default
      translateX = 0;
      translateY = 0;
      zoom = 1;

      // widths of <img> and container
      const imgWidth = img.naturalWidth;
      const imgHeight = img.naturalHeight;
      const containerWidth = parseFloat(
        window.getComputedStyle(container).width
      );
      const containerHeight = parseFloat(
        window.getComputedStyle(container).height
      );

      // how much zooming is needed to fit <img> to container
      const xRatio = imgWidth / containerWidth;
      const yRatio = imgHeight / containerHeight;
      const maxRatio = Math.max(xRatio, yRatio);
      const newZoom = 1 / maxRatio;

      // fit <img> to container according to option
      if (defaultZoom === "shrink") {
        if (maxRatio > 1) zoom = newZoom;
      } else if (defaultZoom === "fit") zoom = newZoom;

      updateTransform();
    };

    // when mouse wheel is rolled anywhere in container
    const onContainerWheel = function (event) {
      if (!event) return;

      // let ctrl + mouse wheel to zoom behave as normal
      if (event.ctrlKey) return;

      // prevent normal scroll behavior
      event.preventDefault();
      event.stopPropagation();

      // point around which to scale img
      const viewRect = container.getBoundingClientRect();
      const viewX = (viewRect.left + viewRect.right) / 2;
      const viewY = (viewRect.top + viewRect.bottom) / 2;
      const originX = centerZoom === "true" ? viewX : mouseX;
      const originY = centerZoom === "true" ? viewY : mouseY;

      // get point on image under origin
      const oldRect = img.getBoundingClientRect();
      const oldPercentX = (originX - oldRect.left) / oldRect.width;
      const oldPercentY = (originY - oldRect.top) / oldRect.height;

      // increment/decrement zoom
      if (event.deltaY < 0) zoomIn();
      if (event.deltaY > 0) zoomOut();

      // get offset between previous image point and origin
      const newRect = img.getBoundingClientRect();
      const offsetX = originX - (newRect.left + newRect.width * oldPercentX);
      const offsetY = originY - (newRect.top + newRect.height * oldPercentY);

      // translate image to keep image point under origin
      translateX += offsetX;
      translateY += offsetY;

      // perform translate
      updateTransform();
    };

    // when container is clicked
    function onContainerClick(event) {
      // if container itself is target of click, and not other
      // element above it
      if (event.target === this) closeLightbox();
    }

    // when mouse button is pressed on image
    const onImageMouseDown = function (event) {
      // store original mouse position relative to image
      clickMouseX = window.mouseX;
      clickMouseY = window.mouseY;
      clickTranslateX = translateX;
      clickTranslateY = translateY;
      event.stopPropagation();
      event.preventDefault();
    };

    // when mouse button is released anywhere in window
    const onWindowMouseUp = function (event) {
      // reset original mouse position
      clickMouseX = undefined;
      clickMouseY = undefined;
      clickTranslateX = undefined;
      clickTranslateY = undefined;

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mouseup", onWindowMouseUp);
    };

    // when mouse is moved anywhere in window
    const onWindowMouseMove = function (event) {
      if (
        clickMouseX === undefined ||
        clickMouseY === undefined ||
        clickTranslateX === undefined ||
        clickTranslateY === undefined
      )
        return;

      // offset image based on original and current mouse position
      translateX = clickTranslateX + window.mouseX - clickMouseX;
      translateY = clickTranslateY + window.mouseY - clickMouseY;
      updateTransform();
      event.preventDefault();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mousemove", onWindowMouseMove);
    };

    // when window is resized
    const onWindowResize = function (event) {
      fit();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("resize", onWindowResize);
    };

    // attach the necessary event listeners
    img.addEventListener("dblclick", fit);
    img.addEventListener("mousedown", onImageMouseDown);
    container.addEventListener("wheel", onContainerWheel);
    container.addEventListener("mousedown", onContainerClick);
    container.addEventListener("touchstart", onContainerClick);
    window.addEventListener("mouseup", onWindowMouseUp);
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("resize", onWindowResize);

    // run fit() after lightbox atttached to document and <img> Loaded
    // so needed container and img dimensions available
    img.addEventListener("load", fit);

    return img;
  }

  // make caption
  function makeCaption(img) {
    const caption = document.createElement("div");
    caption.id = "lightbox_caption";
    const captionSource = img.nextElementSibling;
    if (captionSource.tagName.toLowerCase() === "figcaption") {
      const captionCopy = makeCopy(captionSource);
      caption.innerHTML = captionCopy.innerHTML;
    }

    caption.addEventListener("touchstart", function (event) {
      event.stopPropagation();
    });

    return caption;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // make button to jump to previous image in document
  function makePrevButton(img) {
    const prevButton = document.createElement("button");
    prevButton.id = "lightbox_prev_button";
    prevButton.title = "Jump to the previous image in the document [←]";
    prevButton.classList.add("icon_button", "lightbox_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;

    // attach click listeners to button
    prevButton.addEventListener("click", function () {
      getPrevImg(img).click();
    });

    return prevButton;
  }

  // make button to jump to next image in document
  function makeNextButton(img) {
    const nextButton = document.createElement("button");
    nextButton.id = "lightbox_next_button";
    nextButton.title = "Jump to the next image in the document [→]";
    nextButton.classList.add("icon_button", "lightbox_button");
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;

    // attach click listeners to button
    nextButton.addEventListener("click", function () {
      getNextImg(img).click();
    });

    return nextButton;
  }

  // get previous image in document
  function getPrevImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if < 1
    if (index - 1 >= 0) index--;
    else index = imgs.length - 1;
    return imgs[index];
  }

  // get next image in document
  function getNextImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if > total
    if (index + 1 <= imgs.length - 1) index++;
    else index = 0;
    return imgs[index];
  }

  // close lightbox
  function closeLightbox() {
    focusBody();

    const lightbox = document.getElementById("lightbox_overlay");
    if (lightbox) lightbox.remove();
  }

  // make all elements behind lightbox non-focusable
  function blurBody(overlay) {
    const all = document.querySelectorAll("*");
    for (const element of all) element.tabIndex = -1;
    document.body.classList.add("body_no_scroll");
  }

  // make all elements focusable again
  function focusBody() {
    const all = document.querySelectorAll("*");
    for (const element of all) element.removeAttribute("tabIndex");
    document.body.classList.remove("body_no_scroll");
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* regular <img> in document when hovered */
    img.lightbox_document_img:hover {
      cursor: pointer;
    }

    .body_no_scroll {
      overflow: hidden !important;
    }

    /* screen overlay */
    #lightbox_overlay {
      display: flex;
      flex-direction: column;
      position: fixed;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.75);
      z-index: 3;
    }

    /* middle area containing lightbox image */
    #lightbox_image_container {
      flex-grow: 1;
      display: flex;
      justify-content: center;
      align-items: center;
      overflow: hidden;
      position: relative;
      padding: 20px;
    }

    /* bottom area containing caption */
    #lightbox_bottom_container {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100px;
      min-height: 100px;
      max-height: 100px;
      background: rgba(0, 0, 0, 0.5);
    }

    /* image number info text box */
    #lightbox_number_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      left: 2px;
      top: 0;
      z-index: 4;
    }

    /* zoom info text box */
    #lightbox_zoom_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      right: 2px;
      top: 0;
      z-index: 4;
    }

    /* copy of image caption */
    #lightbox_caption {
      box-sizing: border-box;
      display: inline-block;
      width: 100%;
      max-height: 100%;
      padding: 10px 0;
      text-align: center;
      overflow-y: auto;
      color: #ffffff;
    }

    /* navigation previous/next button */
    .lightbox_button {
      width: 100px;
      height: 100%;
      min-width: 100px;
      min-height: 100%;
      color: #ffffff;
    }

    /* navigation previous/next button when hovered */
    .lightbox_button:hover {
      background: none !important;
    }

    /* navigation button icon */
    .lightbox_button > svg {
      height: 25px;
    }

    /* figure auto-number */
    #lightbox_caption > span:first-of-type {
      font-weight: bold;
      margin-right: 5px;
    }

    /* lightbox image when hovered */
    #lightbox_img:hover {
      cursor: grab;
    }

    /* lightbox image when grabbed */
    #lightbox_img:active {
      cursor: grabbing;
    }
  }

  /* when on screen < 480px wide */
  @media only screen and (max-width: 480px) {
    /* make navigation buttons skinnier on small screens to make more room for caption text */
    .lightbox_button {
      width: 50px;
      min-width: 50px;
    }
  }

  /* always hide lightbox on print */
  @media only print {
    #lightbox_overlay {
      display: none;
    }
  }
</style>
<!-- 
  Link Highlight Plugin

  Makes it such that when a user hovers or focuses a link, other links that have
  the same target will be highlighted. It also makes it such that when clicking
  a link, the target of the link (eg reference, figure, table) is briefly
  highlighted.
-->

<script type="module">
  // whether to also highlight links that go to external urls
  const externalLinks = "false";
  // whether user must click off to unhighlight instead of just
  // un-hovering
  const clickUnhighlight = "false";
  // whether to also highlight links that are unique
  const highlightUnique = "true";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach mouse and focus listeners to link
      link.addEventListener("mouseenter", onLinkFocus);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("mouseleave", onLinkUnhover);
    }

    // attach click and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("hashchange", onHashChange);

    // run hash change on window load in case user has navigated
    // directly to hash
    onHashChange();
  }

  // when link is focused (tabbed to) or hovered
  function onLinkFocus() {
    highlight(this);
  }

  // when link is unhovered
  function onLinkUnhover() {
    if (clickUnhighlight !== "true") unhighlightAll();
  }

  // when the mouse is clicked anywhere in window
  function onClick(event) {
    unhighlightAll();
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) glowElement(target);
  }

  // start glow sequence on an element
  function glowElement(element) {
    const startGlow = function () {
      onGlowEnd();
      element.dataset.glow = "true";
      element.addEventListener("animationend", onGlowEnd);
    };
    const onGlowEnd = function () {
      element.removeAttribute("data-glow");
      element.removeEventListener("animationend", onGlowEnd);
    };
    startGlow();
  }

  // highlight link and all others with same target
  function highlight(link) {
    // force unhighlight all to start fresh
    unhighlightAll();

    // get links with same target
    if (!link) return;
    const sameLinks = getSameLinks(link);

    // if link unique and option is off, exit and don't highlight
    if (sameLinks.length <= 1 && highlightUnique !== "true") return;

    // highlight all same links, and "select" (special highlight) this
    // one
    for (const sameLink of sameLinks) {
      if (sameLink === link) sameLink.setAttribute("data-selected", "true");
      else sameLink.setAttribute("data-highlighted", "true");
    }
  }

  // unhighlight all links
  function unhighlightAll() {
    const links = getLinks();
    for (const link of links) {
      link.setAttribute("data-selected", "false");
      link.setAttribute("data-highlighted", "false");
    }
  }

  // get links with same target
  function getSameLinks(link) {
    const results = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        results.push(otherLink);
    }
    return results;
  }

  // get all links of types we wish to handle
  function getLinks() {
    let query = "a";
    if (externalLinks !== "true") query += '[href^="#"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelectorAll(query);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  @media only screen {
    /* anything with data-highlighted attribute true */
    [data-highlighted="true"] {
      background: #ffeb3b;
    }

    /* anything with data-selected attribute true */
    [data-selected="true"] {
      background: #ff8a65 !important;
    }

    /* animation definition for glow */
    @keyframes highlight_glow {
      0% {
        background: none;
      }
      10% {
        background: #bbdefb;
      }
      100% {
        background: none;
      }
    }

    /* anything with data-glow attribute true */
    [data-glow="true"] {
      animation: highlight_glow 2s;
    }
  }
</style>
<!--
  Table of Contents Plugin

  Provides a "table of contents" (toc) panel on the side of the document that
  allows the user to conveniently navigate between sections of the document.
-->

<script type="module">
  // which types of elements to add links for, in "document.querySelector" format
  const typesQuery = "h1, h2, h3";
  // whether toc starts open. use 'true' or 'false', or 'auto' to
  // use 'true' behavior when screen wide enough and 'false' when not
  const startOpen = "false";
  // whether toc closes when clicking on toc link. use 'true' or
  // 'false', or 'auto' to use 'false' behavior when screen wide
  // enough and 'true' when not
  const clickClose = "auto";
  // if list item is more than this many characters, text will be
  // truncated
  const charLimit = "50";
  // whether or not to show bullets next to each toc item
  const bullets = "false";

  // start script
  function start() {
    // make toc panel and populate with entries (links to document
    // sections)
    const panel = makePanel();
    if (!panel) return;
    makeEntries(panel);
    // attach panel to document after making entries, so 'toc' heading
    // in panel isn't included in toc
    document.body.insertBefore(panel, document.body.firstChild);

    // initial panel state
    if (startOpen === "true" || (startOpen === "auto" && !isSmallScreen()))
      openPanel();
    else closePanel();

    // attach click, scroll, and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("scroll", onScroll);
    window.addEventListener("hashchange", onScroll);
    window.addEventListener("keyup", onKeyUp);
    onScroll();

    // add class to push document body down out of way of toc button
    document.body.classList.add("toc_body_nudge");
  }

  // determine if screen wide enough to fit toc panel
  function isSmallScreen() {
    // in default theme:
    // 816px = 8.5in = width of "page" (<body>) element
    // 260px = min width of toc panel (*2 for both sides of <body>)
    return window.innerWidth < 816 + 260 * 2;
  }

  // when mouse is clicked anywhere in window
  function onClick() {
    if (isSmallScreen()) closePanel();
  }

  // when window is scrolled or hash changed
  function onScroll() {
    highlightViewed();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    // close on esc
    if (event.key === "Escape") closePanel();
  }

  // find entry of currently viewed document section in toc and highlight
  function highlightViewed() {
    const firstId = getFirstInView(typesQuery);

    // get toc entries (links), unhighlight all, then highlight viewed
    const list = document.getElementById("toc_list");
    if (!firstId || !list) return;
    const links = list.querySelectorAll("a");
    for (const link of links) link.dataset.viewing = "false";
    const link = list.querySelector('a[href="#' + firstId + '"]');
    if (!link) return;
    link.dataset.viewing = "true";
  }

  // get first or previous toc listed element in top half of view
  function getFirstInView(query) {
    // get all elements matching query and with id
    const elements = document.querySelectorAll(query);
    const elementsWithIds = [];
    for (const element of elements) {
      if (element.id) elementsWithIds.push(element);
    }

    // get first or previous element in top half of view
    for (let i = 0; i < elementsWithIds.length; i++) {
      const element = elementsWithIds[i];
      const prevElement = elementsWithIds[Math.max(0, i - 1)];
      if (element.getBoundingClientRect().top >= 0) {
        if (element.getBoundingClientRect().top < window.innerHeight / 2)
          return element.id;
        else return prevElement.id;
      }
    }
  }

  // make panel
  function makePanel() {
    // create panel
    const panel = document.createElement("div");
    panel.id = "toc_panel";
    if (bullets === "true") panel.dataset.bullets = "true";

    // create header
    const header = document.createElement("div");
    header.id = "toc_header";

    // create toc button
    const button = document.createElement("button");
    button.id = "toc_button";
    button.innerHTML = document.querySelector(".icon_th_list").innerHTML;
    button.title = "Table of Contents";
    button.classList.add("icon_button");

    // create header text
    const text = document.createElement("h4");
    text.innerHTML = "Table of Contents";

    // create container for toc list
    const list = document.createElement("div");
    list.id = "toc_list";

    // attach click listeners
    panel.addEventListener("click", onPanelClick);
    header.addEventListener("click", onHeaderClick);
    button.addEventListener("click", onButtonClick);

    // attach elements
    header.appendChild(button);
    header.appendChild(text);
    panel.appendChild(header);
    panel.appendChild(list);

    return panel;
  }

  // create toc entries (links) to each element of the specified types
  function makeEntries(panel) {
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) {
      // do not add link if element doesn't have assigned id
      if (!element.id) continue;

      // create link/list item
      const link = document.createElement("a");
      link.classList.add("toc_link");
      switch (element.tagName.toLowerCase()) {
        case "h1":
          link.dataset.level = "1";
          break;
        case "h2":
          link.dataset.level = "2";
          break;
        case "h3":
          link.dataset.level = "3";
          break;
        case "h4":
          link.dataset.level = "4";
          break;
      }
      link.title = element.innerText;
      let text = element.innerText;
      if (text.length > charLimit) text = text.slice(0, charLimit) + "...";
      link.innerHTML = text;
      link.href = "#" + element.id;
      link.addEventListener("click", onLinkClick);

      // attach link
      panel.querySelector("#toc_list").appendChild(link);
    }
  }

  // when panel is clicked
  function onPanelClick(event) {
    // stop click from propagating to window/document and closing panel
    event.stopPropagation();
  }

  // when header itself is clicked
  function onHeaderClick(event) {
    togglePanel();
  }

  // when button is clicked
  function onButtonClick(event) {
    togglePanel();
    // stop header underneath button from also being clicked
    event.stopPropagation();
  }

  // when link is clicked
  function onLinkClick(event) {
    if (clickClose === "true" || (clickClose === "auto" && isSmallScreen()))
      closePanel();
    else openPanel();
  }

  // open panel if closed, close if opened
  function togglePanel() {
    const panel = document.getElementById("toc_panel");
    if (!panel) return;

    if (panel.dataset.open === "true") closePanel();
    else openPanel();
  }

  // open panel
  function openPanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "true";
  }

  // close panel
  function closePanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "false";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- th list icon -->

<template class="icon_th_list">
  <!-- modified from: https://fontawesome.com/icons/th-list -->
  <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
    <path
      fill="currentColor"
      d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* toc panel */
    #toc_panel {
      box-sizing: border-box;
      position: fixed;
      top: 0;
      left: 0;
      background: #ffffff;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      z-index: 2;
    }

    /* toc panel when closed */
    #toc_panel[data-open="false"] {
      min-width: 60px;
      width: 60px;
      height: 60px;
      border-right: solid 1px #bdbdbd;
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc panel when open */
    #toc_panel[data-open="true"] {
      min-width: 260px;
      max-width: 480px;
      /* keep panel edge consistent distance away from "page" edge */
      width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
      bottom: 0;
      border-right: solid 1px #bdbdbd;
    }

    /* toc panel header */
    #toc_header {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 60px;
      margin: 0;
      padding: 20px;
    }

    /* toc panel header when hovered */
    #toc_header:hover {
      cursor: pointer;
    }

    /* toc panel header when panel open */
    #toc_panel[data-open="true"] > #toc_header {
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc open/close header button */
    #toc_button {
      margin-right: 20px;
    }

    /* hide toc list and header text when closed */
    #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
    #toc_panel[data-open="false"] > #toc_list {
      display: none;
    }

    /* toc list of entries */
    #toc_list {
      box-sizing: border-box;
      width: 100%;
      padding: 20px;
      position: absolute;
      top: calc(60px + 1px);
      bottom: 0;
      overflow: auto;
    }

    /* toc entry, link to section in document */
    .toc_link {
      display: block;
      padding: 5px;
      position: relative;
      font-weight: 600;
      text-decoration: none;
    }

    /* toc entry when hovered or when "viewed" */
    .toc_link:hover,
    .toc_link[data-viewing="true"] {
      background: #f5f5f5;
    }

    /* toc entry, level 1 indentation */
    .toc_link[data-level="1"] {
      margin-left: 0;
    }

    /* toc entry, level 2 indentation */
    .toc_link[data-level="2"] {
      margin-left: 20px;
    }

    /* toc entry, level 3 indentation */
    .toc_link[data-level="3"] {
      margin-left: 40px;
    }

    /* toc entry, level 4 indentation */
    .toc_link[data-level="4"] {
      margin-left: 60px;
    }

    /* toc entry bullets */
    #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
      position: absolute;
      left: -15px;
      top: -1px;
      font-size: 1.5em;
    }

    /* toc entry, level 2 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
      content: "\2022";
    }

    /* toc entry, level 3 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
      content: "\25AB";
    }

    /* toc entry, level 4 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
      content: "-";
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* push <body> ("page") element down to make room for toc icon */
    .toc_body_nudge {
      padding-top: 60px;
    }

    /* toc icon when panel closed and not hovered */
    #toc_panel[data-open="false"]:not(:hover) {
      background: rgba(255, 255, 255, 0.75);
    }
  }

  /* always hide toc panel on print */
  @media only print {
    #toc_panel {
      display: none;
    }
  }
</style>
<!-- 
  Tooltips Plugin

  Makes it such that when the user hovers or focuses a link to a citation or
  figure, a tooltip appears with a preview of the reference content, along with
  arrows to navigate between instances of the same reference in the document.
-->

<script type="module">
  // whether user must click off to close tooltip instead of just un-hovering
  const clickClose = "false";
  // delay (in ms) between opening and closing tooltip
  const delay = "100";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach hover and focus listeners to link
      link.addEventListener("mouseover", onLinkHover);
      link.addEventListener("mouseleave", onLinkUnhover);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("touchend", onLinkTouch);
    }

    // attach mouse, key, and resize listeners to window
    window.addEventListener("mousedown", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("keyup", onKeyUp);
    window.addEventListener("resize", onResize);
  }

  // when link is hovered
  function onLinkHover() {
    // function to open tooltip
    const delayOpenTooltip = function () {
      openTooltip(this);
    }.bind(this);

    // run open function after delay
    this.openTooltipTimer = window.setTimeout(delayOpenTooltip, delay);
  }

  // when mouse leaves link
  function onLinkUnhover() {
    // cancel opening tooltip
    window.clearTimeout(this.openTooltipTimer);

    // don't close on unhover if option specifies
    if (clickClose === "true") return;

    // function to close tooltip
    const delayCloseTooltip = function () {
      // if tooltip open and if mouse isn't over tooltip, close
      const tooltip = document.getElementById("tooltip");
      if (tooltip && !tooltip.matches(":hover")) closeTooltip();
    };

    // run close function after delay
    this.closeTooltipTimer = window.setTimeout(delayCloseTooltip, delay);
  }

  // when link is focused (tabbed to)
  function onLinkFocus(event) {
    openTooltip(this);
  }

  // when link is touched on touch screen
  function onLinkTouch(event) {
    // attempt to force hover state on first tap always, and trigger
    // regular link click (and navigation) on second tap
    if (event.target === document.activeElement) event.target.click();
    else {
      document.activeElement.blur();
      event.target.focus();
    }
    if (event.cancelable) event.preventDefault();
    event.stopPropagation();
    return false;
  }

  // when mouse is clicked anywhere in window
  function onClick(event) {
    closeTooltip();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("tooltip_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("tooltip_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeTooltip();
        break;
    }
  }

  // when window is resized or zoomed
  function onResize() {
    closeTooltip();
  }

  // get all links of types we wish to handle
  function getLinks() {
    const queries = [];
    // exclude buttons, anchor links, toc links, etc
    const exclude =
      ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    queries.push('a[href^="#ref-"]' + exclude); // citation links
    queries.push('a[href^="#fig:"]' + exclude); // figure links
    const query = queries.join(", ");
    return document.querySelectorAll(query);
  }

  // get links with same target, get index of link in set, get total
  // same links
  function getSameLinks(link) {
    const sameLinks = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        sameLinks.push(otherLink);
    }

    return {
      elements: sameLinks,
      index: sameLinks.indexOf(link),
      total: sameLinks.length,
    };
  }

  // open tooltip
  function openTooltip(link) {
    // delete tooltip if it exists, start fresh
    closeTooltip();

    // make tooltip element
    const tooltip = makeTooltip(link);

    // if source couldn't be found and tooltip not made, exit
    if (!tooltip) return;

    // make navbar elements
    const navBar = makeNavBar(link);
    if (navBar) tooltip.firstElementChild.appendChild(navBar);

    // attach tooltip to page
    document.body.appendChild(tooltip);

    // position tooltip
    const position = function () {
      positionTooltip(link);
    };
    position();

    // if tooltip contains images, position again after they've loaded
    const imgs = tooltip.querySelectorAll("img");
    for (const img of imgs) img.addEventListener("load", position);
  }

  // close (delete) tooltip
  function closeTooltip() {
    const tooltip = document.getElementById("tooltip");
    if (tooltip) tooltip.remove();
  }

  // make tooltip
  function makeTooltip(link) {
    // get target element that link points to
    const source = getSource(link);

    // if source can't be found, exit
    if (!source) return;

    // create new tooltip
    const tooltip = document.createElement("div");
    tooltip.id = "tooltip";
    const tooltipContent = document.createElement("div");
    tooltipContent.id = "tooltip_content";
    tooltip.appendChild(tooltipContent);

    // make copy of source node and put in tooltip
    const sourceCopy = makeCopy(source);
    tooltipContent.appendChild(sourceCopy);

    // attach mouse event listeners
    tooltip.addEventListener("click", onTooltipClick);
    tooltip.addEventListener("mousedown", onTooltipClick);
    tooltip.addEventListener("touchstart", onTooltipClick);
    tooltip.addEventListener("mouseleave", onTooltipUnhover);

    // (for interaction with lightbox plugin)
    // transfer click on tooltip copied img to original img
    const sourceImg = source.querySelector("img");
    const sourceCopyImg = sourceCopy.querySelector("img");
    if (sourceImg && sourceCopyImg) {
      const clickImg = function () {
        sourceImg.click();
        closeTooltip();
      };
      sourceCopyImg.addEventListener("click", clickImg);
    }

    return tooltip;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
      "class",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // when tooltip is clicked
  function onTooltipClick(event) {
    // when user clicks on tooltip, stop click from transferring
    // outside of tooltip (eg, click off to close tooltip, or eg click
    // off to unhighlight same refs)
    event.stopPropagation();
  }

  // when tooltip is unhovered
  function onTooltipUnhover(event) {
    if (clickClose === "true") return;

    // make sure new mouse/touch/focus no longer over tooltip or any
    // element within it
    const tooltip = document.getElementById("tooltip");
    if (!tooltip) return;
    if (this.contains(event.relatedTarget)) return;

    closeTooltip();
  }

  // make nav bar to go betwen prev/next instances of same reference
  function makeNavBar(link) {
    // find other links to the same source
    const sameLinks = getSameLinks(link);

    // don't show nav bar when singular reference
    if (sameLinks.total <= 1) return;

    // find prev/next links with same target
    const prevLink = getPrevLink(link, sameLinks);
    const nextLink = getNextLink(link, sameLinks);

    // create nav bar
    const navBar = document.createElement("div");
    navBar.id = "tooltip_nav_bar";
    const text = sameLinks.index + 1 + " of " + sameLinks.total;

    // create nav bar prev/next buttons
    const prevButton = document.createElement("button");
    const nextButton = document.createElement("button");
    prevButton.id = "tooltip_prev_button";
    nextButton.id = "tooltip_next_button";
    prevButton.title =
      "Jump to the previous occurence of this item in the document [←]";
    nextButton.title =
      "Jump to the next occurence of this item in the document [→]";
    prevButton.classList.add("icon_button");
    nextButton.classList.add("icon_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;
    nextButton.innerHTML =
      document.querySelector(".icon_caret_right").innerHTML;
    navBar.appendChild(prevButton);
    navBar.appendChild(document.createTextNode(text));
    navBar.appendChild(nextButton);

    // attach click listeners to buttons
    prevButton.addEventListener("click", function () {
      onPrevNextClick(link, prevLink);
    });
    nextButton.addEventListener("click", function () {
      onPrevNextClick(link, nextLink);
    });

    return navBar;
  }

  // get previous link with same target
  function getPrevLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if < 1
    let index;
    if (sameLinks.index - 1 >= 0) index = sameLinks.index - 1;
    else index = sameLinks.total - 1;
    return sameLinks.elements[index];
  }

  // get next link with same target
  function getNextLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if > total
    let index;
    if (sameLinks.index + 1 <= sameLinks.total - 1) index = sameLinks.index + 1;
    else index = 0;
    return sameLinks.elements[index];
  }

  // get element that is target of link or url hash
  function getSource(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector('[id="' + id + '"]');
    if (!target) return;

    // if ref or figure, modify target to get expected element
    if (id.indexOf("ref-") === 0) target = target.querySelector(":nth-child(2)");
    else if (id.indexOf("fig:") === 0) target = target.querySelector("figure");

    return target;
  }

  // when prev/next arrow button is clicked
  function onPrevNextClick(link, prevNextLink) {
    if (link && prevNextLink)
      goToElement(prevNextLink, window.innerHeight * 0.5);
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);
    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // determine position to place tooltip based on link position in
  // viewport and tooltip size
  function positionTooltip(link, left, top) {
    const tooltipElement = document.getElementById("tooltip");
    if (!tooltipElement) return;

    // get convenient vars for position/dimensions of
    // link/tooltip/page/view
    link = getRectInPage(link);
    const tooltip = getRectInPage(tooltipElement);
    const view = getRectInPage();

    // horizontal positioning
    if (left)
      // use explicit value
      left = left;
    else if (link.left + tooltip.width < view.right)
      // fit tooltip to right of link
      left = link.left;
    else if (link.right - tooltip.width > view.left)
      // fit tooltip to left of link
      left = link.right - tooltip.width;
    // center tooltip in view
    else left = (view.right - view.left) / 2 - tooltip.width / 2;

    // vertical positioning
    if (top)
      // use explicit value
      top = top;
    else if (link.top - tooltip.height > view.top)
      // fit tooltip above link
      top = link.top - tooltip.height;
    else if (link.bottom + tooltip.height < view.bottom)
      // fit tooltip below link
      top = link.bottom;
    else {
      // center tooltip in view
      top = view.top + view.height / 2 - tooltip.height / 2;
      // nudge off of link to left/right if possible
      if (link.right + tooltip.width < view.right) left = link.right;
      else if (link.left - tooltip.width > view.left)
        left = link.left - tooltip.width;
    }

    tooltipElement.style.left = left + "px";
    tooltipElement.style.top = top + "px";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* tooltip container */
    #tooltip {
      position: absolute;
      width: 50%;
      min-width: 240px;
      max-width: 75%;
      z-index: 1;
    }

    /* tooltip content */
    #tooltip_content {
      margin-bottom: 5px;
      padding: 20px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
      overflow-wrap: break-word;
    }

    /* tooltip copy of paragraphs and figures */
    #tooltip_content > p,
    #tooltip_content > figure {
      margin: 0;
      max-height: 320px;
      overflow-y: auto;
    }

    /* tooltip copy of <img> */
    #tooltip_content > figure > img,
    #tooltip_content > figure > svg {
      max-height: 260px;
    }

    /* navigation bar */
    #tooltip_nav_bar {
      margin-top: 10px;
      text-align: center;
    }

    /* navigation bar previous/next buton */
    #tooltip_nav_bar > .icon_button {
      position: relative;
      top: 3px;
    }

    /* navigation bar previous button */
    #tooltip_nav_bar > .icon_button:first-of-type {
      margin-right: 5px;
    }

    /* navigation bar next button */
    #tooltip_nav_bar > .icon_button:last-of-type {
      margin-left: 5px;
    }
  }

  /* always hide tooltip on print */
  @media only print {
    #tooltip {
      display: none;
    }
  }
</style>
<!--
  Analytics Plugin (third-party) 
  
  Copy and paste code from Google Analytics or similar service here.
-->
<!-- 
  Annotations Plugin

  Allows public annotation of the  manuscript. See https://web.hypothes.is/.
-->

<script type="module">
  // configuration
  window.hypothesisConfig = function () {
    return {
      branding: {
        accentColor: "#2196f3",
        appBackgroundColor: "#f8f8f8",
        ctaBackgroundColor: "#f8f8f8",
        ctaTextColor: "#000000",
        selectionFontFamily: "Open Sans, Helvetica, sans serif",
        annotationFontFamily: "Open Sans, Helvetica, sans serif",
      },
    };
  };

  // hypothesis client script
  const embed = "https://hypothes.is/embed.js";
  // hypothesis annotation count query url
  const query = "https://api.hypothes.is/api/search?limit=0&url=";

  // start script
  function start() {
    const button = makeButton();
    document.body.insertBefore(button, document.body.firstChild);
    insertCount(button);
  }

  // make button
  function makeButton() {
    // create button
    const button = document.createElement("button");
    button.id = "hypothesis_button";
    button.innerHTML = document.querySelector(".icon_hypothesis").innerHTML;
    button.title = "Hypothesis annotations";
    button.classList.add("icon_button");

    function onClick(event) {
      onButtonClick(event, button);
    }

    // attach click listeners
    button.addEventListener("click", onClick);

    return button;
  }

  // insert annotations count
  async function insertCount(button) {
    // get annotation count from Hypothesis based on url
    let count = "-";
    try {
      const canonical = document.querySelector('link[rel="canonical"]');
      const location = window.location;
      const url = encodeURIComponent((canonical || location).href);
      const response = await fetch(query + url);
      const json = await response.json();
      count = json.total || "-";
    } catch (error) {
      console.log(error);
    }

    // put count into button
    const counter = document.createElement("span");
    counter.id = "hypothesis_count";
    counter.innerHTML = count;
    button.title = "View " + count + " Hypothesis annotations";
    button.append(counter);
  }

  // when button is clicked
  function onButtonClick(event, button) {
    const script = document.createElement("script");
    script.src = embed;
    document.body.append(script);
    button.remove();
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
  <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
  <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
    <path
      fill="currentColor"
      d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  /* hypothesis activation button */
  #hypothesis_button {
    box-sizing: border-box;
    position: fixed;
    top: 0;
    right: 0;
    width: 60px;
    height: 60px;
    background: #ffffff;
    border-radius: 0;
    border-left: solid 1px #bdbdbd;
    border-bottom: solid 1px #bdbdbd;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
    z-index: 2;
  }

  /* hypothesis button svg */
  #hypothesis_button > svg {
    position: relative;
    top: -4px;
  }

  /* hypothesis annotation count */
  #hypothesis_count {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 5px;
  }

  /* side panel */
  .annotator-frame {
    width: 280px !important;
  }

  /* match highlight color to rest of theme */
  .annotator-highlights-always-on .annotator-hl {
    background-color: #ffeb3b !important;
  }

  /* match focused color to rest of theme */
  .annotator-hl.annotator-hl-focused {
    background-color: #ff8a65 !important;
  }

  /* match bucket bar color to rest of theme */
  .annotator-bucket-bar {
    background: #f5f5f5 !important;
  }

  /* always hide button, toolbar, and tooltip on print */
  @media only print {
    #hypothesis_button {
      display: none;
    }

    .annotator-frame {
      display: none !important;
    }

    hypothesis-adder {
      display: none !important;
    }
  }
</style>
<!-- 
  Mathjax Plugin (third-party) 

  Allows the proper rendering of math/equations written in LaTeX.
  See https://www.mathjax.org/.
-->

<script type="text/x-mathjax-config">
  // configuration
  MathJax.Hub.Config({
    "CommonHTML": { linebreaks: { automatic: true } },
    "HTML-CSS": { linebreaks: { automatic: true } },
    "SVG": { linebreaks: { automatic: true } },
    "fast-preview": { disabled: true }
  });
</script>

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
  integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
  crossorigin="anonymous"
></script>

<style>
  /* mathjax containers */
  .math.display > span:not(.MathJax_Preview) {
    /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
    display: flex !important;
    overflow-x: auto !important;
    overflow-y: hidden !important;
    justify-content: center;
    align-items: center;
    margin: 0 !important;
  }

  /* right click menu */
  .MathJax_Menu {
    border-radius: 5px !important;
    border: solid 1px #bdbdbd !important;
    box-shadow: none !important;
  }

  /* equation auto-number */
  span[id^="eq:"] > span.math.display + span {
    font-weight: 600;
  }

  /* equation */
  span[id^="eq:"] > span.math.display > span {
    /* nudge to make room for equation auto-number and anchor */
    margin-right: 60px !important;
  }
</style>
</body>
</html>
